## Documentation

### Introduction

This report is intended for interpretation of inherited DNA variants implicated with cancer susceptibility and inherited cancer syndromes. Variants in *Class 1-5* are limited to a selected set of known cancer predisposition genes, for which the report lists __ONLY__ those variants that are

1. Previously classified in ClinVar (five-level significance scheme: pathogenic/likely pathogenic/VUS/likely benign/benign), or
2. Coding variants not recorded in ClinVar with germline population frequency below the user-defined threshold, i.e. 
    * Minor allele frequency (MAF) < `r cps_report[['metadata']][['config']][['popgen']][['maf_upper_threshold']]`) in the user-defined population set in the gnomAD database


### Annotation resources

The analysis performed in the cancer genome report is based on the following underlying tools and knowledge resources:

* __PCGR databundle version__

  * __`r cps_report$metadata$pcgr_db_release$bundle_version$version`__


* __Software__ 
```{r list_software, echo=F,results='asis'}
for (n in names(cps_report$metadata$pcgr_db_release)) {
  if (n == "mosdepth" | n == "mutationalpatterns" | n == "vcf2maf" | n == "bundle_version") {
    next
  }
  if (cps_report$metadata$pcgr_db_release[[n]]$resource_type == "software") {
    s <- paste0("    * [", cps_report$metadata$pcgr_db_release[[n]]$name, "](", cps_report$metadata$pcgr_db_release[[n]]$url, ") - ", cps_report$metadata$pcgr_db_release[[n]]$description, " (", cps_report$metadata$pcgr_db_release[[n]]$version, ")")
    cat(s, sep = "\n")
  }
}
```
<br><br>

* __Databases/datasets__ 
```{r list_db, echo=F,results='asis'}
for (n in names(cps_report$metadata$pcgr_db_release)) {
  if (cps_report$metadata$pcgr_db_release[[n]]$resource_type != "software") {
    if (n != "cbmdb" & n != "chembl" & n != "corum" & n != "icgc" & n != "bundle_version" & n != "tcga" & n != "cosmic" & n != "dgidb" & n != "ct" &
      n != "kegg" & n != "onekg" & n != "tcga-pcdm" & n != "opentargets") {
      s <- paste0("    * [", cps_report$metadata$pcgr_db_release[[n]]$name, "](", cps_report$metadata$pcgr_db_release[[n]]$url, ") - ", cps_report$metadata$pcgr_db_release[[n]]$description, " (", cps_report$metadata$pcgr_db_release[[n]]$version, ")")
      cat(s, sep = "\n")
    }
  }
}
```

### Variant classification

<br>

All coding, <b>non-ClinVar</b> variants in the set of cancer predisposition genes have been classified according to a <i>five-level pathogenicity scheme</i> (coined <b>CPSR_CLASSIFICATION</b> in the tables above). The scheme has the same five levels as those employed by ClinVar, e.g. pathogenic/likely pathogenic/VUS/likely benign/benign. The classification performed by CPSR is rule-based, implementing refined ACMG criteria, many of which were outlined in [SherLoc](https://www.invitae.com/en/variant-classification/) ([Nykamp et al., Genetics in Medicine, 2017](https://www.ncbi.nlm.nih.gov/pubmed/28492532)). Important attributes of cancer predisposition genes, such as mode of inheritance and mechanism of disease (loss-of-function), have been harvested from [Genomics England PanelApp](https://panelapp.genomicsengland.co.uk/), [Maxwell et al., Am J Hum Genet, 2016](https://www.ncbi.nlm.nih.gov/pubmed/27153395), and [Huang et al., Cell, 2018](https://www.ncbi.nlm.nih.gov/pubmed/29625052)

The ACMG criteria listed in the table below form the basis for the <b>CPSR_CLASSIFICATION</b> implemented in CPSR. Specifically, the <i>score</i> column indicates how much each evidence item contribute to either of the two pathogenicity poles (positive values indicate pathogenic support, negative values indicate benign support). Evidence score along each pole ('B' and 'P') are aggregated, and if there is conflicting or little evidence it will be classified as a VUS. The contribution of ACMG evidence items pr. variant can be seen in the <b>CPSR_CLASSIFICATION_CODE</b> and <b>CPSR_CLASSIFICATION_DOC</b> variables.
<br><br>

#### Calibration of classification thresholds 

How do we derive the variant classification (<i>P, LP, VUS, LB, B</i>) from the aggregated variant pathogenicity score (<i>CPSR_PATHOGENICITY_SCORE</i>)?

We calibrated the thresholds for conversion of pathogenicity scores to categorical variant classification using high-quality ClinVar-classified variants in cancer predisposition genes (see details in [BioRxiv preprint](https://doi.org/10.1101/846089)). The following thresholds are currently used to assign classifications based on pathogenicity scores:

<br>

`r knitr::kable(pcgrr::cpsr_acmg[['score2tier']])`

<br><br>

In the table below, a detailed description of all evidence criteria that are currently used for variant classification in CPSR (green elements indicate criteria that contribute with a benign effect, red elements contribute with a pathogenic effect):<br><br>


```{r acmg_evidence, echo = F, eval = T}
data <- dplyr::filter(
  pcgrr::cpsr_acmg[["evidence_codes"]],
  cpsr_evidence_code != "ACMG_BS2_1" &
    cpsr_evidence_code != "ACMG_BS2_2" &
    cpsr_evidence_code != "ACMG_BS2_3"
) %>%
  dplyr::select(-category) %>%
  dplyr::rename(pole = pathogenicity_pole, category = category_long, score = path_score) %>%
  dplyr::arrange(pole)

DT::datatable(data,
  escape = F, extensions = c("Buttons", "Responsive"),
  options = list(
    pageLength = 10,
    buttons = c("csv", "excel"),
    dom = "Bfrtip"
  )
) %>%
  DT::formatStyle("cpsr_evidence_code", "pole",
    color = "white",
    backgroundColor = DT::styleEqual(c("P", "B"), c("#9E0142", "#077009"))
  )
```


