

## Summary of findings

```{r donut_chart, echo = F, fig.width = width_donut, fig.height = height_donut, eval = cps_report[['content']][['snv_indel']][['eval']]}
p1 <- cpsr::summary_donut_chart(
  cps_report$content$snv_indel$variant_set$tsv, plot_type = "ClinVar")
p2 <- cpsr::summary_donut_chart(
  cps_report$content$snv_indel$variant_set$tsv, plot_type = "Other")

p <- NULL
if (!is.null(p1) & !is.null(p2)) {
  p <- ggpubr::ggarrange(p1, p2, labels = c("", ""), ncol = 2, nrow = 1)
}
if (!is.null(p1) & is.null(p2)) {
  p <- p1
}
if (is.null(p1) & !is.null(p2)) {
  p <- p2
}
p
```


### Variant statistics 

```{r num_cpg_targets, eval = TRUE}

primary_targets <- 
  cps_report[['settings']][['conf']][['gene_panel']][['panel_genes']] |>
  dplyr::filter(PRIMARY_TARGET == T) |>
  dplyr::select(ENTREZGENE) |>
  dplyr::distinct()
```

Variant numbers in the targeted cancer predisposition genes (n = `r NROW(primary_targets)`)

* Number of SNVs: __`r cps_report[['content']][['snv_indel']][['v_stat_cpg']][['n_snv']]`__
* Number of InDels: __`r cps_report[['content']][['snv_indel']][['v_stat_cpg']][['n_indel']]`__
* Number of protein-coding variants: __`r cps_report[['content']][['snv_indel']][['v_stat_cpg']][['n_coding']]`__

<br><br>
