
## Variant classification

```{r prepare_classified_variant_data, echo=F, results='asis'}

dtable_pageLength <- 6
dtable_scrollY <- "700px"

show_class_filters <- list()
missing_class_items <- list()
tot_variants <- list()
for (c in c("class1", "class2", "class3", "class4", "class5")) {
  show_class_filters[[c]] <- list()
  missing_class_items[[c]] <- list()
  tot_variants[[c]] <- list()
  for (m in c("Other", "ClinVar")) {
    tot_variants[[c]][[m]] <- 0
    show_class_filters[[c]][[m]] <- F
    missing_class_items[[c]][[m]] <- T
    if (NROW(cps_report[["content"]][["snv_indel"]][["disp"]][[c]]) > 0) {
      tot_variants[[c]][[m]] <- NROW(cps_report[["content"]][["snv_indel"]][["disp"]][[c]] %>%
        dplyr::filter(CPSR_CLASSIFICATION_SOURCE == m))
    }
    if (tot_variants[[c]][[m]] > 0) {
      show_class_filters[[c]][[m]] <- T
      missing_class_items[[c]][[m]] <- F
    }
  }
}

tag_gnomad <- cps_report[["metadata"]][["config"]][["popgen"]][["vcftag_gnomad"]]
desc_gnomad <- cps_report[["metadata"]][["config"]][["popgen"]][["popdesc_gnomad"]]
formula_gnomad <- as.formula(paste0("~", rlang::sym(tag_gnomad)))
```


<!--
## <span style="color:`r pcgrr::color_palette$pathogenicity$values[1]`">Class 5 - <b>Pathogenic</b> variants</span> {.tabset .tabset-pills .nav-justified}
-->

### Class 5 - <b>Pathogenic</b></span> variants {.tabset .tabset-pills .nav-justified}

<br>

   *  A total of n = __`r tot_variants[['class5']][['ClinVar']]`__ variants are registered with a <i>Pathogenic</i> clinical significance in ClinVar.
   *  A total of n = __`r tot_variants[['class5']][['Other']]`__ <i><b>non-ClinVar</b></i> variants (i.e. not registered in ClinVar) are classified with a <i>Pathogenic</i> significance by CPSR (ACMG criteria - based on population frequency and variant effect).

<br>


```{r active_tab_class5, echo = F, results = "asis"}
if (missing_class_items[["class5"]][["ClinVar"]] == F | (missing_class_items[["class5"]][["Other"]] == T & missing_class_items[["class5"]][["ClinVar"]] == T)) {
  cat("")
  cat("#### ClinVar {.active}")
  cat("")
} else {
  cat("")
  cat("#### ClinVar")
  cat("")
}
```


```{r class5_cpsr_clinvar, echo=F, results = 'asis', eval = show_class_filters[['class5']][['ClinVar']]}

cat("\n")
htmltools::br()


variants_class5_clinvar <-
  dplyr::filter(cps_report[["content"]][["snv_indel"]][["disp"]][["class5"]], CPSR_CLASSIFICATION_SOURCE == "ClinVar")

if (NROW(variants_class5_clinvar) > 2000) {
  cat("<b>NOTE - only considering top 2000 variants (due to limitations with client-side tables) </b><br>", sep = "\n")
  cat("<br>")
  variants_class5_clinvar <-
    head(variants_class5_clinvar, 2000)
}

variants_class5_clinvar <- variants_class5_clinvar %>%
  dplyr::select(-c(
    CPSR_CLASSIFICATION, CPSR_PATHOGENICITY_SCORE,
    CPSR_CLASSIFICATION_DOC, CPSR_CLASSIFICATION_CODE
  ))

variants_class5_1 <- crosstalk::SharedData$new(variants_class5_clinvar)
crosstalk::bscols(
  list(
    crosstalk::filter_select("CONSEQUENCE", "Consequence", variants_class5_1, ~CONSEQUENCE),
    crosstalk::filter_checkbox("GENOTYPE", "Genotype", variants_class5_1, ~GENOTYPE),
    crosstalk::filter_select("SYMBOL", "Gene", variants_class5_1, ~SYMBOL)
  ),
  list(
    crosstalk::filter_select("CLINVAR_PHENOTYPE", "ClinVar phenotype(s)", variants_class5_1, ~CLINVAR_PHENOTYPE),
    crosstalk::filter_slider("CLINVAR_REVIEW_STATUS_STARS", "ClinVar review status stars", variants_class5_1, ~CLINVAR_REVIEW_STATUS_STARS, min = 0, max = 4, step = 1, ticks = T),
    crosstalk::filter_select("CLINVAR_CONFLICTED", "ClinVar conflicting interpretations", variants_class5_1, ~CLINVAR_CONFLICTED),
    crosstalk::filter_slider(tag_gnomad, paste0("MAF gnomAD (", desc_gnomad, ")"), variants_class5_1, formula_gnomad, sep = "", ticks = F)
  )
)



htmltools::br()
htmltools::br()

DT::datatable(variants_class5_1,
  escape = F, extensions = c("Buttons", "Responsive"),
  options = list(
    pageLength = dtable_pageLength,
    # scrollY = dtable_scrollY,
    scrollCollapse = T,
    # autoWidth = T,
    buttons = c("csv", "excel"),
    dom = "Bfrtip",
    initComplete = DT::JS(
      "function(settings, json) {",
      "$(this.api().table().header()).css({'background-color':'#9E0142', 'color': '#fff'});",
      "}"
    )
  )
)

htmltools::br()
htmltools::br()
htmltools::br()
```

```{r active2_tab_class5, echo = F, results = "asis"}
if (missing_class_items[["class5"]][["ClinVar"]] == T & missing_class_items[["class5"]][["Other"]] == F) {
  cat("")
  cat("#### Non-ClinVar {.active}")
  cat("")
} else {
  cat("")
  cat("#### Non-ClinVar")
  cat("")
}
```


```{r class5_cpsr_other, echo=F, results = 'asis', eval = show_class_filters[['class5']][['Other']]}

variants_class5_other <-
  dplyr::filter(cps_report[["content"]][["snv_indel"]][["disp"]][["class5"]], CPSR_CLASSIFICATION_SOURCE == "Other")

if (NROW(variants_class5_other) > 2000) {
  cat("<b>NOTE - only considering top 2000 variants (ranked according to CPSR classification score) </b><br>", sep = "\n")
  cat("<br>")
  variants_class5_other <-
    head(variants_class5_other, 2000)
}

variants_class5_other <- variants_class5_other %>%
  dplyr::select(-c(
    CLINVAR, CLINVAR_REVIEW_STATUS_STARS, CLINVAR_VARIANT_ORIGIN,
    CLINVAR_CLASSIFICATION, CLINVAR_CONFLICTED, CLINVAR_PHENOTYPE
  ))


variants_class5_2 <- crosstalk::SharedData$new(variants_class5_other)
crosstalk::bscols(
  list(
    crosstalk::filter_select("CONSEQUENCE", "Consequence", variants_class5_2, ~CONSEQUENCE),
    crosstalk::filter_checkbox("GENOTYPE", "Genotype", variants_class5_2, ~GENOTYPE),
    crosstalk::filter_select("SYMBOL", "Gene", variants_class5_2, ~SYMBOL)
  ),
  list(
    crosstalk::filter_select("CPSR_CLASSIFICATION_CODE", "CPSR classification (ACMG criteria codes)", variants_class5_2, ~CPSR_CLASSIFICATION_CODE),
    crosstalk::filter_slider("CPSR_PATHOGENICITY_SCORE", "CPSR pathogenicity score", variants_class5_2, ~CPSR_PATHOGENICITY_SCORE, step = 0.5, ticks = T),
    crosstalk::filter_slider(tag_gnomad, paste0("MAF gnomAD (", desc_gnomad, ")"), variants_class5_2, formula_gnomad, sep = "", ticks = F)
  )
)



htmltools::br()
htmltools::br()

DT::datatable(variants_class5_2,
  escape = F, extensions = c("Buttons", "Responsive"),
  options = list(
    pageLength = dtable_pageLength,
    # scrollY = dtable_scrollY,
    scrollCollapse = T,
    buttons = c("csv", "excel"),
    dom = "Bfrtip",
    initComplete = DT::JS(
      "function(settings, json) {",
      "$(this.api().table().header()).css({'background-color':'#9E0142', 'color': '#fff'});",
      "}"
    )
  )
)

htmltools::br()
htmltools::br()
htmltools::br()
```


<!--
## <span style="color:`r pcgrr::color_palette$pathogenicity$values[2]`">Class 4 - <b>Likely Pathogenic</b> variants</span> {.tabset .tabset-pills .nav-justified}
-->
### Class 4 - <b>Likely Pathogenic</b> variants {.tabset .tabset-pills .nav-justified}

<br>

*  A total of n = __`r tot_variants[['class4']][['ClinVar']]`__ variants are recorded with a <i>Likely Pathogenic</i> clinical significance in ClinVar.
*  A total of n = __`r tot_variants[['class4']][['Other']]`__ <i><b>non-ClinVar</b></i> variants (i.e. not registered in ClinVar) are classified with a <i>Likely Pathogenic</i> significance by CPSR (ACMG criteria - based on population frequency and variant effect).

<br><br>


```{r active_tab_class4, echo = F, results = "asis"}
if (missing_class_items[["class4"]][["ClinVar"]] == F | (missing_class_items[["class4"]][["Other"]] == T & missing_class_items[["class4"]][["ClinVar"]] == T)) {
  cat("")
  cat("#### ClinVar {.active}")
  cat("")
} else {
  cat("")
  cat("#### ClinVar")
  cat("")
}
```

```{r class4_cpsr_clinvar, echo=F, results = 'asis', eval = show_class_filters[['class4']][['ClinVar']]}

cat("\n")
htmltools::br()

variants_class4_clinvar <-
  dplyr::filter(cps_report[["content"]][["snv_indel"]][["disp"]][["class4"]], CPSR_CLASSIFICATION_SOURCE == "ClinVar")

if (NROW(variants_class4_clinvar) > 2000) {
  cat("<b>NOTE - only considering top 2000 variants (due to limitations with client-side tables) </b><br>", sep = "\n")
  cat("<br>")
  variants_class4_clinvar <-
    head(variants_class4_clinvar, 2000)
}

variants_class4_clinvar <- variants_class4_clinvar %>%
  dplyr::select(-c(
    CPSR_CLASSIFICATION, CPSR_PATHOGENICITY_SCORE,
    CPSR_CLASSIFICATION_DOC, CPSR_CLASSIFICATION_CODE
  ))

variants_class4_1 <- crosstalk::SharedData$new(variants_class4_clinvar)
crosstalk::bscols(
  list(
    crosstalk::filter_select("CONSEQUENCE", "Consequence", variants_class4_1, ~CONSEQUENCE),
    crosstalk::filter_checkbox("GENOTYPE", "Genotype", variants_class4_1, ~GENOTYPE),
    crosstalk::filter_select("SYMBOL", "Gene", variants_class4_1, ~SYMBOL)
  ),
  list(
    crosstalk::filter_select("CLINVAR_PHENOTYPE", "ClinVar phenotype(s)", variants_class4_1, ~CLINVAR_PHENOTYPE),
    crosstalk::filter_slider("CLINVAR_REVIEW_STATUS_STARS", "ClinVar review status stars", variants_class4_1, ~CLINVAR_REVIEW_STATUS_STARS, min = 0, max = 4, step = 1, ticks = T),
    crosstalk::filter_select("CLINVAR_CONFLICTED", "ClinVar conflicting interpretations", variants_class4_1, ~CLINVAR_CONFLICTED),
    crosstalk::filter_slider(tag_gnomad, paste0("MAF gnomAD (", desc_gnomad, ")"), variants_class4_1, formula_gnomad, sep = "", ticks = F)
  )
)



htmltools::br()
htmltools::br()

DT::datatable(variants_class4_1,
  escape = F, extensions = c("Buttons", "Responsive"),
  options = list(
    pageLength = dtable_pageLength,
    # scrollY = dtable_scrollY,
    scrollCollapse = T,
    buttons = c("csv", "excel"),
    dom = "Bfrtip",
    initComplete = DT::JS(
      "function(settings, json) {",
      "$(this.api().table().header()).css({'background-color':'#D53E4F', 'color': '#fff'});",
      "}"
    )
  )
)
htmltools::br()
htmltools::br()
htmltools::br()
```

```{r active2_tab_class4, echo = F, results = "asis"}
if (missing_class_items[["class4"]][["ClinVar"]] == T & missing_class_items[["class4"]][["Other"]] == F) {
  cat("")
  cat("#### Non-ClinVar {.active}")
  cat("")
} else {
  cat("")
  cat("#### Non-ClinVar")
  cat("")
}
```

```{r class4_cpsr_other, echo=F, results = 'asis', eval = show_class_filters[['class4']][['Other']]}

cat("\n")
htmltools::br()

variants_class4_other <-
  dplyr::filter(cps_report[["content"]][["snv_indel"]][["disp"]][["class4"]], CPSR_CLASSIFICATION_SOURCE == "Other")

if (NROW(variants_class4_other) > 2000) {
  cat("<b>NOTE - only considering top 2000 variants (ranked according to CPSR classification score) </b><br>", sep = "\n")
  cat("<br>")
  variants_class4_other <-
    head(variants_class4_other, 2000)
}

variants_class4_other <- variants_class4_other %>%
  dplyr::select(-c(
    CLINVAR, CLINVAR_REVIEW_STATUS_STARS, CLINVAR_VARIANT_ORIGIN,
    CLINVAR_CLASSIFICATION, CLINVAR_CONFLICTED, CLINVAR_PHENOTYPE
  ))

variants_class4_2 <- crosstalk::SharedData$new(variants_class4_other)
crosstalk::bscols(
  list(
    crosstalk::filter_select("CONSEQUENCE", "Consequence", variants_class4_2, ~CONSEQUENCE),
    crosstalk::filter_checkbox("GENOTYPE", "Genotype", variants_class4_2, ~GENOTYPE),
    crosstalk::filter_select("SYMBOL", "Gene", variants_class4_2, ~SYMBOL)
  ),
  list(
    crosstalk::filter_select("CPSR_CLASSIFICATION_CODE", "CPSR classification (ACMG criteria codes)", variants_class4_2, ~CPSR_CLASSIFICATION_CODE),
    crosstalk::filter_slider("CPSR_PATHOGENICITY_SCORE", "CPSR pathogenicity score", variants_class4_2, ~CPSR_PATHOGENICITY_SCORE, step = 0.5, ticks = T),
    crosstalk::filter_slider(tag_gnomad, paste0("MAF gnomAD (", desc_gnomad, ")"), variants_class4_2, formula_gnomad, sep = "", ticks = F)
  )
)



htmltools::br()
htmltools::br()

DT::datatable(variants_class4_2,
  escape = F, extensions = c("Buttons", "Responsive"),
  options = list(
    pageLength = dtable_pageLength,
    # scrollY = dtable_scrollY,
    scrollCollapse = T,
    buttons = c("csv", "excel"),
    dom = "Bfrtip",
    initComplete = DT::JS(
      "function(settings, json) {",
      "$(this.api().table().header()).css({'background-color':'#D53E4F', 'color': '#fff'});",
      "}"
    )
  )
)

htmltools::br()
htmltools::br()
htmltools::br()
```


<!--
## <span style="color:`r pcgrr::color_palette$pathogenicity$values[3]`">Class 3 - <b>Variants of Uncertain Significance (VUS)</b></span> {.tabset .tabset-pills .nav-justified}
-->

### Class 3 - Variants of <b>Uncertain Significance (VUS)</b> {.tabset .tabset-pills .nav-justified}


<br>

*  A total of n = __`r tot_variants[['class3']][['ClinVar']]`__ variants are recorded with a <i>VUS</i> clinical significance in ClinVar.
*  A total of n = __`r tot_variants[['class3']][['Other']]`__ <i><b>non-ClinVar</b></i> variants (i.e. not registered in ClinVar) are classified with a <i>VUS</i> significance by CPSR (ACMG criteria - based on population frequency and variant effect).

<br><br>


```{r active_tab_class3, echo = F, results = "asis"}
if (missing_class_items[["class3"]][["ClinVar"]] == F | (missing_class_items[["class3"]][["Other"]] == T & missing_class_items[["class3"]][["ClinVar"]] == T)) {
  cat("")
  cat("#### ClinVar {.active}")
  cat("")
} else {
  cat("")
  cat("#### ClinVar")
  cat("")
}
```

```{r class3_cpsr_clinvar, echo=F, results = 'asis', eval = show_class_filters[['class3']][['ClinVar']]}

cat("\n")
htmltools::br()

variants_class3_clinvar <-
  dplyr::filter(cps_report[["content"]][["snv_indel"]][["disp"]][["class3"]], CPSR_CLASSIFICATION_SOURCE == "ClinVar")

if (NROW(variants_class3_clinvar) > 2000) {
  cat("<b>NOTE - only considering top 2000 variants (due to limitations with client-side tables) </b><br>", sep = "\n")
  cat("<br>")
  variants_class3_clinvar <-
    head(variants_class3_clinvar, 2000)
}

variants_class3_clinvar <- variants_class3_clinvar %>%
  dplyr::select(-c(
    CPSR_CLASSIFICATION, CPSR_PATHOGENICITY_SCORE,
    CPSR_CLASSIFICATION_DOC, CPSR_CLASSIFICATION_CODE
  ))

variants_class3_1 <- crosstalk::SharedData$new(variants_class3_clinvar)
crosstalk::bscols(
  list(
    crosstalk::filter_select("CONSEQUENCE", "Consequence", variants_class3_1, ~CONSEQUENCE),
    crosstalk::filter_checkbox("GENOTYPE", "Genotype", variants_class3_1, ~GENOTYPE),
    crosstalk::filter_select("SYMBOL", "Gene", variants_class3_1, ~SYMBOL),
    crosstalk::filter_slider("GERP_SCORE", "Genomic conservation score (GERP)", variants_class3_1, ~GERP_SCORE,
      min = -12.3, max = 6.17, ticks = T
    ),
    crosstalk::filter_slider(tag_gnomad, paste0("MAF gnomAD (", desc_gnomad, ")"), variants_class3_1, formula_gnomad, sep = "", ticks = F)
  ),
  list(
    crosstalk::filter_select("miRNA_TARGET_HIT", "miRNA target gain/loss", variants_class3_1, ~miRNA_TARGET_HIT),
    crosstalk::filter_select("TF_BINDING_SITE_VARIANT", "TF binding site alteration", variants_class3_1, ~TF_BINDING_SITE_VARIANT),
    crosstalk::filter_select("CLINVAR_PHENOTYPE", "ClinVar phenotype(s)", variants_class3_1, ~CLINVAR_PHENOTYPE),
    crosstalk::filter_slider("CLINVAR_REVIEW_STATUS_STARS", "ClinVar review status stars", variants_class3_1, ~CLINVAR_REVIEW_STATUS_STARS, min = 0, max = 4, step = 1, ticks = T),
    crosstalk::filter_select("CLINVAR_CONFLICTED", "ClinVar conflicting interpretations", variants_class3_1, ~CLINVAR_CONFLICTED)
  )
)



htmltools::br()
htmltools::br()

DT::datatable(variants_class3_1,
  escape = F, extensions = c("Buttons", "Responsive"),
  options = list(
    pageLength = dtable_pageLength,
    # scrollY = dtable_scrollY,
    scrollCollapse = T,
    buttons = c("csv", "excel"),
    dom = "Bfrtip",
    initComplete = DT::JS(
      "function(settings, json) {",
      "$(this.api().table().header()).css({'background-color':'#000000', 'color': '#fff'});",
      "}"
    )
  )
)

htmltools::br()
htmltools::br()
htmltools::br()
```

```{r active2_tab_class3, echo = F, results = "asis"}
if (missing_class_items[["class3"]][["ClinVar"]] == T & missing_class_items[["class3"]][["Other"]] == F) {
  cat("")
  cat("#### Non-ClinVar {.active}")
  cat("")
} else {
  cat("")
  cat("#### Non-ClinVar")
  cat("")
}
```


```{r vus_caution_note, echo=F, results = "asis", include = show_class_filters[['class3']][['Other']]}

htmltools::br()

cat('<ul><li><b>IMPORTANT NOTE:</b> Non-ClinVar variants with a CPSR pathogenicity score of 2.0 (maximum score for VUS in CPSR) represent the most challenging collection of variants when it comes to classification. Based on calibration of classification score thresholds with high-confident ClinVar variants, we discovered a heterogeneous mix of _Likely Pathogenic_, _Pathogenic_, and _VUS_ variants that were all attributed with this score. Henceforth, we recommend that these variants should be prioritized for careful examination, and any such cases are <mark style="background-color:#ff7518; font-weight:bold; color:black">conveniently higlighted</mark> in the table below.</li></ul>\n')

htmltools::br()
```



```{r class3_cpsr_other, echo=F, results = 'asis', eval = show_class_filters[['class3']][['Other']]}

cat("\n")
htmltools::br()

variants_class3_other <-
  dplyr::filter(cps_report[["content"]][["snv_indel"]][["disp"]][["class3"]], CPSR_CLASSIFICATION_SOURCE == "Other")

if (NROW(variants_class3_other) > 2000) {
  cat("<b>NOTE - only considering top 2000 variants (ranked according to CPSR classification score) </b><br>", sep = "\n")
  cat("<br>")
  variants_class3_other <-
    head(variants_class3_other, 2000)
}

variants_class3_other <- variants_class3_other %>%
  dplyr::select(-c(
    CLINVAR, CLINVAR_REVIEW_STATUS_STARS, CLINVAR_VARIANT_ORIGIN,
    CLINVAR_CLASSIFICATION, CLINVAR_CONFLICTED, CLINVAR_PHENOTYPE
  ))

variants_class3_2 <- crosstalk::SharedData$new(variants_class3_other)
crosstalk::bscols(
  list(
    crosstalk::filter_select("CONSEQUENCE", "Consequence", variants_class3_2, ~CONSEQUENCE),
    crosstalk::filter_checkbox("GENOTYPE", "Genotype", variants_class3_2, ~GENOTYPE),
    crosstalk::filter_select("SYMBOL", "Gene", variants_class3_2, ~SYMBOL),
    crosstalk::filter_slider("GERP_SCORE", "Genomic conservation score (GERP)", variants_class3_2, ~GERP_SCORE,
      min = -12.3, max = 6.17, ticks = T
    )
  ),
  list(
    crosstalk::filter_select("miRNA_TARGET_HIT", "miRNA target gain/loss", variants_class3_2, ~miRNA_TARGET_HIT),
    crosstalk::filter_select("TF_BINDING_SITE_VARIANT", "TF binding site alteration", variants_class3_2, ~TF_BINDING_SITE_VARIANT),
    crosstalk::filter_select("CPSR_CLASSIFICATION_CODE", "CPSR classification (ACMG criteria codes)", variants_class3_2, ~CPSR_CLASSIFICATION_CODE),
    crosstalk::filter_slider("CPSR_PATHOGENICITY_SCORE", "CPSR pathogenicity score", variants_class3_2, ~CPSR_PATHOGENICITY_SCORE, step = 0.5, ticks = T),
    crosstalk::filter_slider(tag_gnomad, paste0("MAF gnomAD (", desc_gnomad, ")"), variants_class3_2, formula_gnomad, sep = "", ticks = F)
  )
)



htmltools::br()
htmltools::br()

DT::datatable(variants_class3_2,
  escape = F, extensions = c("Buttons", "Responsive"),
  options = list(
    pageLength = dtable_pageLength,
    # scrollY = dtable_scrollY,
    scrollCollapse = T,
    buttons = c("csv", "excel"),
    dom = "Bfrtip",
    initComplete = DT::JS(
      "function(settings, json) {",
      "$(this.api().table().header()).css({'background-color':'#000000', 'color': '#fff'});",
      "}"
    )
  )
) %>%
  DT::formatStyle(
    color = "black",
    columns = "SYMBOL",
    valueColumns = "CPSR_PATHOGENICITY_SCORE",
    backgroundColor = DT::styleEqual(
      c(-1.0, -0.5, 0, 0.5, 1.0, 1.5, 2.0),
      c("#FFFFFF", "#FFFFFF", "#FFFFFF", "#FFFFFF", "#FFFFFF", "#FFFFFF", "#ff7518")
    )
  )

htmltools::br()
htmltools::br()
htmltools::br()
```

<!--
## <span style="color:`r pcgrr::color_palette$pathogenicity$values[4]`">Class 2 - <b>Likely Benign</b> variants</span> {.tabset .tabset-pills .nav-justified}
-->
### Class 2 - <b>Likely Benign</b> variants {.tabset .tabset-pills .nav-justified}

<br>

*  A total of n = __`r tot_variants[['class2']][['ClinVar']]`__ variants are recorded with a <i>Likely Benign</i> clinical significance in ClinVar.
*  A total of n = __`r tot_variants[['class2']][['Other']]`__ <i><b>Other</b></i> variants (i.e. not registered in ClinVar) are classified with a <i>Likely Benign</i> significance by CPSR (ACMG criteria - based on population frequency and variant effect).

<br><br>


```{r active_tab_class2, echo = F, results = "asis"}
if (missing_class_items[["class2"]][["ClinVar"]] == F | (missing_class_items[["class2"]][["Other"]] == T & missing_class_items[["class2"]][["ClinVar"]] == T)) {
  cat("")
  cat("#### ClinVar {.active}")
  cat("")
} else {
  cat("")
  cat("#### ClinVar")
  cat("")
}
```

```{r class2_cpsr_clinvar, echo=F, results = 'asis', eval = show_class_filters[['class2']][['ClinVar']]}

cat("\n")
htmltools::br()

variants_class2_clinvar <-
  dplyr::filter(cps_report[["content"]][["snv_indel"]][["disp"]][["class2"]], CPSR_CLASSIFICATION_SOURCE == "ClinVar")

if (NROW(variants_class2_clinvar) > 2000) {
  cat("<b>NOTE - only considering top 2000 variants (due to limitations with client-side tables) </b><br>", sep = "\n")
  cat("<br>")
  variants_class2_clinvar <-
    head(variants_class2_clinvar, 2000)
}

variants_class2_clinvar <- variants_class2_clinvar %>%
  dplyr::select(-c(
    CPSR_CLASSIFICATION, CPSR_PATHOGENICITY_SCORE,
    CPSR_CLASSIFICATION_DOC, CPSR_CLASSIFICATION_CODE
  ))

variants_class2_1 <- crosstalk::SharedData$new(variants_class2_clinvar)
crosstalk::bscols(
  list(
    crosstalk::filter_select("CONSEQUENCE", "Consequence", variants_class2_1, ~CONSEQUENCE),
    crosstalk::filter_checkbox("GENOTYPE", "Genotype", variants_class2_1, ~GENOTYPE),
    crosstalk::filter_select("SYMBOL", "Gene", variants_class2_1, ~SYMBOL)
  ),
  list(
    crosstalk::filter_select("CLINVAR_PHENOTYPE", "ClinVar phenotype(s)", variants_class2_1, ~CLINVAR_PHENOTYPE),
    crosstalk::filter_slider("CLINVAR_REVIEW_STATUS_STARS", "ClinVar review status stars", variants_class2_1, ~CLINVAR_REVIEW_STATUS_STARS, min = 0, max = 4, step = 1, ticks = T),
    crosstalk::filter_select("CLINVAR_CONFLICTED", "ClinVar conflicting interpretations", variants_class2_1, ~CLINVAR_CONFLICTED),
    crosstalk::filter_slider(tag_gnomad, paste0("MAF gnomAD (", desc_gnomad, ")"), variants_class2_1, formula_gnomad, sep = "", ticks = F)
  )
)



htmltools::br()
htmltools::br()

DT::datatable(variants_class2_1,
  escape = F, extensions = c("Buttons", "Responsive"),
  options = list(
    pageLength = dtable_pageLength,
    # scrollY = dtable_scrollY,
    scrollCollapse = T,
    buttons = c("csv", "excel"),
    dom = "Bfrtip",
    initComplete = DT::JS(
      "function(settings, json) {",
      "$(this.api().table().header()).css({'background-color':'#78C679', 'color': '#fff'});",
      "}"
    )
  )
)

htmltools::br()
htmltools::br()
htmltools::br()
```

```{r active2_tab_class2, echo = F, results = "asis"}
if (missing_class_items[["class2"]][["ClinVar"]] == T & missing_class_items[["class2"]][["Other"]] == F) {
  cat("")
  cat("#### Non-ClinVar {.active}")
  cat("")
} else {
  cat("")
  cat("#### Non-ClinVar")
  cat("")
}
```

```{r class2_cpsr_other, echo=F, results = 'asis', eval = show_class_filters[['class2']][['Other']]}

cat("\n")
htmltools::br()

variants_class2_other <-
  dplyr::filter(cps_report[["content"]][["snv_indel"]][["disp"]][["class2"]], CPSR_CLASSIFICATION_SOURCE == "Other")

if (NROW(variants_class2_other) > 2000) {
  cat("<b>NOTE - only considering top 2000 variants (ranked according to CPSR classification score) </b><br>", sep = "\n")
  cat("<br>")
  variants_class2_other <-
    head(variants_class2_other, 2000)
}

variants_class2_other <- variants_class2_other %>%
  dplyr::select(-c(
    CLINVAR, CLINVAR_REVIEW_STATUS_STARS, CLINVAR_VARIANT_ORIGIN,
    CLINVAR_CLASSIFICATION, CLINVAR_CONFLICTED, CLINVAR_PHENOTYPE
  ))

variants_class2_2 <- crosstalk::SharedData$new(variants_class2_other)
crosstalk::bscols(
  list(
    crosstalk::filter_select("CONSEQUENCE", "Consequence", variants_class2_2, ~CONSEQUENCE),
    crosstalk::filter_checkbox("GENOTYPE", "Genotype", variants_class2_2, ~GENOTYPE),
    crosstalk::filter_select("SYMBOL", "Gene", variants_class2_2, ~SYMBOL)
  ),
  list(
    crosstalk::filter_select("CPSR_CLASSIFICATION_CODE", "CPSR classification (ACMG criteria codes)", variants_class2_2, ~CPSR_CLASSIFICATION_CODE),
    crosstalk::filter_slider("CPSR_PATHOGENICITY_SCORE", "CPSR pathogenicity score", variants_class2_2, ~CPSR_PATHOGENICITY_SCORE, step = 0.5, ticks = T),
    crosstalk::filter_slider(tag_gnomad, paste0("MAF gnomAD (", desc_gnomad, ")"), variants_class2_2, formula_gnomad, sep = "", ticks = F)
  )
)



htmltools::br()
htmltools::br()

DT::datatable(variants_class2_2,
  escape = F, extensions = c("Buttons", "Responsive"),
  options = list(
    pageLength = dtable_pageLength,
    # scrollY = dtable_scrollY,
    scrollCollapse = T,
    buttons = c("csv", "excel"),
    dom = "Bfrtip",
    initComplete = DT::JS(
      "function(settings, json) {",
      "$(this.api().table().header()).css({'background-color':'#78C679', 'color': '#fff'});",
      "}"
    )
  )
)
htmltools::br()
htmltools::br()
htmltools::br()
```

<!--
## <span style="color:`r pcgrr::color_palette$pathogenicity$values[5]`">Class 1 - <b>Benign</b> variants</span> {.tabset .tabset-pills .nav-justified}
-->

### Class 1 - <b>Benign</b> variants {.tabset .tabset-pills .nav-justified}


<br>

*  A total of n = __`r tot_variants[['class1']][['ClinVar']]`__ variants are recorded with a <i>Benign</i> clinical significance in ClinVar.
*  A total of n = __`r tot_variants[['class1']][['Other']]`__ <i><b>Other</b></i> variants (i.e. not registered in ClinVar) are classified with a <i>Benign</i> significance by CPSR (ACMG criteria - based on population frequency and variant effect).

<br><br>


```{r active_tab_class1, echo = F, results = "asis"}
if (missing_class_items[["class1"]][["ClinVar"]] == F | (missing_class_items[["class1"]][["Other"]] == T & missing_class_items[["class1"]][["ClinVar"]] == T)) {
  cat("")
  cat("#### ClinVar {.active}")
  cat("")
} else {
  cat("")
  cat("#### ClinVar")
  cat("")
}
```

```{r class1_cpsr_clinvar, echo=F, results = 'asis', eval = show_class_filters[['class1']][['ClinVar']]}

cat("\n")
htmltools::br()

variants_class1_clinvar <-
  dplyr::filter(cps_report[["content"]][["snv_indel"]][["disp"]][["class1"]], CPSR_CLASSIFICATION_SOURCE == "ClinVar")

if (NROW(variants_class1_clinvar) > 2000) {
  cat("<b>NOTE - only considering top 2000 variants (due to limitations with client-side tables) </b><br>", sep = "\n")
  cat("<br>")
  variants_class1_clinvar <-
    head(variants_class1_clinvar, 2000)
}

variants_class1_clinvar <- variants_class1_clinvar %>%
  dplyr::select(-c(
    CPSR_CLASSIFICATION, CPSR_PATHOGENICITY_SCORE,
    CPSR_CLASSIFICATION_DOC, CPSR_CLASSIFICATION_CODE
  ))

variants_class1_1 <- crosstalk::SharedData$new(variants_class1_clinvar)
crosstalk::bscols(
  list(
    crosstalk::filter_select("CONSEQUENCE", "Consequence", variants_class1_1, ~CONSEQUENCE),
    crosstalk::filter_checkbox("GENOTYPE", "Genotype", variants_class1_1, ~GENOTYPE),
    crosstalk::filter_select("SYMBOL", "Gene", variants_class1_1, ~SYMBOL)
  ),
  list(
    crosstalk::filter_select("CLINVAR_PHENOTYPE", "ClinVar phenotype(s)", variants_class1_1, ~CLINVAR_PHENOTYPE),
    crosstalk::filter_slider("CLINVAR_REVIEW_STATUS_STARS", "ClinVar review status stars", variants_class1_1, ~CLINVAR_REVIEW_STATUS_STARS, min = 0, max = 4, step = 1, ticks = T),
    crosstalk::filter_select("CLINVAR_CONFLICTED", "ClinVar conflicting interpretations", variants_class1_1, ~CLINVAR_CONFLICTED),
    crosstalk::filter_slider(tag_gnomad, paste0("MAF gnomAD (", desc_gnomad, ")"), variants_class1_1, formula_gnomad, sep = "", ticks = F)
  )
)



htmltools::br()
htmltools::br()

DT::datatable(variants_class1_1,
  escape = F,
  extensions = c("Buttons", "Responsive"),
  options = list(
    pageLength = dtable_pageLength,
    # scrollY = dtable_scrollY,
    scrollCollapse = T,
    buttons = c("csv", "excel"),
    dom = "Bfrtip",
    initComplete = DT::JS(
      "function(settings, json) {",
      "$(this.api().table().header()).css({'background-color':'#077009', 'color': '#fff'});",
      "}"
    )
  )
)

htmltools::br()
htmltools::br()
htmltools::br()
```

```{r active2_tab_class1, echo = F, results = "asis"}
if (missing_class_items[["class1"]][["ClinVar"]] == T & missing_class_items[["class1"]][["Other"]] == F) {
  cat("")
  cat("#### Non-ClinVar {.active}")
  cat("")
} else {
  cat("")
  cat("#### Non-ClinVar")
  cat("")
}
```

```{r class1_cpsr_other, echo=F, results = 'asis', eval = show_class_filters[['class1']][['Other']]}

cat("\n")
htmltools::br()

variants_class1_other <-
  dplyr::filter(cps_report[["content"]][["snv_indel"]][["disp"]][["class1"]], CPSR_CLASSIFICATION_SOURCE == "Other")

if (NROW(variants_class1_other) > 2000) {
  cat("<b>NOTE - only considering top 2000 variants (ranked according to CPSR classification score) </b><br>", sep = "\n")
  cat("<br>")
  variants_class1_other <-
    head(variants_class1_other, 2000)
}

variants_class1_other <- variants_class1_other %>%
  dplyr::select(-c(
    CLINVAR, CLINVAR_REVIEW_STATUS_STARS, CLINVAR_VARIANT_ORIGIN,
    CLINVAR_CLASSIFICATION, CLINVAR_CONFLICTED, CLINVAR_PHENOTYPE
  ))

variants_class1_2 <- crosstalk::SharedData$new(variants_class1_other)
crosstalk::bscols(
  list(
    crosstalk::filter_select("CONSEQUENCE", "Consequence", variants_class1_2, ~CONSEQUENCE),
    crosstalk::filter_checkbox("GENOTYPE", "Genotype", variants_class1_2, ~GENOTYPE),
    crosstalk::filter_select("SYMBOL", "Gene", variants_class1_2, ~SYMBOL)
  ),
  list(
    crosstalk::filter_select("CPSR_CLASSIFICATION_CODE", "CPSR classification (ACMG criteria codes)", variants_class1_2, ~CPSR_CLASSIFICATION_CODE),
    crosstalk::filter_slider("CPSR_PATHOGENICITY_SCORE", "CPSR pathogenicity score", variants_class1_2, ~CPSR_PATHOGENICITY_SCORE, step = 0.5, ticks = T),
    crosstalk::filter_slider(tag_gnomad, paste0("MAF gnomAD (", desc_gnomad, ")"), variants_class1_2, formula_gnomad, sep = "", ticks = F)
  )
)



htmltools::br()
htmltools::br()

DT::datatable(variants_class1_2,
  escape = F, extensions = c("Buttons", "Responsive"),
  options = list(
    pageLength = dtable_pageLength,
    # scrollY = dtable_scrollY,
    scrollCollapse = T,
    buttons = c("csv", "excel"),
    dom = "Bfrtip",
    initComplete =
      DT::JS(
        "function(settings, json) {",
        "$(this.api().table().header()).css({'background-color':'#077009', 'color': '#fff'});",
        "}"
      )
  )
)

htmltools::br()
htmltools::br()
htmltools::br()
```


