## Settings

```{r on_off_settings, echo=F}

gencode_versions <- unlist(stringr::str_split_fixed(
  stringr::str_replace(
    cps_report$metadata$pcgr_db_release$gencode$version, "release ", ""
  ),
  "/", 2
))[1, ]

gencode_version <- gencode_versions[2]
if (cps_report[["metadata"]][["genome_assembly"]] == "grch38") {
  gencode_version <- gencode_versions[1]
}

transcript_set_gencode <- paste0("GENCODE - basic set (v", gencode_version, ")")
if (cps_report[["metadata"]][["config"]][["other"]][["vep_gencode_all"]] == T) {
  transcript_set_gencode <- paste0("GENCODE - all transcripts (v", gencode_version, ")")
}
```

### Sample metadata

* Sample name &nbsp;:&nbsp; <span style="background-color:#000000; color:white"><b>&nbsp;`r cps_report[['metadata']][['sample_name']]`&nbsp;</b></span>
<br>

### Report configuration 

The report is generated with _[cpsr](https://github.com/sigven/cpsr)_ version `r cps_report[['metadata']][['cpsr_version']]`, ( _[pcgr](https://github.com/sigven/pcgr)_ version `r cps_report[['metadata']][['pcgr_version']]`), using the following key settings:

* Genome assembly: __`r cps_report[['metadata']][['genome_assembly']]`__
* Report settings
  * R Markdown theme (Bootstrap): __`r cps_report[['metadata']][['config']][['visual']][['report_theme']]`__
  * Datatable display: __`r cps_report[['metadata']][['config']][['visual']][['table_display']]`__
  * Non-floating TOC: __`r cps_report[['metadata']][['config']][['visual']][['nonfloating_toc']]`__
* Control population gnomAD: __`r cps_report[['metadata']][['config']][['popgen']][['popdesc_gnomad']]`__
* Upper MAF threshold (gnomAD) for unclassified variants included in report:  __`r cps_report[['metadata']][['config']][['popgen']][['maf_upper_threshold']]`__
* Ignore non-proteincoding variants in report: __`r cps_report[['metadata']][['config']][['ignore_noncoding']]`__
* Generate CPSR variant classifications also for existing ClinVar variants: __`r cps_report[['metadata']][['config']][['classify_all']]`__
* Ignore ClinVar variants in target genes if they are reported only for non-cancer phenotypes: 
__`r cps_report[['metadata']][['config']][['clinvar_ignore_noncancer']]`__
* Include [secondary findings](https://www.ncbi.nlm.nih.gov/clinvar/docs/acmg/) in report: __`r cps_report[['metadata']][['config']][['secondary_findings']]`__
* Include GWAS hits in report: __`r cps_report[['metadata']][['config']][['gwas']][['run']]`__
    * Minimum p-value for association: __`r cps_report[['metadata']][['config']][['gwas']][['p_value_min']]`__
* Variant Effect Predictor (VEP) settings:
  * Transcript set: __`r transcript_set_gencode`__
  * Transcript pick order: __`r cps_report[['metadata']][['config']][['other']][['vep_pick_order']]`__
  * Regulatory regions annotation: __`r cps_report[['metadata']][['config']][['other']][['vep_regulatory']]`__
