<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="CPSR is a computational workflow that interprets germline variants identified from
             next-generation sequencing in the context of cancer predisposition. The workflow is
             integrated with the framework that underlies the Personal Cancer Genome Reporter
             (PCGR - https://github.com/sigven/pcgr). While PCGR is intended for reporting
             and analysis of somatic variants detected in a tumor, CPSR is intended for reporting
             and ranking of germline variants in protein-coding genes that are implicated in cancer
             predisposition and inherited cancer syndromes.">
<title>Cancer Predisposition Sequencing Reporter (CPSR) • CPSR</title>
<script src="deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="deps/bootstrap-5.1.0/bootstrap.min.css" rel="stylesheet">
<script src="deps/bootstrap-5.1.0/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.rawgit.com/afeld/bootstrap-toc/v1.0.1/dist/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="pkgdown.js"></script><meta property="og:title" content="Cancer Predisposition Sequencing Reporter (CPSR)">
<meta property="og:description" content="CPSR is a computational workflow that interprets germline variants identified from
             next-generation sequencing in the context of cancer predisposition. The workflow is
             integrated with the framework that underlies the Personal Cancer Genome Reporter
             (PCGR - https://github.com/sigven/pcgr). While PCGR is intended for reporting
             and analysis of somatic variants detected in a tumor, CPSR is intended for reporting
             and ranking of germline variants in protein-coding genes that are implicated in cancer
             predisposition and inherited cancer syndromes.">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-dark navbar-expand-lg bg-primary"><div class="container">
    
    <a class="navbar-brand me-2" href="index.html">CPSR</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.7.7</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="active nav-item">
  <a class="nav-link" href="index.html">Intro</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="articles/installation.html">Installation</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="articles/running.html">Running</a>
</li>
<li class="nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="articles/input.html">Input files</a>
    <a class="dropdown-item" href="articles/virtual_panels.html">Virtual gene panels</a>
    <a class="dropdown-item" href="articles/variant_classification.html">ACMG classification</a>
    <a class="dropdown-item" href="articles/output.html">Output files</a>
    <a class="dropdown-item" href="articles/annotation_resources.html">Annotation resources</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="articles/CHANGELOG.html">CHANGELOG</a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="nav-link" href="reference/index.html">Reference</a>
</li>
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/sigven/cpsr/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-home">
<div class="row">
  <main id="main" class="col-md-9"><p><br><br></p>
<div class="section level3 unlisted">
<h3 class="unlisted" id="cancer-predisposition-sequencing-reporter">Cancer Predisposition Sequencing Reporter<a class="anchor" aria-label="anchor" href="#cancer-predisposition-sequencing-reporter"></a>
</h3>
<p><br></p>
<p>The <em>Cancer Predisposition Sequencing Reporter (CPSR)</em> is a
computational workflow that <strong>interprets germline
variants</strong> identified from next-generation sequencing <strong>in
the context of cancer predisposition</strong>. The workflow is
integrated with the framework that underlies <a href="https://github.com/sigven/pcgr" class="external-link">Personal Cancer Genome Reporter -
PCGR</a>. While <em>PCGR</em> is primarily intended for reporting and
analysis of somatic variants detected in a tumor, <em>CPSR</em> is
intended for reporting and ranking of germline variants in
protein-coding genes that are implicated in cancer predisposition and
inherited cancer syndromes.</p>
<p><em>CPSR</em> accepts a query file with raw germline variant
calls(SNVs/InDels) from a single sample, encoded in the <a href="https://samtools.github.io/hts-specs/VCFv4.2.pdf" class="external-link">VCF format</a>.
Through the use several different <a href="articles/virtual_panels.html">virtual cancer predisposition gene
panels</a>, the user can flexibly put a restriction on which genes and
findings are displayed in the cancer predisposition report.</p>
<p>Snapshots of sections in the cancer predisposition genome report:</p>
<p><img src="img/cpsr_views.png"></p>
<p>The software performs extensive variant annotation on the selected
geneset and produces an interactive HTML report, in which the user can
investigate:</p>
<ul>
<li>
<strong>ClinVar variants</strong> - pre-classified variants
according to a five-level tier scheme in ClinVar (Pathogenic to
Benign)</li>
<li>
<strong>Novel variants</strong> - classified by CPSR through ACMG
criteria (variant frequency levels and functional effects) into to a
five-level tier scheme (Pathogenic to Benign)</li>
<li>
<strong>Variant biomarkers</strong> - cancer predisposition variants
with reported implications for prognosis, diagnosis or therapeutic
regimens</li>
<li>
<strong>Secondary findings (optional)</strong> - pathogenic ClinVar
variants in the <a href="https://www.ncbi.nlm.nih.gov/clinvar/docs/acmg/" class="external-link">ACMG recommended
list for reporting of secondary findings</a>
</li>
<li>
<strong>GWAS hits (optional)</strong> - variants overlapping with
previously identified hits in genome-wide association studies (GWAS) of
cancer phenotypes (i.e. low to moderate risk conferring alleles), using
<a href="https://www.ebi.ac.uk/gwas/" class="external-link">NHGRI-EBI Catalog of published
genome-wide association studies</a> as the underlying source.</li>
</ul>
<p>The variant sets can be interactively explored and filtered further
through different types of filters (phenotypes, genes, variant
consequences, population MAF etc.). Importantly, the unclassified
(i.e. non-ClinVar) variants are assigned a <em>pathogenicity score</em>
based on the aggregation of scores according to previously established
<a href="https://www.ncbi.nlm.nih.gov/pubmed/25741868" class="external-link">ACMG
criteria</a>. The ACMG criteria includes cancer-specific criteria, as
outlined and specified in several previous studies (<a href="https://www.ncbi.nlm.nih.gov/pubmed/29625052" class="external-link">Huang et al.,
<em>Cell</em>, 2018</a>; <a href="https://www.ncbi.nlm.nih.gov/pubmed/28492532" class="external-link">Nykamp et al.,
<em>Genet Med.</em>, 2017</a>; <a href="https://www.ncbi.nlm.nih.gov/pubmed/27153395" class="external-link">Maxwell et al.,
<em>Am J Hum Genet.</em>, 2016</a>; <a href="https://www.ncbi.nlm.nih.gov/pubmed/27181684" class="external-link">Amendola et al.,
<em>Am J Hum Genet.</em>, 2016</a>). See also <a href="https://github.com/sigven/cpsr#related-work" class="external-link"><em>Related
work</em></a> below).</p>
</div>
<div class="section level3">
<h3 id="news">News<a class="anchor" aria-label="anchor" href="#news"></a>
</h3>
<ul>
<li>
<em>February 2022</em>: <strong>x.x.x release</strong>
<ul>
<li>Complete restructure of code and Conda installation routines,
contributed largely by the great <a href="https://github.com/pdiakumis" class="external-link">@pdiakumis</a>
</li>
<li>Updated bundle (ClinVar, CancerMine, UniprotKB, PanelApp, CIViC,
GWAS catalog)</li>
<li>Software upgrade (VEP, R/BioConductor)</li>
<li>New documentation site (<a href="https://sigven.github.io/cpsr" class="uri">https://sigven.github.io/cpsr</a>)</li>
</ul>
</li>
<li>
<em>June 30th 2021</em>: <strong>0.6.2 release</strong>
<ul>
<li>Updated bundle (ClinVar, CancerMine, UniprotKB, PanelApp, CIViC,
GWAS catalog)</li>
<li>Software upgrade (VEP, R/BioConductor)</li>
<li><a href="http://cpsr.readthedocs.io/en/latest/CHANGELOG.html" class="external-link">CHANGELOG</a></li>
</ul>
</li>
<li>
<em>November 30th 2020</em>: <strong>0.6.1 release</strong>
<ul>
<li>Updated bundle (ClinVar, CancerMine, UniprotKB, CIViC, GWAS
catalog)</li>
<li><a href="http://cpsr.readthedocs.io/en/latest/CHANGELOG.html" class="external-link">CHANGELOG</a></li>
</ul>
</li>
</ul>
</div>
<div class="section level3">
<h3 id="example-report">Example report<a class="anchor" aria-label="anchor" href="#example-report"></a>
</h3>
<p><a href="https://doi.org/10.5281/zenodo.5035960" class="external-link"><img src="https://zenodo.org/badge/DOI/10.5281/zenodo.5035960.svg" alt="DOI"></a></p>
</div>
<div class="section level3">
<h3 id="getting-started">Getting started<a class="anchor" aria-label="anchor" href="#getting-started"></a>
</h3>
<ul>
<li><p><a href="articles/installation.html">Installation
instructions</a></p></li>
<li><p><a href="articles/running.html#example-run">Run through an
example</a></p></li>
<li>
<p>Learn more about</p>
<ol style="list-style-type: decimal">
<li>Details regarding <a href="articles/input.html">CPSR input
files</a>, and how they should be formatted</li>
<li>The types and contents of <a href="articles/output.html">CPSR output
files</a>
</li>
<li>
<a href="articles/variant_classification.html">ACMG variant
classification procedure</a> used in CPSR</li>
<li>The list of <a href="articles/virtual_panels.html">virtual gene
panels</a> available in CPSR</li>
</ol>
</li>
</ul>
</div>
<div class="section level3">
<h3 id="related-work">Related work<a class="anchor" aria-label="anchor" href="#related-work"></a>
</h3>
<ul>
<li><a href="https://github.com/ding-lab/CharGer" class="external-link">CharGer -
Characterization of Germline variants</a></li>
<li><a href="https://pathoman.mskcc.org/" class="external-link">PathoMan - Pathogenicity of
Mutation Analyzer (Beta)</a></li>
<li><a href="https://www.invitae.com/en/variant-classification/" class="external-link">SherLoc
- Variant classification</a></li>
</ul>
</div>
<div class="section level3">
<h3 id="contact">Contact<a class="anchor" aria-label="anchor" href="#contact"></a>
</h3>
<p><a href="mailto:sigven@ifi.uio.no" class="email">sigven@ifi.uio.no</a></p>
</div>

  </main><aside class="col-md-3"><div class="links">
<h2 data-toc-skip>Links</h2>
<ul class="list-unstyled">
<li><a href="https://github.com/sigven/cpsr/" class="external-link">Browse source code</a></li>
<li><a href="https://github.com/sigven/cpsr/issues" class="external-link">Report a bug</a></li>
</ul>
</div>

<div class="license">
<h2 data-toc-skip>License</h2>
<ul class="list-unstyled">
<li><a href="LICENSE.html">Full license</a></li>
<li><small><a href="https://opensource.org/licenses/mit-license.php" class="external-link">MIT</a> + file <a href="LICENSE-text.html">LICENSE</a></small></li>
</ul>
</div>


<div class="citation">
<h2 data-toc-skip>Citation</h2>
<ul class="list-unstyled">
<li><p><a href="authors.html#citation">Citing CPSR</a></p></li>
</ul>
</div>

<div class="developers">
<h2 data-toc-skip>Developers</h2>
<ul class="list-unstyled">
<li>
<a href="https://github.com/sigven" class="external-link">Sigve Nakken</a> <br><small class="roles"> Author, maintainer </small> <a href="https://orcid.org/0000-0001-8468-2050" target="orcid.widget" aria-label="ORCID" class="external-link"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
<li>
<a href="https://github.com/pdiakumis" class="external-link">Peter Diakumis</a> <br><small class="roles"> Author, contributor </small> <a href="https://orcid.org/0000-0002-7502-7545" target="orcid.widget" aria-label="ORCID" class="external-link"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
</ul>
</div>



  </aside>
</div>


    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by <a href="https://github.com/sigven" class="external-link">Sigve Nakken</a>,
<a href="https://github.com/pdiakumis" class="external-link">Peter Diakumis</a>.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a>
2.0.2.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
