
## Genomic biomarkers 

- Sample variants (__class 4/5__) in the targeted cancer predisposition genes that overlap with reported clinical evidence items from [CIViC](https://civicdb.org) and [CGI biomarker database](https://www.cancergenomeinterpreter.org/biomarkers) are listed. 
- Note the same variant may overlap multiple evidence items, pending on the _disease context_, _therapeutic context_, or underlying evidence source (e.g. publication)
- Total number of _clinical evidence items_ that have been matched with query variants:
    * Predisposing: __`r NROW(cps_report[['content']][['snv_indel']][['clin_eitem']][['predisposing']][['any']])`__ evidence items
    * Predictive: __`r NROW(cps_report[['content']][['snv_indel']][['clin_eitem']][['predictive']][['any']])`__ evidence items
    * Prognostic: __`r NROW(cps_report[['content']][['snv_indel']][['clin_eitem']][['prognostic']][['any']])`__ evidence items
    * Diagnostic: __`r NROW(cps_report[['content']][['snv_indel']][['clin_eitem']][['diagnostic']][['any']])`__ evidence items
    * Functional: __`r NROW(cps_report[['content']][['snv_indel']][['clin_eitem']][['functional']][['any']])`__ evidence items

<br><br>


```{r prepare_germline_data}
#| echo: false
#| output: asis
#| eval: true

show_germline_filters <- list()
missing_germline_items <- list()
biomarker_present <- F
for (type in names(cps_report$content$snv_indel$clin_eitem)) {
  show_germline_filters[[type]] <- F
  missing_germline_items[[type]] <- T
  if (NROW(cps_report[["content"]][["snv_indel"]][["clin_eitem"]][[type]][['any']]) > 0) {
    show_germline_filters[[type]] <- T
    missing_germline_items[[type]] <- F
    biomarker_present <- T
  }
  if (NROW(cps_report[["content"]][["snv_indel"]][["clin_eitem"]][[type]][['any']]) > 2000){
    cps_report[["content"]][["snv_indel"]][["clin_eitem"]][[type]][['any']] <- 
      head(cps_report[["content"]][["snv_indel"]][["clin_eitem"]][[type]][['any']], 2000)
  }
}
```


```{r biomarker_note}
#| echo: false
#| output: asis
#| include: !expr biomarker_present

cat("<b>NOTE:</b> Biomarkers registered in CIViC/CGI are provided at different _resolutions_ (i.e. filter <b>Biomarker mapping</b>). The accuracy of a match between a variant in the tumor sample and a reported biomarker will vary accordingly (highlighted by gene symbols with different color backgrounds):\n\n")

cat('<ul><li>Biomarker match at the <mark style="background-color:black; font-weight:bold; color:white">&nbsp; genomic position/amino acid/codon level &nbsp;</mark></li>')
cat(paste0('<br><li>Biomarker match at the <mark style="background-color:', cps_report$settings$conf$visual_reporting$color_palette[["warning"]][["values"]][1], '; color:white; font-weight:bold">&nbsp; exon/gene level&nbsp; </mark></li></ul>\n'))

htmltools::br()
```

::: {.panel-tabset}

```{r active_tab_cpsr_predisposing}
#| echo: false
#| output: asis
#| eval: true

if (missing_germline_items[["predisposing"]] == F |
  (missing_germline_items[["diagnostic"]] == T &
    missing_germline_items[["prognostic"]] == T &
    missing_germline_items[["predictive"]] == T)) {
  cat("")
  cat("### Predisposing {.active}")
  cat("")
} else {
  cat("")
  cat("### Predisposing")
  cat("")
}
```


<br>

```{r germline_predisposing}
#| eval: !expr show_germline_filters$predisposing
#| output: asis
#| echo: false

variants_germline_predisposing_shared <- crosstalk::SharedData$new(
  cps_report[["content"]][["snv_indel"]][["clin_eitem"]][["predisposing"]][["any"]])
crosstalk::bscols(
  list(
    crosstalk::filter_select("CANCER_TYPE", "Cancer type", variants_germline_predisposing_shared, ~CANCER_TYPE),
    crosstalk::filter_select("CLINICAL_SIGNIFICANCE", "Clinical significance", variants_germline_predisposing_shared, ~CLINICAL_SIGNIFICANCE),
    crosstalk::filter_select("EVIDENCE_LEVEL", "Evidence level", variants_germline_predisposing_shared, ~EVIDENCE_LEVEL)
  ),
  list(
    crosstalk::filter_select("SYMBOL", "Gene", variants_germline_predisposing_shared, ~SYMBOL),
    crosstalk::filter_select("BIOMARKER_MAPPING", "Biomarker mapping", variants_germline_predisposing_shared, ~BIOMARKER_MAPPING)
  )
)

##cat("The table below lists all variant-evidence item associations:", sep = "\n")
htmltools::br()
htmltools::br()

DT::datatable(variants_germline_predisposing_shared,
  escape = F, extensions = c("Buttons", "Responsive"),
  options =
    list(
      buttons = c("csv", "excel"),
      pageLength = 6,
      dom = "Bfrtip"
    )) |>
   DT::formatStyle("EVIDENCE_LEVEL",
    backgroundColor = DT::styleEqual(
      cps_report$settings$conf$visual_reporting$color_palette$clinical_evidence$levels,
      cps_report$settings$conf$visual_reporting$color_palette$clinical_evidence$values
    )
  ) |>
  DT::formatStyle(
    color = "white", "SYMBOL", "BIOMARKER_MAPPING", fontWeight = "bold", `text-align` = "center",
    backgroundColor = DT::styleEqual(
      c("genomic", "hgvsp", "codon", "hgvsp_nonprincipal", "exon", "gene"),
      c("#000", "#000", "#000","#000",
        cps_report$settings$conf$visual_reporting$color_palette[["warning"]][["values"]][1],
        cps_report$settings$conf$visual_reporting$color_palette[["warning"]][["values"]][1])
    )
  )
```

```{r germline_predisposing_missing}
#| echo: false
#| output: asis
#| eval: !expr missing_germline_items$predisposing

cat("<i>No variant-evidence item associations found.</i>", sep = "\n")
cat("\n")
```

<br><br><br>


```{r active_tab_cpsr_predictive}
#| echo: false
#| output: asis

if (missing_germline_items[["predictive"]] == F &
  missing_germline_items[["predisposing"]] == T) {
  cat("")
  cat("### Predictive {.active}")
  cat("")
} else {
  cat("")
  cat("### Predictive")
  cat("")
}
```


<br>

```{r germline_predictive}
#| eval: !expr show_germline_filters$predictive
#| output: asis
#| echo: false

variants_germline_predictive_shared <- crosstalk::SharedData$new(
  cps_report[["content"]][["snv_indel"]][["clin_eitem"]][["predictive"]][["any"]])
crosstalk::bscols(
  list(
    crosstalk::filter_select("CANCER_TYPE", "Cancer type", variants_germline_predictive_shared, ~CANCER_TYPE),
    crosstalk::filter_select("CLINICAL_SIGNIFICANCE", "Clinical significance", variants_germline_predictive_shared, ~CLINICAL_SIGNIFICANCE),
    crosstalk::filter_select("EVIDENCE_LEVEL", "Evidence level", variants_germline_predictive_shared, ~EVIDENCE_LEVEL)
  ),
  list(
    crosstalk::filter_select("SYMBOL", "Gene", variants_germline_predictive_shared, ~SYMBOL),
    crosstalk::filter_select("BIOMARKER_MAPPING", "Biomarker mapping", variants_germline_predictive_shared, ~BIOMARKER_MAPPING),
    crosstalk::filter_select("THERAPEUTIC_CONTEXT", "Therapeutic context", variants_germline_predictive_shared, ~THERAPEUTIC_CONTEXT)
  )
)

##cat("The table below lists all variant-evidence item associations:", sep = "\n")
htmltools::br()
htmltools::br()

DT::datatable(variants_germline_predictive_shared, escape = F, extensions = c("Buttons", "Responsive"), options = list(
  buttons = c("csv", "excel"),
  pageLength = 6, dom = "Bfrtip")) |>
   DT::formatStyle("EVIDENCE_LEVEL",
    backgroundColor = DT::styleEqual(
      cps_report$settings$conf$visual_reporting$color_palette$clinical_evidence$levels,
      cps_report$settings$conf$visual_reporting$color_palette$clinical_evidence$values
    )
  ) |>
  DT::formatStyle(
    color = "white", "SYMBOL", "BIOMARKER_MAPPING", fontWeight = "bold", `text-align` = "center",
    backgroundColor = DT::styleEqual(
      c("genomic", "hgvsp", "codon", "hgvsp_nonprincipal", "exon", "gene"),
      c("#000", "#000", "#000","#000",
        cps_report$settings$conf$visual_reporting$color_palette[["warning"]][["values"]][1],
        cps_report$settings$conf$visual_reporting$color_palette[["warning"]][["values"]][1])
    )
  )
```

```{r germline_predictive_missing}
#| echo: false
#| output: asis
#| eval: !expr missing_germline_items$predictive

cat("<i>No variant-evidence item associations found.</i>", sep = "\n")
cat("\n")
```

<br><br><br>


```{r active_tab_cpsr_prognostic}
#| echo: false
#| output: asis

if (missing_germline_items[["prognostic"]] == F &
  missing_germline_items[["predisposing"]] == T &
  missing_germline_items[["predictive"]] == T) {
  cat("")
  cat("### Prognostic {.active}")
  cat("")
} else {
  cat("")
  cat("### Prognostic")
  cat("")
}
```


<br>

```{r germline_prognostic}
#| eval: !expr show_germline_filters$prognostic
#| output: asis
#| echo: false

variants_germline_prognostic_shared <- crosstalk::SharedData$new(
  cps_report[["content"]][["snv_indel"]][["clin_eitem"]][["prognostic"]][['any']])
crosstalk::bscols(
  list(
    crosstalk::filter_select("SYMBOL", "Gene", variants_germline_prognostic_shared, ~SYMBOL),
    crosstalk::filter_select("CANCER_TYPE", "Cancer type", variants_germline_prognostic_shared, ~CANCER_TYPE),
    crosstalk::filter_select("CLINICAL_SIGNIFICANCE", "Clinical significance", variants_germline_prognostic_shared, ~CLINICAL_SIGNIFICANCE)
  ),
  list(
    crosstalk::filter_select("EVIDENCE_LEVEL", "Evidence level", variants_germline_prognostic_shared, ~EVIDENCE_LEVEL),
    crosstalk::filter_select("BIOMARKER_MAPPING", "Biomarker mapping", variants_germline_prognostic_shared, ~BIOMARKER_MAPPING)
  )
)
##cat("The table below lists all variant-evidence item associations:", sep = "\n")
htmltools::br()
htmltools::br()

DT::datatable(variants_germline_prognostic_shared, escape = F, extensions = c("Buttons", "Responsive"), options = list(
  buttons = c("csv", "excel"),
  pageLength = 6, dom = "Bfrtip")) |>
   DT::formatStyle("EVIDENCE_LEVEL",
    backgroundColor = DT::styleEqual(
      cps_report$settings$conf$visual_reporting$color_palette$clinical_evidence$levels,
      cps_report$settings$conf$visual_reporting$color_palette$clinical_evidence$values
    )
  ) |>
 DT::formatStyle(
    color = "white", "SYMBOL", "BIOMARKER_MAPPING", fontWeight = "bold", `text-align` = "center",
    backgroundColor = DT::styleEqual(
      c("genomic", "hgvsp", "codon", "hgvsp_nonprincipal", "exon", "gene"),
      c("#000", "#000", "#000","#000",
        cps_report$settings$conf$visual_reporting$color_palette[["warning"]][["values"]][1],
        cps_report$settings$conf$visual_reporting$color_palette[["warning"]][["values"]][1])
    )
  )
```

```{r germline_prognostic_missing}
#| echo: false
#| output: asis
#| eval: !expr missing_germline_items$prognostic

cat("<i>No variant-evidence item associations found.</i>", sep = "\n")
cat("\n")
```

<br><br><br>

```{r active_tab_cpsr_diagnostic}
#| echo: false
#| output: asis

if (missing_germline_items[["diagnostic"]] == F &
  missing_germline_items[["prognostic"]] == T &
  missing_germline_items[["predisposing"]] == T &
  missing_germline_items[["predictive"]] == T) {
  cat("")
  cat("### Diagnostic {.active}")
  cat("")
} else {
  cat("")
  cat("### Diagnostic")
  cat("")
}
```


<br>

```{r germline_diagnostic, echo=F, results='asis', eval = show_germline_filters[['diagnostic']]}
#| eval: !expr show_germline_filters$diagnostic
#| output: asis
#| echo: false

variants_germline_diagnostic_shared <- crosstalk::SharedData$new(
  cps_report[["content"]][["snv_indel"]][["clin_eitem"]][["diagnostic"]][['any']])
crosstalk::bscols(
  list(
    crosstalk::filter_select("SYMBOL", "Gene", variants_germline_diagnostic_shared, ~SYMBOL),
    crosstalk::filter_select("CANCER_TYPE", "Cancer type", variants_germline_diagnostic_shared, ~CANCER_TYPE),
    crosstalk::filter_select("CLINICAL_SIGNIFICANCE", "Clinical association", variants_germline_diagnostic_shared, ~CLINICAL_SIGNIFICANCE)
  ),
  list(
    crosstalk::filter_select("EVIDENCE_LEVEL", "Evidence level", variants_germline_diagnostic_shared, ~EVIDENCE_LEVEL),
    crosstalk::filter_select("BIOMARKER_MAPPING", "Biomarker mapping", variants_germline_diagnostic_shared, ~BIOMARKER_MAPPING)
  )
)
##cat("The table below lists all variant-evidence item associations:", sep = "\n")
htmltools::br()
htmltools::br()

DT::datatable(variants_germline_diagnostic_shared,
  escape = F, extensions = c("Buttons", "Responsive"),
  options = list(
    buttons = c("csv", "excel"),
    pageLength = 6, dom = "Bfrtip"
  )) |>
  DT::formatStyle("EVIDENCE_LEVEL",
    backgroundColor = DT::styleEqual(
      cps_report$settings$conf$visual_reporting$color_palette$clinical_evidence$levels,
      cps_report$settings$conf$visual_reporting$color_palette$clinical_evidence$values
    )
  ) |>
  DT::formatStyle(
    color = "white", "SYMBOL", "BIOMARKER_MAPPING", fontWeight = "bold", `text-align` = "center",
    backgroundColor = DT::styleEqual(
      c("genomic", "hgvsp", "codon", "hgvsp_nonprincipal", "exon", "gene"),
      c("#000", "#000", "#000","#000",
        cps_report$settings$conf$visual_reporting$color_palette[["warning"]][["values"]][1],
        cps_report$settings$conf$visual_reporting$color_palette[["warning"]][["values"]][1])
    )
  )
```

```{r germline_diagnostic_missing}
#| echo: false
#| output: asis
#| eval: !expr missing_germline_items$diagnostic

cat("<i>No variant-evidence item associations found.</i>", sep = "\n")
cat("\n")
```

:::

<br><br><br>


