
## Genomic biomarkers 

```{r prepare_germline_data}
#| echo: false
#| output: asis
#| eval: true

show_germline_filters <- list()
missing_germline_items <- list()
biomarker_present <- F
etype_set <- list()

for (etype in c("Predictive","Diagnostic","Predisposing","Prognostic")) {
  show_germline_filters[[tolower(etype)]] <- F
  missing_germline_items[[tolower(etype)]] <- T
  
  if(NROW(cps_report[["content"]][["snv_indel"]]$callset$variant_display$bm) == 0){
    next
  }
  
  etype_set[[tolower(etype)]] <- 
    cps_report[["content"]][["snv_indel"]]$callset$variant_display$bm |>
    dplyr::filter(.data$BM_EVIDENCE_TYPE == etype)

  if (NROW(etype_set[[tolower(etype)]]) > 0) {
    etype_set[[tolower(etype)]] <- 
      etype_set[[tolower(etype)]] |>
      dplyr::arrange(.data$BM_EVIDENCE_LEVEL)
    show_germline_filters[[tolower(etype)]] <- T
    missing_germline_items[[tolower(etype)]] <- F
    biomarker_present <- T
  }
  if (NROW(etype_set[[tolower(etype)]]) > 2000){
    etype_set[[tolower(etype)]] <- 
      head(etype_set[[tolower(etype)]], 2000)
  }
}
```

- Germline DNA variants (__class 4/5__) in the targeted cancer predisposition genes that match with reported clinical evidence items from [CIViC](https://civicdb.org) and [CGI biomarker database](https://www.cancergenomeinterpreter.org/biomarkers) are listed. 

- Total number of _unique variants_ with clinical evidence associations: __N = `r cps_report[['content']][['snv_indel']]$v_stat_bm$n_var_eitems`__

- Total number of detected _variant-clinical evidence associations_:
    * Predictive: __`r cps_report[['content']][['snv_indel']]$v_stat_bm$n_eitems_predictive`__ 
    * Predisposing: __`r cps_report[['content']][['snv_indel']]$v_stat_bm$n_eitems_predisposing`__
    * Prognostic: __`r cps_report[['content']][['snv_indel']]$v_stat_bm$n_eitems_prognostic`__ 
    * Diagnostic: __`r cps_report[['content']][['snv_indel']]$v_stat_bm$n_eitems_diagnostic`__

::: {.callout-note}
## Note - biomarker matching

As indicated from the variant-association numbers above, the same germline mutation may potentially match with multiple evidence items, pending on e.g. the _disease context_, _therapeutic context_, _biomarker resolution_, or underlying _evidence source_ (i.e. publication)
:::

```{r biomarker_note}
#| echo: false
#| output: asis
#| include: true

cat("\n::: {.callout-note}\n## Note - biomarker resolution\n\n Biomarkers registered in CIViC/CGI are provided at different _resolutions_ (i.e. filter <b>Biomarker resolution</b>). The accuracy of a match between a variant in the sample and a clinical evidence item (biomarker) will vary accordingly (highlighted by gene symbols with different color backgrounds):\n\n")

cat('<ul><li>Biomarker match at the <mark style="background-color:black; font-weight:bold; color:white">&nbsp; genomic position/amino acid/codon level &nbsp;</mark></li>')
cat(paste0('<br><li>Biomarker match at the <mark style="background-color:', 
           pcgrr::color_palette$warning,
           '; color:white; font-weight:bold">&nbsp; exon/gene level&nbsp; </mark></li></ul>\n\n:::'))

htmltools::br()
```

::: {.panel-tabset}

```{r active_tab_cpsr_predisposing}
#| echo: false
#| output: asis
#| eval: true

if (missing_germline_items[["predisposing"]] == F |
  (missing_germline_items[["diagnostic"]] == T &
    missing_germline_items[["prognostic"]] == T &
    missing_germline_items[["predictive"]] == T)) {
  cat("")
  cat("### Predisposing {.active}")
  cat("")
} else {
  cat("")
  cat("### Predisposing")
  cat("")
}
```


<br>

```{r germline_predisposing}
#| eval: !expr show_germline_filters$predisposing
#| output: asis
#| echo: false

variants_germline_predisposing_shared <- crosstalk::SharedData$new(
  etype_set[['predisposing']])
crosstalk::bscols(
  list(
    crosstalk::filter_select("BM_CANCER_TYPE", "Cancer type", variants_germline_predisposing_shared, ~BM_CANCER_TYPE),
    crosstalk::filter_select("BM_CLINICAL_SIGNIFICANCE", "Clinical significance", variants_germline_predisposing_shared, ~BM_CLINICAL_SIGNIFICANCE),
    crosstalk::filter_select("BM_EVIDENCE_LEVEL", "Evidence level", variants_germline_predisposing_shared, ~BM_EVIDENCE_LEVEL)
  ),
  list(
    crosstalk::filter_select("SYMBOL", "Gene", variants_germline_predisposing_shared, ~SYMBOL),
    crosstalk::filter_select("BM_RESOLUTION", "Biomarker resolution", variants_germline_predisposing_shared, ~BM_RESOLUTION)
  )
)

##cat("The table below lists all variant-evidence item associations:", sep = "\n")
htmltools::br()
htmltools::br()

DT::datatable(variants_germline_predisposing_shared,
  escape = F, extensions = c("Buttons", "Responsive"),
  options =
    list(
      buttons = c("csv", "excel"),
      pageLength = 6,
      dom = "Bfrtip"
    )) |>
   DT::formatStyle("BM_EVIDENCE_LEVEL",
    backgroundColor = DT::styleEqual(
      pcgrr::color_palette$clinical_evidence$levels,
      pcgrr::color_palette$clinical_evidence$values
    )
  ) |>
  DT::formatStyle(
    color = "white", 
    "SYMBOL", 
    "BM_RESOLUTION", 
    fontWeight = "bold", 
    `text-align` = "center",
    backgroundColor = DT::styleEqual(
      c("genomic", "hgvsp", "codon", 
        "hgvsp_nonprincipal", "exon", "gene"),
      c("#000", "#000", "#000","#000",
        pcgrr::color_palette$warning,
        pcgrr::color_palette$warning)
    )
  )
```

```{r germline_predisposing_missing}
#| echo: false
#| output: asis
#| eval: !expr missing_germline_items$predisposing

cat("<i>No variant-evidence item associations found.</i>", sep = "\n")
cat("\n")
```

<br><br><br>


```{r active_tab_cpsr_predictive}
#| echo: false
#| output: asis

if (missing_germline_items[["predictive"]] == F &
  missing_germline_items[["predisposing"]] == T) {
  cat("")
  cat("### Predictive {.active}")
  cat("")
} else {
  cat("")
  cat("### Predictive")
  cat("")
}
```


<br>

```{r germline_predictive}
#| eval: !expr show_germline_filters$predictive
#| output: asis
#| echo: false

variants_germline_predictive_shared <- crosstalk::SharedData$new(
  etype_set[['predictive']])
crosstalk::bscols(
  list(
    crosstalk::filter_select("BM_CANCER_TYPE", "Cancer type", variants_germline_predictive_shared, ~BM_CANCER_TYPE),
    crosstalk::filter_select("BM_CLINICAL_SIGNIFICANCE", "Clinical significance", variants_germline_predictive_shared, ~BM_CLINICAL_SIGNIFICANCE),
    crosstalk::filter_select("BM_EVIDENCE_LEVEL", "Evidence level", variants_germline_predictive_shared, ~BM_EVIDENCE_LEVEL)
  ),
  list(
    crosstalk::filter_select("SYMBOL", "Gene", variants_germline_predictive_shared, ~SYMBOL),
    crosstalk::filter_select("BM_RESOLUTION", "Biomarker resolution", 
                             variants_germline_predictive_shared, ~BM_RESOLUTION),
    crosstalk::filter_select("BM_THERAPEUTIC_CONTEXT", "Therapeutic context", variants_germline_predictive_shared, ~BM_THERAPEUTIC_CONTEXT)
  )
)

##cat("The table below lists all variant-evidence item associations:", sep = "\n")
htmltools::br()
htmltools::br()

DT::datatable(
  variants_germline_predictive_shared, 
  escape = F, extensions = c("Buttons", "Responsive"), 
  options = list(
    buttons = c("csv", "excel"),
    pageLength = 6, dom = "Bfrtip")) |>
  DT::formatStyle(
    "BM_EVIDENCE_LEVEL",
    backgroundColor = DT::styleEqual(
      pcgrr::color_palette$clinical_evidence$levels,
      pcgrr::color_palette$clinical_evidence$values
    )
  ) |>
  DT::formatStyle(
    color = "white", 
    "SYMBOL", 
    "BM_RESOLUTION", 
    fontWeight = "bold", 
    `text-align` = "center",
    backgroundColor = DT::styleEqual(
      c("genomic", "hgvsp", "codon", 
        "hgvsp_nonprincipal", "exon", "gene"),
      c("#000", "#000", "#000","#000",
        pcgrr::color_palette$warning,
        pcgrr::color_palette$warning)
    )
  )
```

```{r germline_predictive_missing}
#| echo: false
#| output: asis
#| eval: !expr missing_germline_items$predictive

cat("<i>No variant-evidence item associations found.</i>", sep = "\n")
cat("\n")
```

<br><br><br>


```{r active_tab_cpsr_prognostic}
#| echo: false
#| output: asis

if (missing_germline_items[["prognostic"]] == F &
  missing_germline_items[["predisposing"]] == T &
  missing_germline_items[["predictive"]] == T) {
  cat("")
  cat("### Prognostic {.active}")
  cat("")
} else {
  cat("")
  cat("### Prognostic")
  cat("")
}
```


<br>

```{r germline_prognostic}
#| eval: !expr show_germline_filters$prognostic
#| output: asis
#| echo: false

variants_germline_prognostic_shared <- crosstalk::SharedData$new(
  etype_set[['prognostic']])
crosstalk::bscols(
  list(
    crosstalk::filter_select("SYMBOL", "Gene", variants_germline_prognostic_shared, ~SYMBOL),
    crosstalk::filter_select("BM_CANCER_TYPE", "Cancer type", variants_germline_prognostic_shared, ~BM_CANCER_TYPE),
    crosstalk::filter_select("BM_CLINICAL_SIGNIFICANCE", "Clinical significance", variants_germline_prognostic_shared, ~BM_CLINICAL_SIGNIFICANCE)
  ),
  list(
    crosstalk::filter_select("BM_EVIDENCE_LEVEL", "Evidence level", variants_germline_prognostic_shared, ~BM_EVIDENCE_LEVEL),
    crosstalk::filter_select("BM_RESOLUTION", "Biomarker resolution", variants_germline_prognostic_shared, ~BM_RESOLUTION)
  )
)
##cat("The table below lists all variant-evidence item associations:", sep = "\n")
htmltools::br()
htmltools::br()

DT::datatable(variants_germline_prognostic_shared, escape = F, extensions = c("Buttons", "Responsive"), options = list(
  buttons = c("csv", "excel"),
  pageLength = 6, dom = "Bfrtip")) |>
   DT::formatStyle("BM_EVIDENCE_LEVEL",
    backgroundColor = DT::styleEqual(
      pcgrr::color_palette$clinical_evidence$levels,
      pcgrr::color_palette$clinical_evidence$values
    )
  ) |>
 DT::formatStyle(
    color = "white", 
    "SYMBOL", 
    "BM_RESOLUTION", 
    fontWeight = "bold", 
    `text-align` = "center",
    backgroundColor = DT::styleEqual(
      c("genomic", "hgvsp", "codon", "hgvsp_nonprincipal", "exon", "gene"),
      c("#000", "#000", "#000","#000",
        pcgrr::color_palette$warning,
        pcgrr::color_palette$warning)
    )
  )
```

```{r germline_prognostic_missing}
#| echo: false
#| output: asis
#| eval: !expr missing_germline_items$prognostic

cat("<i>No variant-evidence item associations found.</i>", sep = "\n")
cat("\n")
```

<br><br><br>

```{r active_tab_cpsr_diagnostic}
#| echo: false
#| output: asis

if (missing_germline_items[["diagnostic"]] == F &
  missing_germline_items[["prognostic"]] == T &
  missing_germline_items[["predisposing"]] == T &
  missing_germline_items[["predictive"]] == T) {
  cat("")
  cat("### Diagnostic {.active}")
  cat("")
} else {
  cat("")
  cat("### Diagnostic")
  cat("")
}
```


<br>

```{r germline_diagnostic, echo=F, results='asis', eval = show_germline_filters[['diagnostic']]}
#| eval: !expr show_germline_filters$diagnostic
#| output: asis
#| echo: false

variants_germline_diagnostic_shared <- crosstalk::SharedData$new(
  etype_set[['diagnostic']])
crosstalk::bscols(
  list(
    crosstalk::filter_select("SYMBOL", "Gene", variants_germline_diagnostic_shared, ~SYMBOL),
    crosstalk::filter_select("BM_CANCER_TYPE", "Cancer type", variants_germline_diagnostic_shared, ~BM_CANCER_TYPE),
    crosstalk::filter_select("BM_CLINICAL_SIGNIFICANCE", "Clinical association", variants_germline_diagnostic_shared, ~BM_CLINICAL_SIGNIFICANCE)
  ),
  list(
    crosstalk::filter_select("BM_EVIDENCE_LEVEL", "Evidence level", variants_germline_diagnostic_shared, ~BM_EVIDENCE_LEVEL),
    crosstalk::filter_select("BM_RESOLUTION", "Biomarker resolution", variants_germline_diagnostic_shared, ~BM_RESOLUTION)
  )
)
##cat("The table below lists all variant-evidence item associations:", sep = "\n")
htmltools::br()
htmltools::br()

DT::datatable(variants_germline_diagnostic_shared,
  escape = F, extensions = c("Buttons", "Responsive"),
  options = list(
    buttons = c("csv", "excel"),
    pageLength = 6, dom = "Bfrtip"
  )) |>
  DT::formatStyle("BM_EVIDENCE_LEVEL",
    backgroundColor = DT::styleEqual(
      pcgrr::color_palette$clinical_evidence$levels,
      pcgrr::color_palette$clinical_evidence$values
    )
  ) |>
  DT::formatStyle(
    color = "white", 
    "SYMBOL",
    "BM_RESOLUTION", 
    fontWeight = "bold", 
    `text-align` = "center",
    backgroundColor = DT::styleEqual(
      c("genomic", "hgvsp", "codon", 
        "hgvsp_nonprincipal", "exon", "gene"),
      c("#000", "#000", "#000","#000",
        pcgrr::color_palette$warning,
        pcgrr::color_palette$warning)
    )
  )
```

```{r germline_diagnostic_missing}
#| echo: false
#| output: asis
#| eval: !expr missing_germline_items$diagnostic

cat("<i>No variant-evidence item associations found.</i>", sep = "\n")
cat("\n")
```

:::

<br><br><br>
