## Settings

```{r on_off_settings}
#| echo: false
#| eval: true

gencode_metadata <- 
  cps_report$settings$reference_data$source_metadata |>
  dplyr::filter(source_abbreviation == "gencode")

transcript_set_gencode <- paste0("GENCODE - basic set (v", gencode_metadata$source_version, ")")
if (as.logical(cps_report[["settings"]][["conf"]][["vep"]][["vep_gencode_basic"]]) == F) {
  transcript_set_gencode <- paste0("GENCODE - all transcripts (v", gencode_metadata$source_version, ")")
}
```

* Sample name &nbsp;:&nbsp; <span style="background-color:#007a74; font-size:1.2em; color:white"><b>&nbsp;&nbsp;`r cps_report[['settings']][['sample_id']]`&nbsp;&nbsp;</b></span>
<br>

::: {.callout-note collapse="true"}

## Run configuration and report settings

The report is generated with _[cpsr](https://github.com/sigven/cpsr)_ version `r cps_report[['settings']][['software']][['cpsr_version']]`, ( _[pcgr](https://github.com/sigven/pcgr)_ version `r cps_report[['settings']][['software']][['pcgr_version']]`), using the following key settings:

:::: {.columns}

::: {.column width="47.5%"}

* Genome assembly: __`r cps_report[['settings']][['genome_assembly']]`__
* Control population gnomAD: __`r cps_report[['settings']][['conf']][['variant_classification']][['popdesc_gnomad']]`__
* Upper MAF threshold (gnomAD) for non-ClinVar-classified variants:  __`r cps_report[['settings']][['conf']][['variant_classification']][['maf_upper_threshold']]`__
* Ignore non-protein-coding variants: __`r !as.logical(cps_report[['settings']][['conf']][['other']][['show_noncoding']])`__
* Generate CPSR variant classifications also for classified ClinVar variants: __`r as.logical(cps_report[['settings']][['conf']][['variant_classification']][['classify_all']])`__
* Report ClinVar variants if they are only associated with non-cancer phenotypes/diseases: 
__`r as.logical(cps_report[['settings']][['conf']][['variant_classification']][['clinvar_report_noncancer']])`__

:::

::: {.column width="5%"}
<!-- empty column to create gap -->
:::

::: {.column width="47.5%"}

* Include [secondary findings](https://www.ncbi.nlm.nih.gov/clinvar/docs/acmg/): __`r as.logical(cps_report[['settings']][['conf']][['variant_classification']][['secondary_findings']])`__
* Include [cancer-related pharmacogenomics findings](https://www.ncbi.nlm.nih.gov/clinvar/docs/acmg/): __`r as.logical(cps_report[['settings']][['conf']][['variant_classification']][['pgx_findings']])`__
* Include GWAS hits: __`r as.logical(cps_report[['settings']][['conf']][['variant_classification']][['gwas_findings']])`__
* Variant Effect Predictor (VEP) settings:
  * Transcript set: __`r transcript_set_gencode`__
  * Transcript pick order: __`r stringr::str_replace_all(cps_report[['settings']][['conf']][['vep']][['vep_pick_order']],",",", ")`__
  * Regulatory regions annotation: __`r as.logical(cps_report[['settings']][['conf']][['vep']][['vep_regulatory']])`__
  * Buffer size: __`r cps_report$settings$conf$vep$vep_buffer_size`__
  * Number of forks: __`r cps_report$settings$conf$vep$vep_n_forks`__

:::

::::

:::
