
##  Secondary findings

```{r prepare_sf_data}
#| output: asis
#| echo: false

tot_sf_variants <- NROW(
  cps_report[["content"]][["snv_indel"]]$callset$variant_display$sf)
acmg_version <- cps_report$settings$reference_data$source_metadata[
  cps_report$settings$reference_data$source_metadata$source_abbreviation == "acmg_sf","source_version"]
acmg_url <- cps_report$settings$reference_data$source_metadata[
  cps_report$settings$reference_data$source_metadata$source_abbreviation == "acmg_sf","source_url"]
genotypes_unavailable <- !(as.logical(
  cps_report$settings$conf$sample_properties$gt_detected))

```

```{r gt_unavailable}
#| eval: !expr genotypes_unavailable
#| output: asis
#| echo: false

cat("::: {.callout-warning}\n## Warning - genotype data missing\n\n*Secondary findings not assessed due to lacking genotype information among sample variants*. CPSR was unable to pull this data from the input VCF (<i>'GT'</i> FORMAT tag not present within sample genotype information)\n\n:::")

```


*  For the sample in question, a total of __N = `r tot_sf_variants`__ variants in [genes recommended for secondary findings reporting (ACMG `r acmg_version`)](`r acmg_url`) 
are registered with a <i>Pathogenic/Likely pathogenic</i> clinical significance in ClinVar.

<br>

```{r sf_cancer}
#| eval: !expr tot_sf_variants > 0
#| output: asis
#| echo: false

htmltools::br()
htmltools::br()

DT::datatable(
  cps_report[["content"]][["snv_indel"]]$callset$variant_display$sf,
  escape = F, extensions = c("Buttons", "Responsive"),
  options = list(
    pageLength = 6,
    buttons = c("csv", "excel"),
    dom = "Bfrtip",
    initComplete = DT::JS(
      "function(settings, json) {",
      "$(this.api().table().header()).css({'background-color': '#8E9092', 'color': '#fff'});",
      "}"
    )
  )
)

htmltools::br()
htmltools::br()
```
