

<!DOCTYPE html>
<!--[if IE 8]><html class="no-js lt-ie9" lang="en" > <![endif]-->
<!--[if gt IE 8]><!--> <html class="no-js" lang="en" > <!--<![endif]-->
<head>
  <meta charset="utf-8">
  
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  <title>Getting started &mdash; Cancer Predisposition Sequencing Reporter 2.1.0 documentation</title>
  

  
  
  
  

  

  
  
    

  

  <link rel="stylesheet" href="_static/css/theme.css" type="text/css" />
  <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
    <link rel="author" title="About these documents" href="about.html" />
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="CPSR annotation resources" href="annotation_resources.html" />
    <link rel="prev" title="About" href="about.html" /> 

  
  <script src="_static/js/modernizr.min.js"></script>

</head>

<body class="wy-body-for-nav">

   
  <div class="wy-grid-for-nav">

    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search">
          

          
            <a href="index.html" class="icon icon-home"> Cancer Predisposition Sequencing Reporter
          

          
          </a>

          
            
            
              <div class="version">
                2.1.0
              </div>
            
          

          
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>

          
        </div>

        <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
          
            
            
              
            
            
              <p class="caption"><span class="caption-text">Table of Contents</span></p>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="about.html">About</a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="#">Getting started</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#step-0-python">STEP 0: Python</a></li>
<li class="toctree-l2"><a class="reference internal" href="#step-1-installation-of-docker">STEP 1: Installation of Docker</a></li>
<li class="toctree-l2"><a class="reference internal" href="#step-2-download-run-script-data-bundle-and-pull-docker-image">STEP 2: Download run script/data bundle, and pull Docker image</a></li>
<li class="toctree-l2"><a class="reference internal" href="#step-3-input-preprocessing">STEP 3: Input preprocessing</a></li>
<li class="toctree-l2"><a class="reference internal" href="#step-4-configuration">STEP 4: Configuration</a></li>
<li class="toctree-l2"><a class="reference internal" href="#step-5-run-example">STEP 5: Run example</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="annotation_resources.html">CPSR annotation resources</a></li>
<li class="toctree-l1"><a class="reference internal" href="input.html">Input</a></li>
<li class="toctree-l1"><a class="reference internal" href="output.html">Output</a></li>
<li class="toctree-l1"><a class="reference internal" href="variant_classification.html">Variant classification (ACMG)</a></li>
<li class="toctree-l1"><a class="reference internal" href="CHANGELOG.html">CHANGELOG</a></li>
</ul>

            
          
        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" aria-label="top navigation">
        
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="index.html">Cancer Predisposition Sequencing Reporter</a>
        
      </nav>


      <div class="wy-nav-content">
        
        <div class="rst-content">
        
          















<div role="navigation" aria-label="breadcrumbs navigation">

  <ul class="wy-breadcrumbs">
    
      <li><a href="index.html">Docs</a> &raquo;</li>
        
      <li>Getting started</li>
    
    
      <li class="wy-breadcrumbs-aside">
        
            
            <a href="_sources/getting_started.rst.txt" rel="nofollow"> View page source</a>
          
        
      </li>
    
  </ul>

  
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
            
  <div class="section" id="getting-started">
<h1>Getting started<a class="headerlink" href="#getting-started" title="Permalink to this headline">¶</a></h1>
<div class="section" id="step-0-python">
<h2>STEP 0: Python<a class="headerlink" href="#step-0-python" title="Permalink to this headline">¶</a></h2>
<p>An installation of Python (version <em>3.6</em>) is required to run CPSR. Check
that Python is installed by typing <code class="docutils literal notranslate"><span class="pre">python</span> <span class="pre">--version</span></code> in your terminal
window. In addition, a <a class="reference external" href="https://github.com/uiri/toml">Python library</a>
for parsing configuration files encoded with
<a class="reference external" href="https://github.com/toml-lang/toml">TOML</a> is needed. To install,
simply run the following command:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">pip</span> <span class="n">install</span> <span class="n">toml</span>
</pre></div>
</div>
</div>
<div class="section" id="step-1-installation-of-docker">
<h2>STEP 1: Installation of Docker<a class="headerlink" href="#step-1-installation-of-docker" title="Permalink to this headline">¶</a></h2>
<ol class="arabic simple">
<li><a class="reference external" href="https://docs.docker.com/engine/installation/">Install the Docker
engine</a> on your
preferred platform<ul>
<li>installing <a class="reference external" href="https://docs.docker.com/engine/installation/linux/">Docker on
Linux</a></li>
<li>installing <a class="reference external" href="https://docs.docker.com/engine/installation/mac/">Docker on Mac
OS</a></li>
<li>NOTE: We have not yet been able to perform enough testing on the
Windows platform, and we have received feedback that particular
versions of Docker/Windows do not work with CPSR (an example being
<a class="reference external" href="https://github.com/docker/toolbox/issues/607">mounting of data
volumes</a>)</li>
</ul>
</li>
<li>Test that Docker is running, e.g.&nbsp;by typing <code class="docutils literal notranslate"><span class="pre">docker</span> <span class="pre">ps</span></code> or
<code class="docutils literal notranslate"><span class="pre">docker</span> <span class="pre">images</span></code> in the terminal window</li>
<li>Adjust the computing resources dedicated to the Docker, i.e.:<ul>
<li>Memory: minimum 5GB</li>
<li>CPUs: minimum 4</li>
<li><a class="reference external" href="https://docs.docker.com/docker-for-mac/#advanced">How to - Mac OS
X</a></li>
</ul>
</li>
</ol>
</div>
<div class="section" id="step-2-download-run-script-data-bundle-and-pull-docker-image">
<h2>STEP 2: Download run script/data bundle, and pull Docker image<a class="headerlink" href="#step-2-download-run-script-data-bundle-and-pull-docker-image" title="Permalink to this headline">¶</a></h2>
<ol class="arabic">
<li><p class="first">Download and unpack the <a class="reference external" href="https://github.com/sigven/cpsr/releases/tag/v0.3.0">CPSR pre-release
(0.3.0)</a></p>
</li>
<li><p class="first">Pull the PCGR Docker image (<em>dev version</em>):
<code class="docutils literal notranslate"><span class="pre">docker</span> <span class="pre">pull</span> <span class="pre">sigven/pcgr:dev</span></code></p>
</li>
<li><p class="first">Download and unpack the latest PCGR data bundles</p>
<ul class="simple">
<li><a class="reference external" href="https://drive.google.com/open?id=1OL5C994HDaeadASz7KzMhPoXfdSiyhNy">grch37 data bundle -
20181119</a>
(approx 9Gb)</li>
<li><a class="reference external" href="https://drive.google.com/open?id=1CZNc87E0K5AK2RDSNU57FqLp0H1skpUh">grch38 data bundle -
20181119</a>
(approx 14Gb)</li>
<li><em>Unpacking</em>:
<code class="docutils literal notranslate"><span class="pre">gzip</span> <span class="pre">-dc</span> <span class="pre">pcgr.databundle.grch37.YYYYMMDD.tgz</span> <span class="pre">|</span> <span class="pre">tar</span> <span class="pre">xvf</span> <span class="pre">-</span></code></li>
</ul>
<p>A <em>data/</em> folder should now have been produced</p>
</li>
</ol>
</div>
<div class="section" id="step-3-input-preprocessing">
<h2>STEP 3: Input preprocessing<a class="headerlink" href="#step-3-input-preprocessing" title="Permalink to this headline">¶</a></h2>
<p>The CPSR workflow accepts one type of input file</p>
<ul class="simple">
<li>An unannotated, single-sample VCF file (&gt;= v4.2) with germline DNA
variants (SNVs/InDels)</li>
<li>We <strong>strongly</strong> recommend that the input VCF is compressed and
indexed using <a class="reference external" href="http://www.htslib.org/doc/tabix.html">bgzip</a> and
<a class="reference external" href="http://www.htslib.org/doc/tabix.html">tabix</a></li>
<li>If the input VCF contains multi-allelic sites, these will be subject
to <a class="reference external" href="http://genome.sph.umich.edu/wiki/Vt#Decompose">decomposition</a></li>
<li>Variants used for reporting should be designated as ‘PASS’ in the VCF
FILTER column (non-PASS variants are ignored in the report)</li>
</ul>
</div>
<div class="section" id="step-4-configuration">
<h2>STEP 4: Configuration<a class="headerlink" href="#step-4-configuration" title="Permalink to this headline">¶</a></h2>
<p>A few elements of the workflow can be figured using the <em>cpsr</em>
configuration file, encoded in
<a class="reference external" href="https://github.com/toml-lang/toml">TOML</a> (an easy to read file
format).</p>
<p>The initial step of the workflow performs <a class="reference external" href="https://github.com/EBIvariation/vcf-validator">VCF
validation</a> on the
input VCF file. This procedure is very strict, and often causes the
workflow to return an error due to various violations of the VCF
specification. If the user trusts that the most critical parts of the
input VCF is properly encoded, a setting in the configuration file
(<code class="docutils literal notranslate"><span class="pre">vcf_validation</span> <span class="pre">=</span> <span class="pre">false</span></code>) can be used to turn off VCF validation.</p>
<p>An exhaustive, predefined list of 209 cancer predisposition/syndrome
genes can also be configured.</p>
<p>See section on <a class="reference external" href="input.md">Input</a> for more details wrt. configuration.</p>
</div>
<div class="section" id="step-5-run-example">
<h2>STEP 5: Run example<a class="headerlink" href="#step-5-run-example" title="Permalink to this headline">¶</a></h2>
<p>Run the workflow with <strong>cpsr.py</strong>, which takes the following arguments
and options:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>usage: cpsr.py [-h] [--input_vcf INPUT_VCF] [--force_overwrite] [--version]
        [--basic] [--docker-uid DOCKER_USER_ID] [--no-docker]
        pcgr_base_dir output_dir {grch37,grch38} configuration_file
        sample_id

Cancer Predisposition Sequencing Reporter (CPSR) - report of cancer-predisposing
germline variants

positional arguments:
pcgr_base_dir         Directory that contains the PCGR data bundle
                directory, e.g. ~/pcgr-dev
output_dir            Output directory
{grch37,grch38}       Genome assembly build: grch37 or grch38
configuration_file    Configuration file (TOML format)
sample_id             Sample identifier - prefix for output files

optional arguments:
-h, --help            show this help message and exit
--input_vcf INPUT_VCF
                VCF input file with somatic query variants
                (SNVs/InDels). (default: None)
--force_overwrite     By default, the script will fail with an error if any
                output file already exists. You can force the
                overwrite of existing result files by using this flag
                (default: False)
--version             show program&#39;s version number and exit
--basic               Run functional variant annotation on VCF through
                VEP/vcfanno, omit report generation (STEP 4) (default:
                False)
--docker-uid DOCKER_USER_ID
                Docker user ID. Default is the host system user ID. If
                you are experiencing permission errors, try setting
                this up to root (`--docker-uid root`) (default: None)
--no-docker           Run the CPSR workflow in a non-Docker mode (see
                install_no_docker/ folder for instructions (default:
                False)
</pre></div>
</div>
<p>The <em>cpsr</em> software bundle contains an example VCF file. It also
contains a configuration file (<em>cpsr.toml</em>).</p>
<p>Analysis of the example VCF can be performed by the following command:</p>
<p><code class="docutils literal notranslate"><span class="pre">python</span> <span class="pre">~/cpsr-0.3.0/cpsr.py</span> <span class="pre">--input_vcf</span> <span class="pre">~/cpsr-0.3.0/example.vcf.gz</span></code>
<code class="docutils literal notranslate"><span class="pre">~/pcgr-dev</span> <span class="pre">~/cpsr-0.3.0</span> <span class="pre">grch37</span> <span class="pre">~/cpsr-0.3.0/cpsr.toml</span> <span class="pre">example</span></code></p>
<p>Note that the example command also refers to the PCGR directory
(<em>pcgr-dev</em>), which contains the data bundle that are necessary for both
<em>PCGR</em> and <em>CPSR</em>.</p>
<p>This command will run the Docker-based <em>cpsr</em> workflow and produce the
following output files in the <em>cpsr</em> folder:</p>
<ol class="arabic simple">
<li><strong>example.cpsr.grch37.pass.vcf.gz (.tbi)</strong> - Bgzipped VCF file with
functional/clinical annotations</li>
<li><strong>example.cpsr.grch37.pass.tsv.gz</strong> - Compressed TSV file (generated
with <a class="reference external" href="https://github.com/sigven/vcf2tsv">vcf2tsv</a>) with
functional/clinical annotations</li>
<li><strong>example.cpsr.grch37.html</strong> - Interactive HTML report with
clinically relevant variants in cancer predisposition genes organized
into tiers</li>
<li><strong>example.cpsr.grch37.json.gz</strong> - Compressed JSON dump of HTML report
content</li>
<li><strong>example.cpsr.snvs_indels.tiers.grch37.tsv</strong> - TSV file with most
important annotations of tier-structured SNVs/InDels</li>
</ol>
</div>
</div>


           </div>
           
          </div>
          <footer>
  
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
      
        <a href="annotation_resources.html" class="btn btn-neutral float-right" title="CPSR annotation resources" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right"></span></a>
      
      
        <a href="about.html" class="btn btn-neutral" title="About" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left"></span> Previous</a>
      
    </div>
  

  <hr/>

  <div role="contentinfo">
    <p>
        &copy; Copyright 2018, Sigve Nakken.

    </p>
  </div>
  Built with <a href="http://sphinx-doc.org/">Sphinx</a> using a <a href="https://github.com/rtfd/sphinx_rtd_theme">theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>. 

</footer>

        </div>
      </div>

    </section>

  </div>
  


  

    <script type="text/javascript">
        var DOCUMENTATION_OPTIONS = {
            URL_ROOT:'./',
            VERSION:'2.1.0',
            LANGUAGE:'None',
            COLLAPSE_INDEX:false,
            FILE_SUFFIX:'.html',
            HAS_SOURCE:  true,
            SOURCELINK_SUFFIX: '.txt'
        };
    </script>
      <script type="text/javascript" src="_static/jquery.js"></script>
      <script type="text/javascript" src="_static/underscore.js"></script>
      <script type="text/javascript" src="_static/doctools.js"></script>

  

  <script type="text/javascript" src="_static/js/theme.js"></script>

  <script type="text/javascript">
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script>  

  <style>
    /* Sidebar header (and topbar for mobile) */
    .wy-side-nav-search, .wy-nav-top {
      background: #0B6121;
    }
    /* Sidebar */
    /*.wy-nav-side {*/
    /*  background: #ff0000;*/
    /*}*/
  </style>


</body>
</html>