---
title: "Cancer Predisposition Variant Report by CPSR"
subtitle: " - _functional interpretation of cancer-predisposing DNA variants_"
date: '`r strftime(Sys.time(), format = "%B %d, %Y")`'
bibliography: cpsr_bibliography.bib
title-block-banner: "#007a74"
title-block-banner-color: "white"
linkcolor: "#007a74"
include-in-header:
  - text: |
      <link rel="shortcut icon" href="https://github.com/sigven/cpsr/raw/master/pkgdown/favicon/favicon-16x16.png" />
      <style>
      .panel-tabset > .tab-content {
        border: none;
      }
      </style>
nocite: |
  @Nakken2021-mb, @Huang2018-ah, @Richards2015-kj, @Maxwell2016-mr, @Amendola2016-pu, @Griffith2017-do, @Martin2019-nq, @Nykamp2017-rj, Dienstmann2015-od
format: 
  html:
    embed-resources: true
    smooth-scroll: true
    page-layout: full
    fontsize: 0.9em
    toc: true
    toc_depth: 2
    grid:
      body-width: 920px
css: cpsr.css

---

```{r global_options}
#| include: false
#| eval: true

knitr::opts_chunk$set(echo = F, warning = FALSE, dpi = 72, error = F, eval = T)
options(scipen = 999)


width_donut <- 10
height_donut <- 7

cps_report <- readRDS(file="../../cps_test.rds")

set_all <- NROW(cps_report$content$snv_indel$variant_set$tsv)
no_content <- !(as.logical(cps_report[['content']][['snv_indel']][['eval']]))


set_clinvar <- 0
set_other <- 0
if (cps_report$content$snv_indel$eval == T) {
  if (cps_report$content$snv_indel$v_stat$n > 0) {
    set_clinvar <- 
      NROW(
        cps_report$content$snv_indel$variant_set$tsv |> 
          dplyr::filter(!is.na(CLINVAR_CLASSIFICATION))
        )
    set_other <- NROW(
      cps_report$content$snv_indel$variant_set$tsv |> 
        dplyr::filter(is.na(CLINVAR_CLASSIFICATION) & 
                        !is.na(CPSR_CLASSIFICATION) > 0)
      )
  }
}

if (set_other == 0 | set_clinvar == 0) {
  width_donut <- 6
  height_donut <- 4
}

logo_fname_path <- system.file("extdata", "logo.png", package = "cpsr")

```

<a href="https://sigven.github.io/cpsr">
  <img src="`r logo_fname_path`" title="" alt="" align="right" width="80"/>
</a>

```{r settings_cpsr}
#| child: 'quarto/cpsr_settings.qmd'
```


```{r panel_cpsr}
#| child: 'quarto/cpsr_virtual_panel.qmd'
```

```{r summary_cpsr}
#| child: 'quarto/cpsr_summary.qmd'
```

```{r no_variants}
#| eval: !expr no_content
#| output: asis
cat("<br><ul>")
cat("<li><b>NO</b> variants found in the selected virtual gene panel/custom geneset</li>")
cat("</ul><br>")
```


```{r classification_cpsr}
#| child: 'quarto/cpsr_classification.qmd'
#| eval: !expr as.logical(cps_report[['content']][['snv_indel']][['eval']])
```

<br><br>

```{r biomarkers_cpsr} 
#| child: 'quarto/cpsr_biomarkers.qmd'
#| eval: !expr as.logical(cps_report[['content']][['snv_indel']][['eval']])
```

<br><br>

```{r secondary_findings_cpsr}
#| child: 'quarto/cpsr_secondary_findings.qmd'
#| eval: !expr as.logical(cps_report$settings$conf$variant_classification$secondary_findings)
```

```{r update_global_options}
#| include: false
options(scipen = 0)
```

```{r gwas_cpsr}
#| child: 'quarto/cpsr_gwas.qmd'
#| eval: !expr as.logical(cps_report$settings$conf$variant_classification$gwas_findings)
```

```{r documentation_cpsr, child='quarto/cpsr_documentation.qmd', eval = TRUE}
```

<br><br>


## References

::: {#refs}
:::

```{r cpsr_disclaimer}
#| eval: true
#| output: asis

cat('<p><br><br><br><b>MEDICAL DISCLAIMER</b>:<i>The information contained in this cancer predisposition genome report is intended for research purposes only. We make no representations or warranties of any kind, expressed or implied, about the completeness, accuracy, reliability, suitability or availability with respect to the sequencing report or the information, products, services, for interpretation or use in clinical practice, or otherwise contained in the report for any purpose. Any reliance you place on information in the report is therefore strictly at your own risk. In no event will we be liable for any loss or damage including without limitation, indirect or consequential loss or damage, or any loss or damage whatsoever arising from loss of data or profits arising out of, or in connection with, the use of this genome report.</i><br><br></p>')

```

