<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="cpsr">
<title>Changelog • CPSR</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.11/clipboard.min.js" integrity="sha512-7O5pXpc0oCRrxk8RUfDYFgn0nO1t+jLuIOQdOMRp4APB7uZ4vSjspzp5y6YDtDs4VzUSTbWzBFZ/LKJhnyFOKw==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Changelog">
<meta property="og:description" content="cpsr">
<meta property="og:image" content="https://sigven.github.io/cpsr/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-info" data-bs-theme="light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">CPSR</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">2.0.0</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="../articles/installation.html">Installation</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../articles/running.html">Running</a>
</li>
<li class="nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/input.html">Input files</a>
    <a class="dropdown-item" href="../articles/virtual_panels.html">Virtual gene panels</a>
    <a class="dropdown-item" href="../articles/variant_classification.html">Variant classification</a>
    <a class="dropdown-item" href="../articles/output.html">Output files</a>
    <a class="dropdown-item" href="../articles/annotation_resources.html">Annotation resources</a>
  </div>
</li>
<li class="active nav-item">
  <a class="nav-link" href="../articles/CHANGELOG.html">CHANGELOG</a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/sigven/cpsr/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>Changelog</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/sigven/cpsr/blob/HEAD/vignettes/CHANGELOG.Rmd" class="external-link"><code>vignettes/CHANGELOG.Rmd</code></a></small>
      <div class="d-none name"><code>CHANGELOG.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="v2-0-0">v2.0.0<a class="anchor" aria-label="anchor" href="#v2-0-0"></a>
</h2>
<ul>
<li>Date: <strong>2024-06-26</strong>
</li>
<li>Data updates
<ul>
<li>ClinVar</li>
<li>GWAS catalog</li>
<li>CIViC</li>
<li>GENCODE</li>
<li>Cancer Gene Census</li>
<li>PanelApp</li>
<li>Disease Ontology/EFO</li>
<li>UniProt KB</li>
</ul>
</li>
</ul>
<div class="section level5">
<h5 id="addedchanged">Added/changed<a class="anchor" aria-label="anchor" href="#addedchanged"></a>
</h5>
<ul>
<li>New report generation framework - <a href="https://quarto.org" class="external-link">quarto</a>
<ul>
<li>multiple options related to Rmarkdown output are now deprecated</li>
</ul>
</li>
<li>Re-organized data bundle structure
<ul>
<li>Users need to download an assembly-specific VEP cache separately
from PCGR/CPSR, and provide its path to the new required argument
<code>--vep_dir</code> in the <code>cpsr</code> command</li>
</ul>
</li>
<li>Re-engineered data bundle generation pipeline</li>
<li>Improved data bundle documentation
<ul>
<li>An HTML report with an overview of the contents of the data bundle
is shipped with the reference data itself, also available <a href="https://rpubs.com/sigven/pcgr_refdata" class="external-link">here (grch38)</a>.</li>
</ul>
</li>
<li>Cleaned up code base for reporting and classification</li>
<li>A multi-sheet Excel workbook output with variant classifications and
biomarker findings are provided, suitable e.g. for aggregation of
results across samples</li>
<li>Singularity/Apptainer support</li>
<li>argument name changes to <code>cpsr</code>:
<ul>
<li>
<code>--pcgr_dir</code> is now named <code>--refdata_dir</code>
</li>
<li>
<code>--clinvar_ignore_noncancer</code> is now named
<code>--clinvar_report_noncancer</code>, meaning that variants found in
ClinVar, yet attributed to <em>non-cancer related</em> phenotypes, are
now excluded from reporting by default)</li>
<li>
<code>--vep_gencode_all</code> is now named
<code>--vep_gencode_basic</code>, meaning that the gene variant
annotation is now using <em>all</em> GENCODE transcripts by default, not
only the <em>basic</em> set)</li>
<li>
<code>--preserved_info_tags</code> is now named
<code>--retained_info_tags</code>
</li>
<li>
<code>--basic</code> is now named <code>--no_reporting</code>
</li>
</ul>
</li>
<li>LOFTEE plugin in VEP removed as loss-of-function variant classifier
(due to low level of maintenance, and outdated dependency requirements).
For now, a simplified LoF-annotation is used as a replacement, looking
primarily at CSQ types (<code>stop_gained</code>,
<code>frameshift_variant</code>, <code>splice_acceptor_variant</code>,
<code>splice_donor_variant</code>). Furthermore, frameshift/stop-gain
variants that are found within the last 5% of the coding sequence length
are deemed non-LOF, as are splice donor variants not disrupting the
canonical site (GC&gt;GT). An even more advanced LoF-annotation is
planned for a future release.</li>
<li>Biomarkers are matched much more comprehensively than in previous
versions, matching at the genomic level, codon, exon, amino acid and
gene level (both principal and non-principal transcript matches)</li>
<li>Some bugs in the retrieval of secondary variant findings have been
fixed</li>
</ul>
</div>
<div class="section level5">
<h5 id="removed">Removed<a class="anchor" aria-label="anchor" href="#removed"></a>
</h5>
<ul>
<li>Options for configuring Rmarkdown output,
i.e. <code>--report_theme</code>, <code>report_nonfloating_toc</code>,
<code>report_table_display</code> argument in the <code>cpsr</code>
command is deprecated</li>
<li><code>--gwas_p_value</code></li>
<li>
<code>--no_vcf_validate</code> - VCF validation is simplified, not
relying on the strict <em>vcf-validator</em>
</li>
</ul>
</div>
<div class="section level4">
<h4 id="section"></h4>
</div>
</div>
<div class="section level2">
<h2 id="v1-0-1">v1.0.1<a class="anchor" aria-label="anchor" href="#v1-0-1"></a>
</h2>
<ul>
<li>Date: <strong>2022-11-11</strong>
</li>
</ul>
<div class="section level5">
<h5 id="added">Added<a class="anchor" aria-label="anchor" href="#added"></a>
</h5>
<ul>
<li>Logo</li>
</ul>
</div>
</div>
<div class="section level2">
<h2 id="v1-0-0">v1.0.0<a class="anchor" aria-label="anchor" href="#v1-0-0"></a>
</h2>
<ul>
<li><p>Date: <strong>2022-02-25</strong></p></li>
<li>
<p>Data updates</p>
<ul>
<li>ClinVar</li>
<li>GWAS catalog</li>
<li>CIViC</li>
<li>CancerMine</li>
<li>KEGG</li>
<li>Disease Ontology/EFO</li>
<li>Open Targets Platform,</li>
<li>UniProt KB</li>
<li>GENCODE</li>
</ul>
</li>
<li><p>Improved detection of secondary variant findings</p></li>
<li><p>CPSR is now implemented as a dedicated R package, depending also
on the <a href="https://github.com/sigven/pcgr/tree/master/pcgrr" class="external-link">pcgrr</a> R
package. Running the complete CPSR workflow relies upon installation of
<a href="https://github.com/sigven/pcgr" class="external-link">PCGR</a>.</p></li>
</ul>
</div>
<div class="section level2">
<h2 id="v0-6-2">v0.6.2<a class="anchor" aria-label="anchor" href="#v0-6-2"></a>
</h2>
<ul>
<li><p>Date: <strong>2021-06-30</strong></p></li>
<li><p>Data updates: ClinVar, PanelApp, GWAS catalog, CIViC, CancerMine,
dbNSFP, KEGG, Disease Ontology/EFO, Open Targets Platform, UniProt KB,
GENCODE</p></li>
<li><p>Software upgrades: R v4.1, Bioconductor v3.13, VEP (104)
++</p></li>
<li><p>Improved GWAS track for cancer phenotypes</p></li>
</ul>
<div class="section level5">
<h5 id="changed">Changed<a class="anchor" aria-label="anchor" href="#changed"></a>
</h5>
<ul>
<li>TOML-based configuration for CPSR is abandoned, all options to CPSR
are now configured through the command-line parameters</li>
</ul>
</div>
<div class="section level5">
<h5 id="added-1">Added<a class="anchor" aria-label="anchor" href="#added-1"></a>
</h5>
<ul>
<li>
<p>Command-line options</p>
<ul>
<li>Previously set in TOML file
<ul>
<li><code>--pop_gnomad</code></li>
<li><code>--report_theme</code></li>
<li>
<code>--preserved_info_tags</code> (previously
<code>custom_tags</code> (TOML))</li>
<li><code>--custom_list_name</code></li>
<li><code>--gwas_p_value</code></li>
<li>
<code>--vcfanno_n_proc</code> (previously
<code>n_vcfanno_proc (TOML)</code>)</li>
<li>
<code>--vep_n_forks</code> (previously
<code>n_vep_forks (TOML)</code>)</li>
<li><code>--vep_pick_order</code></li>
<li>
<code>--vep_no_intergenic</code> (previously
<code>vep_skip_intergenic (TOML)</code>)</li>
</ul>
</li>
<li>New options
<ul>
<li>
<code>--report_nonfloating_toc</code> (<strong>NEW</strong>) - add
the TOC at the top of the HTML report, not floating at the left of the
document</li>
<li>
<code>--report_table_display</code> (<strong>NEW</strong>) - choose
level of comprehensiveness in interactive data tables (full versus light
(default))</li>
</ul>
</li>
</ul>
</li>
<li><p>Improved support for noncoding variant interpretation, primarily
in the context of variants of uncertain significance (VUS). Annotations
will show variants that disrupt/create microRNA target sites (dbMTS),
and variants that overlap transcription factor binding sites (critical
and non-critical positions). Genomic conservation scores (GERP) are also
provided.</p></li>
<li><p>Approx. 100 protein-coding genes have been appended to the <a href="superpanel.html">CPSR superpanel (panel 0)</a></p></li>
</ul>
</div>
<div class="section level5">
<h5 id="fixed">Fixed<a class="anchor" aria-label="anchor" href="#fixed"></a>
</h5>
<ul>
<li>Code typo (<a href="https://github.com/sigven/cpsr/issues/33" class="external-link">issue
#33</a>)</li>
</ul>
</div>
<div class="section level5">
<h5 id="removed-1">Removed<a class="anchor" aria-label="anchor" href="#removed-1"></a>
</h5>
<ul>
<li>Command-line Options
<ul>
<li>
<code>--conf</code> - configuration file</li>
</ul>
</li>
</ul>
</div>
</div>
<div class="section level2">
<h2 id="v0-6-1">v0.6.1<a class="anchor" aria-label="anchor" href="#v0-6-1"></a>
</h2>
<ul>
<li>Date: <strong>2020-11-30</strong>
</li>
</ul>
<div class="section level5">
<h5 id="added-2">Added<a class="anchor" aria-label="anchor" href="#added-2"></a>
</h5>
<ul>
<li>Increased number of genes in panel 0: All genes in 42 virtual panels
related to cancer conditions in Genomics England PanelApp now also
contributes toward panel 0</li>
<li>Added option in main script
(<code>--clinvar_ignore_noncancer</code>) that will exclude any query
variants (from HTML report and TSV/JSON output) that have been reported
and classified for non-cancer related conditions only (in ClinVar)
<ul>
<li>this to exclude variants associated with non-cancer related
phenotypes</li>
</ul>
</li>
<li>For the variant biomarker table, the resolution of the reported
biomarker mapping is highlighted with designated background colors for
the gene (exact/codon - black vs. exon/gene - orange)</li>
</ul>
</div>
<div class="section level5">
<h5 id="fixed-1">Fixed<a class="anchor" aria-label="anchor" href="#fixed-1"></a>
</h5>
<ul>
<li>Bug in GWAS hits retrieval, <a href="https://github.com/sigven/cpsr/issues/18" class="external-link">Issue #30</a>
</li>
<li>Custom VCF tags (as specified by user in configuration file) not
shown in output TSV files</li>
</ul>
</div>
<div class="section level5">
<h5 id="changed-1">Changed<a class="anchor" aria-label="anchor" href="#changed-1"></a>
</h5>
<ul>
<li>Removed DisGeNET annotations from output (associations from Open
Targets Platform serve same purpose)</li>
<li>Renamed report section <strong>Genomic Biomarkers</strong> to
<strong>Variant Biomarkers</strong>
</li>
<li>Option <code>--incidental_findings</code> changed back to
<code>--secondary_findings</code> - recommended term to use according to
ACMG</li>
<li>Removed <em>MOD (mechanism-of-disease)</em> from TSV output
file</li>
</ul>
</div>
</div>
<div class="section level2">
<h2 id="v0-6-0rc">v0.6.0rc<a class="anchor" aria-label="anchor" href="#v0-6-0rc"></a>
</h2>
<ul>
<li><p>Date: <strong>2020-09-24</strong></p></li>
<li><p>Data updates: ClinVar, GWAS catalog, GENCODE, CIViC, CancerMine,
UniProt KB, dbNSFP, Pfam, KEGG, Open Targets Platform, Genomics England
PanelApp</p></li>
<li><p>Software updates: VEP 101</p></li>
</ul>
<div class="section level5">
<h5 id="fixed-2">Fixed<a class="anchor" aria-label="anchor" href="#fixed-2"></a>
</h5>
<ul>
<li>Duplicated entries in incidental findings</li>
</ul>
</div>
<div class="section level5">
<h5 id="changed-2">Changed<a class="anchor" aria-label="anchor" href="#changed-2"></a>
</h5>
<ul>
<li>All arguments to <code>cpsr.py</code> are now non-positional</li>
<li>Arguments to <code>cpsr.py</code> are divided into two groups:
<em>required</em> and <em>optional</em>
</li>
<li>
<code>secondary_findings</code> is now coined
<code>incidental_findings</code>
</li>
<li>Option <strong><em>gwas:gwas_hits</em></strong> in CPSR
configuration file is now optional argument <code>--gwas_findings</code>
in <code>cpsr.py</code>
</li>
<li>Option <strong><em>classification:clinvar_cpsr</em></strong> in CPSR
configuration file is now optional argument <code>--classify_all</code>
in <code>cpsr.py</code>
</li>
<li>Option <strong><em>maf_imits:maf_gnomad</em></strong> in CPSR
configuration file is now optional argument
<code>--maf_upper_threshold</code> in <code>cpsr.py</code>
</li>
<li>Option <strong><em>secondary_findings:show_sf</em></strong> in CPSR
configuration file is now optional argument
<code>--incidental_findings</code> in <code>cpsr.py</code>
</li>
<li>Virtual panels is now displayed through HTML (previously static
ggplot plot)</li>
<li>
<strong>Settings</strong> section of report is now divived into
three:
<ul>
<li>Sample metadata</li>
<li>Report configuration</li>
<li>Virtual panel</li>
</ul>
</li>
<li>Classifications of genes as tumor suppressors/oncogenes are now
based on a combination of CancerMine citation count and presence in
Network of Cancer Genes</li>
</ul>
</div>
<div class="section level5">
<h5 id="added-3">Added<a class="anchor" aria-label="anchor" href="#added-3"></a>
</h5>
<ul>
<li>Missing ACMG criterion for classification of silent and intronic
variants outside of splice regions (<em>ACMG_BP7</em>)</li>
<li>Missing ACMG criterion for classification of variants in promoter
and untranslated regions (<em>ACMG_BP3</em>)</li>
<li>Possibility to create custom virtual panel - any combination of
genes from panel 0 provided as a single-column text file with argument
<code>--custom_list</code>
</li>
<li>Ensured that non-empty datatables pr. tier (<strong>ClinVar</strong>
and <strong>Non-ClinVar</strong>) are set as the active tab</li>
<li>Improved documentation of variant classification in the
<strong>References</strong> section</li>
<li>DOIs available for all references</li>
</ul>
</div>
</div>
<div class="section level2">
<h2 id="v0-5-2">v0.5.2<a class="anchor" aria-label="anchor" href="#v0-5-2"></a>
</h2>
<ul>
<li>Date: <strong>2019-11-18</strong>
</li>
</ul>
<div class="section level5">
<h5 id="changed-3">Changed<a class="anchor" aria-label="anchor" href="#changed-3"></a>
</h5>
<ul>
<li>Definition of pathogenic range (wrt to variant frequency) takes into
account population- and position-specific allele numbers - no longer
defined only by allele counts (i.e. <em>AC</em>) but by <em>AC</em> and
<em>AN</em>
</li>
<li>Moved virtual panel identifier from positional argument to optional
argument (<code>--panel_id</code>) in <code>cpsr.py</code>
</li>
</ul>
</div>
<div class="section level5">
<h5 id="added-4">Added<a class="anchor" aria-label="anchor" href="#added-4"></a>
</h5>
<ul>
<li>Ability to analyze custom panels, provided through option
<code>--custom_panel</code>. Needs to be defined as BED file with four
columns, i.e. chromosome, start, stop, genesymbol</li>
</ul>
</div>
</div>
<div class="section level2">
<h2 id="v0-5-1">v0.5.1<a class="anchor" aria-label="anchor" href="#v0-5-1"></a>
</h2>
<ul>
<li>Date: <strong>2019-10-14</strong>
</li>
</ul>
<div class="section level5">
<h5 id="fixed-3">Fixed<a class="anchor" aria-label="anchor" href="#fixed-3"></a>
</h5>
<ul>
<li>Bug in <code>cpsr_validate_input.py</code>, <a href="https://github.com/sigven/cpsr/issues/18" class="external-link">GitHub Issue</a>
</li>
<li>Bug when there are zero variants with a ‘PASS’ status in VCF -
omitting report generation</li>
</ul>
</div>
</div>
<div class="section level2">
<h2 id="v0-5-0">v0.5.0<a class="anchor" aria-label="anchor" href="#v0-5-0"></a>
</h2>
<ul>
<li>Date: <strong>2019-09-23</strong>
</li>
</ul>
<div class="section level5">
<h5 id="fixed-4">Fixed<a class="anchor" aria-label="anchor" href="#fixed-4"></a>
</h5>
<ul>
<li>Bug in implementation of ACMG criteria; genes without a known
loss-of-function mechanism were handled inappropriately</li>
<li>Bug in assignment of heterozygous/homozygous states (input VCF)</li>
<li>Bug in implementation of ACMG_PS1 - Same amino acid change as
previously pathogenic variant</li>
<li>Improved consequence prioritisation for variants with transcript
consequences in multiple, distinct cancer predisposition genes</li>
<li>Upper MAF threshold (as given by user) only applied for unclassified
(i.e. non-ClinVar variants)</li>
<li>Handling of non-coding variants (synonymous, upstream_variants) in
the report, no longer excluded</li>
</ul>
</div>
<div class="section level5">
<h5 id="added-5">Added<a class="anchor" aria-label="anchor" href="#added-5"></a>
</h5>
<ul>
<li>Section on <em>genomic biomarkers</em>; indicating which variants in
the query VCF that overlaps with existing germline biomarkers
(CIViC)</li>
</ul>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by <a href="https://github.com/sigven" class="external-link">Sigve Nakken</a>, <a href="https://github.com/pdiakumis" class="external-link">Peter Diakumis</a>.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.9.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
