<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="cpsr">
<title>Output files â€¢ CPSR</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.1.0/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.1.0/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.rawgit.com/afeld/bootstrap-toc/v1.0.1/dist/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Output files">
<meta property="og:description" content="cpsr">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-dark navbar-expand-lg bg-primary"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">CPSR</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.7.7</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="../index.html">Intro</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../articles/installation.html">Installation</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../articles/running.html">Running</a>
</li>
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/input.html">Input files</a>
    <a class="dropdown-item" href="../articles/virtual_panels.html">Virtual gene panels</a>
    <a class="dropdown-item" href="../articles/variant_classification.html">ACMG classification</a>
    <a class="dropdown-item" href="../articles/output.html">Output files</a>
    <a class="dropdown-item" href="../articles/annotation_resources.html">Annotation resources</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="../articles/CHANGELOG.html">CHANGELOG</a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/sigven/cpsr/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main"><div class="page-header">
      <img src="" class="logo" alt=""><h1>Output files</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/sigven/cpsr/blob/HEAD/vignettes/output.Rmd" class="external-link"><code>vignettes/output.Rmd</code></a></small>
      <div class="d-none name"><code>output.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="output">Output<a class="anchor" aria-label="anchor" href="#output"></a>
</h2>
<div class="section level3">
<h3 id="interactive-html-report">Interactive HTML report<a class="anchor" aria-label="anchor" href="#interactive-html-report"></a>
</h3>
<p>An interactive and tier-structured HTML report that lists variants in
in known cancer predisposition genes is provided with the following
naming convention:</p>
<p>&lt;<em>sample_id</em>&gt;.<strong>cpsr</strong>.&lt;<em>genome_assembly</em>&gt;.<strong>html</strong></p>
<ul>
<li>The <strong>sample_id</strong> is provided as input by the user, and
reflects a unique identifier of the tumor-normal sample pair to be
analyzed.</li>
</ul>
<p>The report is structured in five main sections, described in more
detail below:</p>
<ol style="list-style-type: decimal">
<li>
<strong>Settings</strong>
<ul>
<li>Lists key configurations provided by user, including the list of
genes that constitute the virtual gene panel in the report</li>
</ul>
</li>
<li>
<strong>Summary of findings</strong>
<ul>
<li>Summarizes the findings through donut charts
<ul>
<li>Number of variants in each of the five variant classification
levels</li>
</ul>
</li>
</ul>
</li>
<li>
<strong>Variant classification</strong>
<ul>
<li>For all coding variants in the selected cancer predisposition
geneset, interactive variant tables are shown for each level
(<strong>ClinVar</strong> and <strong>non-ClinVar (Other)</strong>
variants combined):
<ul>
<li>Pathogenic</li>
<li>Likely Pathogenic</li>
<li>Variants of Uncertain Significance (VUS)</li>
<li>Likely Benign</li>
<li>Benign</li>
</ul>
</li>
<li>Biomarkers
<ul>
<li>Reported clinical evidence items from <a href="https://civicdb.org" class="external-link">CIViC</a> that overlap with variants in the
query set are reported in four distinct tabs (Predictive / Prognostic /
Diagnostic / Predisposing)</li>
</ul>
</li>
<li>Secondary findings
<ul>
<li>Pathogenic variants in the <a href="https://www.ncbi.nlm.nih.gov/clinvar/docs/acmg/" class="external-link">ACMG recommended
list of genes for report of secondary/incidental findings</a>
</li>
</ul>
</li>
<li>GWAS hits
<ul>
<li>Low-risk variants found in genome-wide association studies of cancer
phenotypes (NHGRI-EBI Catalog)</li>
</ul>
</li>
</ul>
</li>
<li>
<strong>Documentation</strong>
<ul>
<li>Introduction
<ul>
<li>Short overview of the predisposition report - aims and contents</li>
</ul>
</li>
<li>Annotation resources
<ul>
<li>Underlying tools, databases and annotation sources (with
versions)</li>
</ul>
</li>
<li>Variant classification
<ul>
<li>Overview of how CPSR performs variant classification of variants not
recorded in ClinVar, listing ACMG criteria and associated scores</li>
</ul>
</li>
<li>References
<ul>
<li>Supporting scientific literature (Interpretation/implementation of
ACMG critera etc.)</li>
</ul>
</li>
</ul>
</li>
</ol>
<div class="section level4">
<h4 id="interactive-datatables">Interactive datatables<a class="anchor" aria-label="anchor" href="#interactive-datatables"></a>
</h4>
<p>The interactive datatables contain a number of hyperlinked
annotations similar to those defined for the annotated VCF file,
including the following:</p>
<ul>
<li>SYMBOL - Gene symbol (Entrez/NCBI)</li>
<li>PROTEIN_CHANGE - Amino acid change (VEP)</li>
<li>GENE_NAME - gene name description (Entrez/NCBI)</li>
<li>PROTEIN_DOMAIN - PFAM protein domain</li>
<li>PROTEIN_FEATURE - UniProt feature overlapping variant site</li>
<li>CDS_CHANGE - Coding sequence change</li>
<li>CONSEQUENCE - VEP consequence (primary transcript)</li>
<li>LOSS_OF_FUNCTION - Predicted loss-of-function variant</li>
<li>RMSK_HIT - Overlap with repeats as annotated by RepeatMasker</li>
<li>HGVSc - from VEP</li>
<li>HGVSp - from VEP</li>
<li>NCBI_REFSEQ - Transcript accession ID(s) (NCBI RefSeq)</li>
<li>ONCOGENE - Predicted proto-oncogene (CancerMine/NCG)</li>
<li>TUMOR_SUPPRESSOR - known tumor suppressor gene (CancerMine/NCG)</li>
<li>PREDICTED_EFFECT - Effect predictions (deleterious/benign) from
dbNSFP</li>
<li>VEP_ALL_CSQ - All VEP transcript block consequences</li>
<li>DBSNP - dbSNP rsID</li>
<li>GENOMIC_CHANGE - Variant ID</li>
<li>GENOME_VERSION - Genome assembly</li>
</ul>
</div>
</div>
<div class="section level3">
<h3 id="json">JSON<a class="anchor" aria-label="anchor" href="#json"></a>
</h3>
<p>A JSON file (gzipped) that stores the HTML report content is
provided. This file will easen the process of extracting particular
parts of the report for further analysis.</p>
<p>The JSON contains two main objects, <em>metadata</em> and
<em>content</em>, where the former contains information about the
settings, data versions, and the latter contains the various sections of
the report.</p>
<p>The JSON file can be used <a href="https://sigven.github.io/pcgr/articles/input.html#cpsr-report" class="external-link">as
input to PCGR</a>, in order to populate a somatic genome report with
germline findings.</p>
<p>At present, there is no detailed schema documented for the JSON
structure.</p>
</div>
<div class="section level3">
<h3 id="variant-call-format---vcf">Variant call format - VCF<a class="anchor" aria-label="anchor" href="#variant-call-format---vcf"></a>
</h3>
<p>A VCF file containing annotated, germline calls (single nucleotide
variants and insertion/deletions) is generated with the following naming
convention:</p>
<p>&lt;<em>sample_id</em>&gt;.<strong>cpsr</strong>.&lt;<em>genome_assembly</em>&gt;.<strong>vcf.gz
(.tbi)</strong></p>
<p>Here, the <strong>sample_id</strong> is provided as input by the
user, and reflects a unique identifier of the tumor-normal sample pair
to be analyzed. Following common standards, the annotated VCF file is
compressed with <a href="http://www.htslib.org/doc/bgzip.html" class="external-link">bgzip</a>
and indexed with <a href="http://www.htslib.org/doc/tabix.html" class="external-link">tabix</a>. Below follows a
description of all annotations/tags present in the VCF INFO column after
processing with the CPSR annotation pipeline:</p>
<div class="section level5">
<h5 id="vep-consequence-annotations">
<em>VEP consequence annotations</em><a class="anchor" aria-label="anchor" href="#vep-consequence-annotations"></a>
</h5>
<ul>
<li>CSQ - Complete consequence annotations from VEP. Format:
Allele|Consequence|IMPACT|SYMBOL|Gene|Feature_type|Feature|BIOTYPE|EXON|INTRON|HGVSc
|HGVSp|cDNA_position|CDS_position|Protein_position|Amino_acids|Codons|Existing_variation|
ALLELE_NUM|DISTANCE|STRAND|FLAGS|PICK|VARIANT_CLASS|SYMBOL_SOURCE|HGNC_ID|
CANONICAL|MANE|TSL|APPRIS|CCDS|ENSP|SWISSPROT|TREMBL|UNIPARC|RefSeq|DOMAINS|
HGVS_OFFSET|AF|AFR_AF|AMR_AF|EAS_AF|EUR_AF|SAS_AF|gnomAD_AF|gnomAD_AFR_AF|
gnomAD_AMR_AF|gnomAD_ASJ_AF|gnomAD_EAS_AF|gnomAD_FIN_AF|gnomAD_NFE_AF|
gnomAD_OTH_AF|gnomAD_SAS_AF|CLIN_SIG|SOMATIC|PHENO|CHECK_REF|
MOTIF_NAME|MOTIF_POS|HIGH_INF_POS|MOTIF_SCORE_CHANGE|TRANSCRIPTION_FACTORS|NearestExonJB</li>
<li>Consequence - Impact modifier for the consequence type (picked by
VEPâ€™s â€“flag_pick_allele option)</li>
<li>Gene - Ensembl stable ID of affected gene (picked by VEPâ€™s
â€“flag_pick_allele option)</li>
<li>Feature_type - Type of feature. Currently one of Transcript,
RegulatoryFeature, MotifFeature (picked by VEPâ€™s â€“flag_pick_allele
option)</li>
<li>Feature - Ensembl stable ID of feature (picked by VEPâ€™s
â€“flag_pick_allele option)</li>
<li>cDNA_position - Relative position of base pair in cDNA sequence
(picked by VEPâ€™s â€“flag_pick_allele option)</li>
<li>CDS_position - Relative position of base pair in coding sequence
(picked by VEPâ€™s â€“flag_pick_allele option)</li>
<li>CDS_CHANGE - Coding, transcript-specific sequence annotation (picked
by VEPâ€™s â€“flag_pick_allele option)</li>
<li>AMINO_ACID_START - Protein position indicating absolute start of
amino acid altered (fetched from Protein_position)</li>
<li>AMINO_ACID_END - Protein position indicating absolute end of amino
acid altered (fetched from Protein_position)</li>
<li>Protein_position - Relative position of amino acid in protein
(picked by VEPâ€™s â€“flag_pick_allele option)</li>
<li>Amino_acids - Only given if the variant affects the protein-coding
sequence (picked by VEPâ€™s â€“flag_pick_allele option)</li>
<li>Codons - The alternative codons with the variant base in upper case
(picked by VEPâ€™s â€“flag_pick_allele option)</li>
<li>IMPACT - Impact modifier for the consequence type (picked by VEPâ€™s
â€“flag_pick_allele option)</li>
<li>VARIANT_CLASS - Sequence Ontology variant class (picked by VEPâ€™s
â€“flag_pick_allele option)</li>
<li>SYMBOL - Gene symbol (picked by VEPâ€™s â€“flag_pick_allele option)</li>
<li>SYMBOL_ENTREZ - Official gene symbol as provided by NCBIâ€™s Entrez
gene</li>
<li>SYMBOL_SOURCE - The source of the gene symbol (picked by VEPâ€™s
â€“flag_pick_allele option)</li>
<li>STRAND - The DNA strand (1 or -1) on which the transcript/feature
lies (picked by VEPâ€™s â€“flag_pick_allele option)</li>
<li>ENSP - The Ensembl protein identifier of the affected transcript
(picked by VEPâ€™s â€“flag_pick_allele option)</li>
<li>FLAGS - Transcript quality flags: cds_start_NF: CDS 5â€™, incomplete
cds_end_NF: CDS 3â€™ incomplete (picked by VEPâ€™s â€“flag_pick_allele
option)</li>
<li>SWISSPROT - Best match UniProtKB/Swiss-Prot accession of protein
product (picked by VEPâ€™s â€“flag_pick_allele option)</li>
<li>TREMBL - Best match UniProtKB/TrEMBL accession of protein product
(picked by VEPâ€™s â€“flag_pick_allele option)</li>
<li>UNIPARC - Best match UniParc accession of protein product (picked by
VEPâ€™s â€“flag_pick_allele option)</li>
<li>HGVSc - The HGVS coding sequence name (picked by VEPâ€™s
â€“flag_pick_allele option)</li>
<li>HGVSp - The HGVS protein sequence name (picked by VEPâ€™s
â€“flag_pick_allele option)</li>
<li>HGVSp_short - The HGVS protein sequence name, short version (picked
by VEPâ€™s â€“flag_pick_allele option)</li>
<li>HGVS_OFFSET - Indicates by how many bases the HGVS notations for
this variant have been shifted (picked by VEPâ€™s â€“flag_pick_allele
option)</li>
<li>NearestExonJB - VEP plugin that finds nearest exon junction for a
coding sequence variant. Format: Ensembl exon identifier+distanceto exon
boundary+boundary type(start/end)+exon length</li>
<li>MOTIF_NAME - The source and identifier of a transcription factor
binding profile aligned at this position (picked by VEPâ€™s
â€“flag_pick_allele option)</li>
<li>MOTIF_POS - The relative position of the variation in the aligned
TFBP (picked by VEPâ€™s â€“flag_pick_allele option)</li>
<li>HIGH_INF_POS - A flag indicating if the variant falls in a high
information position of a transcription factor binding profile (TFBP)
(picked by VEPâ€™s â€“flag_pick_allele option)</li>
<li>MOTIF_SCORE_CHANGE - The difference in motif score of the reference
and variant sequences for the TFBP (picked by VEPâ€™s â€“flag_pick_allele
option)</li>
<li>CELL_TYPE - List of cell types and classifications for regulatory
feature (picked by VEPâ€™s â€“flag_pick_allele option)</li>
<li>CANONICAL - A flag indicating if the transcript is denoted as the
canonical transcript for this gene (picked by VEPâ€™s â€“flag_pick_allele
option)</li>
<li>CCDS - The CCDS identifier for this transcript, where applicable
(picked by VEPâ€™s â€“flag_pick_allele option)</li>
<li>INTRON - The intron number (out of total number) (picked by VEPâ€™s
â€“flag_pick_allele option)</li>
<li>INTRON_POSITION - Relative position of intron variant to nearest
exon/intron junction (NearestExonJB VEP plugin)</li>
<li>EXON_POSITION - Relative position of exon variant to nearest
intron/exon junction (NearestExonJB VEP plugin)</li>
<li>EXON - The exon number (out of total number) (picked by VEPâ€™s
â€“flag_pick_allele option)</li>
<li>LAST_EXON - Logical indicator for last exon of transcript (picked by
VEPâ€™s â€“flag_pick_allele option)</li>
<li>LAST_INTRON - Logical indicator for last intron of transcript
(picked by VEPâ€™s â€“flag_pick_allele option)</li>
<li>DISTANCE - Shortest distance from variant to transcript (picked by
VEPâ€™s â€“flag_pick_allele option)</li>
<li>BIOTYPE - Biotype of transcript or regulatory feature (picked by
VEPâ€™s â€“flag_pick_allele option)</li>
<li>TSL - Transcript support level (picked by VEPâ€™s â€“flag_pick_allele
option)&gt;</li>
<li>PUBMED - PubMed ID(s) of publications that cite existing variant -
VEP</li>
<li>PHENO - Indicates if existing variant is associated with a
phenotype, disease or trait - VEP</li>
<li>GENE_PHENO - Indicates if overlapped gene is associated with a
phenotype, disease or trait - VEP</li>
<li>ALLELE_NUM - Allele number from input; 0 is reference, 1 is first
alternate etc - VEP</li>
<li>REFSEQ_MATCH - The RefSeq transcript match status; contains a number
of flags indicating whether this RefSeq transcript matches the
underlying reference sequence and/or an Ensembl transcript (picked by
VEPâ€™s â€“flag_pick_allele option)</li>
<li>PICK - Indicates if this block of consequence data was picked by
VEPâ€™s â€“flag_pick_allele option</li>
<li>VEP_ALL_CSQ - All VEP transcript block consequences
(Consequence:SYMBOL:Feature_type:Feature:BIOTYPE) - VEP</li>
<li>EXONIC_STATUS - Indicates if variant consequence type is â€˜exonicâ€™ or
â€˜nonexonicâ€™. We define â€˜exonicâ€™ as any variants with the following
consequences:
<ul>
<li>stop_gained / stop_lost</li>
<li>start_lost</li>
<li>frameshift_variant</li>
<li>missense_variant</li>
<li>splice_donor_variant</li>
<li>splice_acceptor_variant</li>
<li>inframe_insertion / inframe_deletion</li>
<li>synonymous_variant</li>
<li>protein_altering</li>
</ul>
</li>
<li>CODING_STATUS - Indicates if primary variant consequence type is
â€˜codingâ€™ or â€˜noncodingâ€™. â€˜codingâ€™ variants are here defined as those
with an â€˜exonicâ€™ status, with the exception of synonymous variants</li>
<li>NULL_VARIANT - Primary variant consequence type is frameshift or
stop_gained/stop_lost</li>
<li>SPLICE_DONOR_RELEVANT - Logical indicating if variant is located at
a particular location near the splice donor site (+3A/G, +4A or
+5G)</li>
<li>REGULATORY_ANNOTATION - Comma-separated list of all variant
annotations of Feature_type RegulatoryFeature and MotifFeature. Format:
<consequence>|<feature_type>|<feature>|<biotype>|<motif_name>|<motif_pos>|<high_inf_pos>|<motif_score_change>|<transcription_factors></transcription_factors></motif_score_change></high_inf_pos></motif_pos></motif_name></biotype></feature></feature_type></consequence>
</li>
</ul>
</div>
<div class="section level5">
<h5 id="gene-information">
<em>Gene information</em><a class="anchor" aria-label="anchor" href="#gene-information"></a>
</h5>
<ul>
<li>ENTREZ_ID - <a href="http://www.ncbi.nlm.nih.gov/gene" class="external-link">Entrez</a>
gene identifier</li>
<li>APPRIS - Principal isoform flags according to the <a href="http://appris.bioinfo.cnio.es/#/downloads" class="external-link">APPRIS principal
isoform database</a>
</li>
<li>MANE_SELECT - Indicating if the transcript is the MANE Select or
MANE Plus Clinical transcript for the gene (picked by VEPâ€™s
â€“flag_pick_allele_gene option)</li>
<li>UNIPROT_ID - <a href="http://www.uniprot.org" class="external-link">UniProt</a>
identifier</li>
<li>UNIPROT_ACC - <a href="http://www.uniprot.org" class="external-link">UniProt</a>
accession(s)</li>
<li>ENSEMBL_GENE_ID - Ensembl gene identifier for VEPâ€™s picked
transcript (<em>ENSGXXXXXXX</em>)</li>
<li>ENSEMBL_TRANSCRIPT_ID - Ensembl transcript identifier for VEPâ€™s
picked transcript (<em>ENSTXXXXXX</em>)</li>
<li>ENSEMBL_PROTEIN_ID - Ensembl corresponding protein identifier for
VEPâ€™s picked transcript</li>
<li>REFSEQ_MRNA - Corresponding RefSeq transcript(s) identifier for
VEPâ€™s picked transcript (<em>NM_XXXXX</em>)</li>
<li>TRANSCRIPT_MANE_SELECT - MANE select transcript identifer: one
high-quality representative transcript per protein-coding gene that is
well-supported by experimental data and represents the biology of the
gene</li>
<li>TRANSCRIPT_MANE_PLUS_CLINICAL - transcripts chosen to supplement
MANE Select when needed for clinical variant reporting</li>
<li>GENCODE_TAG - tag for gencode transcript (basic etc)</li>
<li>GENCODE_TRANSCRIPT_TYPE - type of transcript (protein-coding
etc.)</li>
<li>CORUM_ID - Associated protein complexes (identifiers) from <a href="http://mips.helmholtz-muenchen.de/corum/" class="external-link">CORUM</a>
</li>
<li>TUMOR_SUPPRESSOR - Indicates whether gene is predicted as a tumor
suppressor gene, from Network of Cancer Genes (NCG) &amp; the CancerMine
text-mining resource</li>
<li>TUMOR_SUPPRESSOR_EVIDENCE - Underlying evidence for gene being a
tumor suppressor. Format:
NCG:&lt;TRUE|FALSE&gt;&amp;CancerMine:&lt;LC|MC|HC&gt;:num_citations</li>
<li>ONCOGENE - Indicates whether gene is predicted as an oncogene, from
Network of Cancer Genes (NCG) &amp; the CancerMine text-mining
resource</li>
<li>ONCOGENE_EVIDENCE - Underlying evidence for gene being an oncogene.
Format:
NCG:&lt;TRUE|FALSE&gt;&amp;CancerMine:&lt;LC|MC|HC&gt;:num_citations</li>
<li>CANCER_SUSCEPTIBILITY_CUI - MedGen concept unique identifier (CUI)
for cancer phenotype</li>
<li>CANCER_SYNDROME_CUI - MedGen concept unique identifier (CUI) for
cancer syndrome</li>
<li>CANCER_PREDISPOSITION_SOURCE - Data source for susceptibility gene
(panel <em>0</em>: NCGC, CGC_94, TCGA_PANCAN, PANEL_APP, OTHER)</li>
<li>CANCER_PREDISPOSITION_MOI - Mode of inheritance for susceptibility
gene (AR/AD)</li>
<li>CANCER_PREDISPOSITION_MOD - Mechanism of disease for susceptibility
gene (Lof/GoF)</li>
<li>PROB_EXAC_LOF_INTOLERANT - dbNSFP_gene: the probability of being
loss-of-function intolerant (intolerant of both heterozygous and
homozygous lof variants) based on ExAC r0.3 data</li>
<li>PROB_EXAC_LOF_INTOLERANT_HOM - dbNSFP_gene: the probability of being
intolerant of homozygous, but not heterozygous lof variants based on
ExAC r0.3 data</li>
<li>PROB_EXAC_LOF_TOLERANT_NULL - dbNSFP_gene: the probability of being
tolerant of both heterozygous and homozygous lof variants based on ExAC
r0.3 data</li>
<li>PROB_EXAC_NONTCGA_LOF_INTOLERANT - dbNSFP_gene: the probability of
being loss-of-function intolerant (intolerant of both heterozygous and
homozygous lof variants) based on ExAC r0.3 nonTCGA subset</li>
<li>PROB_EXAC_NONTCGA_LOF_INTOLERANT_HOM - dbNSFP_gene: the probability
of being intolerant of homozygous, but not heterozygous lof variants
based on ExAC r0.3 nonTCGA subset</li>
<li>PROB_EXAC_NONTCGA_LOF_TOLERANT_NULL - dbNSFP_gene: the probability
of being tolerant of both heterozygous and homozygous lof variants based
on ExAC r0.3 nonTCGA subset</li>
<li>PROB_GNOMAD_LOF_INTOLERANT - dbNSFP_gene: the probability of being
loss-of-function intolerant (intolerant of both heterozygous and
homozygous lof variants based on gnomAD 2.1 data</li>
<li>PROB_GNOMAD_LOF_INTOLERANT_HOM - dbNSFP_gene: the probability of
being intolerant of homozygous, but not heterozygous lof variants based
on gnomAD 2.1 data</li>
<li>PROB_GNOMAD_LOF_TOLERANT_NULL - dbNSFP_gene: the probability of
being tolerant of both heterozygous and homozygous lof variants based on
gnomAD 2.1 data</li>
<li>PROB_HAPLOINSUFFICIENCY - dbNSFP_gene: Estimated probability of
haploinsufficiency of the gene (from <a href="http://dx.doi.org/10.1371/journal.pgen.1001154" class="external-link uri">http://dx.doi.org/10.1371/journal.pgen.1001154</a>)</li>
<li>ESSENTIAL_GENE_CRISPR - dbNSFP_gene: Essential (E) or Non-essential
phenotype-changing (N) based on large scale CRISPR experiments. from <a href="http://dx.doi.org/10.1126/science.aac7041" class="external-link uri">http://dx.doi.org/10.1126/science.aac7041</a>
</li>
<li>ESSENTIAL_GENE_CRISPR2 - dbNSFP_gene: Essential (E),
context-Specific essential (S), or Non-essential phenotype-changing (N)
based on large scale CRISPR experiments. from <a href="http://dx.doi.org/10.1016/j.cell.2015.11.015" class="external-link uri">http://dx.doi.org/10.1016/j.cell.2015.11.015</a>
</li>
</ul>
</div>
<div class="section level5">
<h5 id="variant-effect-and-protein-coding-information">
<em>Variant effect and protein-coding information</em><a class="anchor" aria-label="anchor" href="#variant-effect-and-protein-coding-information"></a>
</h5>
<ul>
<li><p>MUTATION_HOTSPOT - mutation hotspot codon in <a href="http://cancerhotspots.org/" class="external-link">cancerhotspots.org</a>. Format:
gene_symbol | codon | q-value</p></li>
<li><p>MUTATION_HOTSPOT_TRANSCRIPT - hotspot-associated transcripts
(Ensembl transcript ID)</p></li>
<li><p>MUTATION_HOTSPOT_CANCERTYPE - hotspot-associated cancer types
(from cancerhotspots.org)</p></li>
<li><p>UNIPROT_FEATURE - Overlapping protein annotations from <a href="http://www.uniprot.org" class="external-link">UniProt KB</a></p></li>
<li><p>PFAM_DOMAIN - Pfam domain identifier (from VEP)</p></li>
<li>
<p>EFFECT_PREDICTIONS - All predictions of effect of variant on
protein function and pre-mRNA splicing from <a href="https://sites.google.com/site/jpopgen/dbNSFP" class="external-link">database of
non-synonymous functional predictions - dbNSFP v4.2</a>. Predicted
effects are provided by different sources/algorithms (separated by
â€˜&amp;â€™), T = Tolerated, N = Neutral, D = Damaging:</p>
<ol style="list-style-type: decimal">
<li><a href="https://sift.bii.a-star.edu.sg/" class="external-link">SIFT</a></li>
<li>
<a href="http://www.mutationtaster.org/" class="external-link">MutationTaster</a> (data
release Nov 2015)</li>
<li>
<a href="http://mutationassessor.org/" class="external-link">MutationAssessor</a> (release
3)</li>
<li>
<a href="http://fathmm.biocompute.org.uk" class="external-link">FATHMM</a> (v2.3)</li>
<li>
<a href="http://provean.jcvi.org/index.php" class="external-link">PROVEAN</a> (v1.1 Jan
2015)</li>
<li><a href="http://fathmm.biocompute.org.uk/fathmmMKL.htm" class="external-link">FATHMM_MKL</a></li>
<li><a href="https://www.nature.com/articles/s41588-018-0167-z" class="external-link">PRIMATEAI</a></li>
<li><a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5570203/" class="external-link">DEOGEN2</a></li>
<li>
<a href="https://www.biorxiv.org/content/10.1101/2021.04.09.438706v1" class="external-link">DBNSFP_CONSENSUS_RNN</a>
(Ensembl/consensus prediction, based on deep learning)</li>
<li>
<a href="http://nar.oxfordjournals.org/content/42/22/13534" class="external-link">SPLICE_SITE_EFFECT_ADA</a>
(Ensembl/consensus prediction of splice-altering SNVs, based on adaptive
boosting)</li>
<li>
<a href="http://nar.oxfordjournals.org/content/42/22/13534" class="external-link">SPLICE_SITE_EFFECT_RF</a>
(Ensembl/consensus prediction of splice-altering SNVs, based on random
forest)</li>
<li><a href="http://bejerano.stanford.edu/MCAP" class="external-link">M-CAP</a></li>
<li><a href="http://mutpred.mutdb.org" class="external-link">MutPred</a></li>
<li><a href="http://mendel.stanford.edu/SidowLab/downloads/gerp/" class="external-link">GERP</a></li>
<li><a href="https://doi.org/10.1002/humu.23158" class="external-link">BayesDel</a></li>
<li><a href="https://doi.org/10.1093/nar/gkaa288" class="external-link">LIST-S2</a></li>
<li><a href="https://www.nature.com/articles/s41467-017-00443-5" class="external-link">ALoFT</a></li>
</ol>
</li>
<li><p>DBNSFP_BAYESDEL_ADDAF - predicted effect from BayesDel
(dbNSFP)</p></li>
<li><p>DBNSFP_LIST_S2 - predicted effect from LIST-S2 (dbNSFP)</p></li>
<li><p>DBNSFP_SIFT - predicted effect from SIFT (dbNSFP)</p></li>
<li><p>DBNSFP_PROVEAN - predicted effect from PROVEAN (dbNSFP)</p></li>
<li><p>DBNSFP_MUTATIONTASTER - predicted effect from MUTATIONTASTER
(dbNSFP)</p></li>
<li><p>DBNSFP_MUTATIONASSESSOR - predicted effect from MUTATIONASSESSOR
(dbNSFP)</p></li>
<li><p>DBNSFP_M_CAP - predicted effect from M-CAP (dbNSFP)</p></li>
<li><p>DBNSFP_ALOFTPRED - predicted effect from ALoFT (dbNSFP)</p></li>
<li><p>DBNSFP_MUTPRED - score from MUTPRED (dbNSFP)</p></li>
<li><p>DBNSFP_FATHMM - predicted effect from FATHMM (dbNSFP)</p></li>
<li><p>DBNSFP_PRIMATEAI - predicted effect from PRIMATEAI
(dbNSFP)</p></li>
<li><p>DBNSFP_DEOGEN2 - predicted effect from DEOGEN2 (dbNSFP)</p></li>
<li><p>DBNSFP_GERP - evolutionary constraint measure from GERP
(dbNSFP)</p></li>
<li><p>DBNSFP_FATHMM_MKL - predicted effect from FATHMM-mkl
(dbNSFP)</p></li>
<li><p>DBNSFP_META_RNN - predicted effect from ensemble prediction (deep
learning - dbNSFP)</p></li>
<li><p>DBNSFP_SPLICE_SITE_RF - predicted effect of splice site
disruption, using random forest (dbscSNV)</p></li>
<li><p>DBNSFP_SPLICE_SITE_ADA - predicted effect of splice site
disruption, using boosting (dbscSNV)</p></li>
</ul>
</div>
<div class="section level5">
<h5 id="variant-frequenciesannotations-in-germline-databases">
<em>Variant frequencies/annotations in germline databases</em><a class="anchor" aria-label="anchor" href="#variant-frequenciesannotations-in-germline-databases"></a>
</h5>
<ul>
<li>AFR_AF_GNOMAD - African/American germline allele frequency (<a href="http://gnomad.broadinstitute.org/" class="external-link">Genome Aggregation Database
release 2.1</a>)</li>
<li>AMR_AF_GNOMAD - American germline allele frequency (<a href="http://gnomad.broadinstitute.org/" class="external-link">Genome Aggregation Database
release 2.1</a>)</li>
<li>GLOBAL_AF_GNOMAD - Adjusted global germline allele frequency (<a href="http://gnomad.broadinstitute.org/" class="external-link">Genome Aggregation Database
release 2.1</a>)</li>
<li>SAS_AF_GNOMAD - South Asian germline allele frequency (<a href="http://gnomad.broadinstitute.org/" class="external-link">Genome Aggregation Database
release 2.1</a>)</li>
<li>EAS_AF_GNOMAD - East Asian germline allele frequency (<a href="http://gnomad.broadinstitute.org/" class="external-link">Genome Aggregation Database
release 2.1</a>)</li>
<li>FIN_AF_GNOMAD - Finnish germline allele frequency (<a href="http://gnomad.broadinstitute.org/" class="external-link">Genome Aggregation Database
release 2.1</a>)</li>
<li>NFE_AF_GNOMAD - Non-Finnish European germline allele frequency (<a href="http://gnomad.broadinstitute.org/" class="external-link">Genome Aggregation Database
release 2.1</a>)</li>
<li>OTH_AF_GNOMAD - Other germline allele frequency (<a href="http://gnomad.broadinstitute.org/" class="external-link">Genome Aggregation Database
release 2.1</a>)</li>
<li>ASJ_AF_GNOMAD - Ashkenazi Jewish allele frequency (<a href="http://gnomad.broadinstitute.org/" class="external-link">Genome Aggregation Database
release 2.1</a>)</li>
<li>NON_CANCER_AF_ASJ - Alternate allele frequency for samples of
Ashkenazi Jewish ancestry in the non_cancer subset (<a href="http://gnomad.broadinstitute.org" class="external-link">gnomAD 2.1.1</a>)</li>
<li>NON_CANCER_AF_EAS - Alternate allele frequency for samples of East
Asian ancestry in the non_cancer subset (<a href="http://gnomad.broadinstitute.org" class="external-link">gnomAD 2.1.1</a>)</li>
<li>NON_CANCER_AF_AFR - Alternate allele frequency for samples of
African-American/African ancestry in the non_cancer subset (<a href="http://gnomad.broadinstitute.org" class="external-link">gnomAD 2.1.1</a>)</li>
<li>NON_CANCER_AF_AMR - Alternate allele frequency for samples of Latino
ancestry in the non_cancer subset (<a href="http://gnomad.broadinstitute.org" class="external-link">gnomAD 2.1.1</a>)</li>
<li>NON_CANCER_AF_OTH - Alternate allele frequency for samples of Other
ancestry in the non_cancer subset (<a href="http://gnomad.broadinstitute.org" class="external-link">gnomAD 2.1.1</a>)</li>
<li>NON_CANCER_AF_NFE - Alternate allele frequency for samples of
Non-Finnish European ancestry in the non_cancer subset (<a href="http://gnomad.broadinstitute.org" class="external-link">gnomAD 2.1.1</a>)</li>
<li>NON_CANCER_AF_FIN - Alternate allele frequency for samples of
Finnish ancestry in the non_cancer subset (<a href="http://gnomad.broadinstitute.org" class="external-link">gnomAD 2.1.1</a>)</li>
<li>NON_CANCER_AF_SAS - Alternate allele frequency for samples of South
Asian ancestry in the non_cancer subset (<a href="http://gnomad.broadinstitute.org" class="external-link">gnomAD 2.1.1</a>)</li>
<li>NON_CANCER_AF_GLOBAL - Alternate allele frequency in the non_cancer
subset (<a href="http://gnomad.broadinstitute.org" class="external-link">gnomAD
2.1.1</a>)</li>
<li>NON_CANCER_AC_ASJ - Alternate allele count for samples of Ashkenazi
Jewish ancestry in the non_cancer subset (<a href="http://gnomad.broadinstitute.org" class="external-link">gnomAD 2.1.1</a>)</li>
<li>NON_CANCER_AC_EAS - Alternate allele count for samples of East Asian
ancestry in the non_cancer subset (<a href="http://gnomad.broadinstitute.org" class="external-link">gnomAD 2.1.1</a>)</li>
<li>NON_CANCER_AC_AFR - Alternate allele count for samples of
African-American/African ancestry in the non_cancer subset (<a href="http://gnomad.broadinstitute.org" class="external-link">gnomAD 2.1.1</a>)</li>
<li>NON_CANCER_AC_AMR - Alternate allele count for samples of Latino
ancestry in the non_cancer subset (<a href="http://gnomad.broadinstitute.org" class="external-link">gnomAD 2.1.1</a>)</li>
<li>NON_CANCER_AC_OTH - Alternate allele count for samples of Other
ancestry in the non_cancer subset (<a href="http://gnomad.broadinstitute.org" class="external-link">gnomAD 2.1.1</a>)</li>
<li>NON_CANCER_AC_NFE - Alternate allele frequency for samples of
Non-Finnish European ancestry in the non_cancer subset (<a href="http://gnomad.broadinstitute.org" class="external-link">gnomAD 2.1.1</a>)</li>
<li>NON_CANCER_AC_FIN - Alternate allele count for samples of Finnish
ancestry in the non_cancer subset (<a href="http://gnomad.broadinstitute.org" class="external-link">gnomAD 2.1.1</a>)</li>
<li>NON_CANCER_AC_SAS - Alternate allele count for samples of South
Asian ancestry in the non_cancer subset (<a href="http://gnomad.broadinstitute.org" class="external-link">gnomAD 2.1.1</a>)</li>
<li>NON_CANCER_AC_GLOBAL - Alternate allele count in the non_cancer
subset (<a href="http://gnomad.broadinstitute.org" class="external-link">gnomAD
2.1.1</a>)</li>
<li>NON_CANCER_AN_ASJ - Total number of alleles in samples of Ashkenazi
Jewish ancestry in the non_cancer subset (<a href="http://gnomad.broadinstitute.org" class="external-link">gnomAD 2.1.1</a>)</li>
<li>NON_CANCER_AN_EAS - Total number of alleles in samples of East Asian
ancestry in the non_cancer subset (<a href="http://gnomad.broadinstitute.org" class="external-link">gnomAD 2.1.1</a>)</li>
<li>NON_CANCER_AN_AFR - Total number of alleles in samples of
African-American/African ancestry in the non_cancer subset (<a href="http://gnomad.broadinstitute.org" class="external-link">gnomAD 2.1.1</a>)</li>
<li>NON_CANCER_AN_AMR - Total number of alleles in samples of Latino
ancestry in the non_cancer subset (<a href="http://gnomad.broadinstitute.org" class="external-link">gnomAD 2.1.1</a>)</li>
<li>NON_CANCER_AN_OTH - Total number of alleles in samples of Other
ancestry in the non_cancer subset (<a href="http://gnomad.broadinstitute.org" class="external-link">gnomAD 2.1.1</a>)</li>
<li>NON_CANCER_AN_NFE - Total number of alleles in samples of
Non-Finnish European ancestry in the non_cancer subset (<a href="http://gnomad.broadinstitute.org" class="external-link">gnomAD 2.1.1</a>)</li>
<li>NON_CANCER_AN_FIN - Total number of alleles in samples of Finnish
ancestry in the non_cancer subset (<a href="http://gnomad.broadinstitute.org" class="external-link">gnomAD 2.1.1</a>)</li>
<li>NON_CANCER_AN_SAS - Total number of alleles in samples of South
Asian ancestry in the non_cancer subset (<a href="http://gnomad.broadinstitute.org" class="external-link">gnomAD 2.1.1</a>)</li>
<li>NON_CANCER_AN_GLOBAL - Total number of alleles in the non_cancer
subset (<a href="http://gnomad.broadinstitute.org" class="external-link">gnomAD
2.1.1</a>)</li>
<li>NON_CANCER_NHOMALT_ASJ - Count of homozygous individuals in samples
of Ashkenazi Jewish ancestry in the non_cancer subset (<a href="http://gnomad.broadinstitute.org" class="external-link">gnomAD 2.1.1</a>)</li>
<li>NON_CANCER_NHOMALT_EAS - Count of homozygous individuals in samples
of East Asian ancestry in the non_cancer subset (<a href="http://gnomad.broadinstitute.org" class="external-link">gnomAD 2.1.1</a>)</li>
<li>NON_CANCER_NHOMALT_AFR - Count of homozygous individuals in samples
of African-American/African ancestry in the non_cancer subset (<a href="http://gnomad.broadinstitute.org" class="external-link">gnomAD 2.1.1</a>)</li>
<li>NON_CANCER_NHOMALT_AMR - Count of homozygous individuals in samples
of Latino ancestry in the non_cancer subset (<a href="http://gnomad.broadinstitute.org" class="external-link">gnomAD 2.1.1</a>)</li>
<li>NON_CANCER_NHOMALT_OTH - Count of homozygous individuals in samples
of Other ancestry in the non_cancer subset (<a href="http://gnomad.broadinstitute.org" class="external-link">gnomAD 2.1.1</a>)</li>
<li>NON_CANCER_NHOMALT_NFE - Count of homozygous individuals in samples
of Non-Finnish European ancestry in the non_cancer subset (<a href="http://gnomad.broadinstitute.org" class="external-link">gnomAD 2.1.1</a>)</li>
<li>NON_CANCER_NHOMALT_FIN - Count of homozygous individuals in samples
of Finnish ancestry in the non_cancer subset (<a href="http://gnomad.broadinstitute.org" class="external-link">gnomAD 2.1.1</a>)</li>
<li>NON_CANCER_NHOMALT_SAS - Count of homozygous individuals in samples
of South Asian ancestry in the non_cancer subset (<a href="http://gnomad.broadinstitute.org" class="external-link">gnomAD 2.1.1</a>)</li>
<li>NON_CANCER_NHOMALT_GLOBAL - Count of homozygous individuals in
samples in the non_cancer subset (<a href="http://gnomad.broadinstitute.org" class="external-link">gnomAD 2.1.1</a>)</li>
<li>AFR_AF_1KG - <a href="http://www.1000genomes.org" class="external-link">1000G Project -
phase 3</a> germline allele frequency for samples from AFR
(African)</li>
<li>AMR_AF_1KG - <a href="http://www.1000genomes.org" class="external-link">1000G Project -
phase 3</a> germline allele frequency for samples from AMR (Ad Mixed
American)</li>
<li>EAS_AF_1KG - <a href="http://www.1000genomes.org" class="external-link">1000G Project -
phase 3</a> germline allele frequency for samples from EAS (East
Asian)</li>
<li>EUR_AF_1KG - <a href="http://www.1000genomes.org" class="external-link">1000G Project -
phase 3</a> germline allele frequency for samples from EUR
(European)</li>
<li>SAS_AF_1KG - <a href="http://www.1000genomes.org" class="external-link">1000G Project -
phase 3</a> germline allele frequency for samples from SAS (South
Asian)</li>
<li>GLOBAL_AF_1KG - <a href="http://www.1000genomes.org" class="external-link">1000G Project -
phase 3</a> germline allele frequency for all 1000G project samples
(global)</li>
<li>DBSNPRSID - <a href="http://www.ncbi.nlm.nih.gov/SNP/" class="external-link">dbSNP</a>
reference ID, as provided by VEP</li>
</ul>
</div>
<div class="section level5">
<h5 id="clinical-associations">
<em>Clinical associations</em><a class="anchor" aria-label="anchor" href="#clinical-associations"></a>
</h5>
<ul>
<li>CLINVAR_MSID - <a href="http://www.ncbi.nlm.nih.gov/clinvar" class="external-link">ClinVar</a> Measure
Set/Variant ID</li>
<li>CLINVAR_ALLELE_ID - <a href="http://www.ncbi.nlm.nih.gov/clinvar" class="external-link">ClinVar</a> allele ID</li>
<li>CLINVAR_PMID - Associated Pubmed IDs for variant in <a href="http://www.ncbi.nlm.nih.gov/clinvar" class="external-link">ClinVar</a> - germline
state-of-origin</li>
<li>CLINVAR_HGVSP - Protein variant expression using HGVS
nomenclature</li>
<li>CLINVAR_PMID_SOMATIC - Associated Pubmed IDs for variant in <a href="http://www.ncbi.nlm.nih.gov/clinvar" class="external-link">ClinVar</a> - somatic
state-of-origin</li>
<li>CLINVAR_CONFLICTED - Variant has conflicting interpretations</li>
<li>CLINVAR_CLNSIG - Clinical significance for variant in <a href="http://www.ncbi.nlm.nih.gov/clinvar" class="external-link">ClinVar</a> - germline
state-of-origin</li>
<li>CLINVAR_CLASSIFICATION - Clean clinical significance on a five-level
scheme</li>
<li>CLINVAR_CLNSIG_SOMATIC - Clinical significance for variant in <a href="http://www.ncbi.nlm.nih.gov/clinvar" class="external-link">ClinVar</a> - somatic
state-of-origin</li>
<li>CLINVAR_MEDGEN_CUI - Associated <a href="https://www.ncbi.nlm.nih.gov/medgen/" class="external-link">MedGen</a> concept
identifiers (<em>CUIs</em>) - germline state-of-origin</li>
<li>CLINVAR_MEDGEN_CUI_SOMATIC - Associated <a href="https://www.ncbi.nlm.nih.gov/medgen/" class="external-link">MedGen</a> concept
identifiers (<em>CUIs</em>) - somatic state-of-origin</li>
<li>CLINVAR_MOLECULAR_EFFECT - Variant effect according to ClinVar
annotation</li>
<li>CLINVAR_VARIANT_ORIGIN - Origin of variant (somatic, germline, de
novo etc.) for variant in <a href="http://www.ncbi.nlm.nih.gov/clinvar" class="external-link">ClinVar</a>
</li>
<li>CLINVAR_REVIEW_STATUS_STARS - Rating of the <a href="http://www.ncbi.nlm.nih.gov/clinvar" class="external-link">ClinVar</a> variant (0-4
stars) with respect to level of review</li>
<li>GWAS_HIT - variant associated with cancer phenotype from genome-wide
association study (NHGRI-EBI GWAS catalog)</li>
<li>OPENTARGETS_DISEASE_ASSOCS - Associations between protein targets
and disease based on multiple lines of evidence (mutations,affected
pathways,GWAS, literature etc). Format:
CUI:EFO_ID:IS_DIRECT:OVERALL_SCORE</li>
<li>OPENTARGETS_TRACTABILITY_COMPOUND - Confidence for the existence of
a modulator (small molecule) that interacts with the target to elicit a
desired biological effect</li>
<li>OPENTARGETS_TRACTABILITY_ANTIBODY - Confidence for the existence of
a modulator (antibody) that interacts with the target to elicit a
desired biological effect</li>
</ul>
</div>
</div>
<div class="section level3">
<h3 id="tab-separated-values-tsv">Tab-separated values (TSV)<a class="anchor" aria-label="anchor" href="#tab-separated-values-tsv"></a>
</h3>
<p>We provide a tab-separated values file with most important
variant/gene annotations. The file has the following naming
convention:</p>
<p>&lt;<em>sample_id</em>&gt;.<strong>cpsr</strong>.&lt;<em>genome_assembly</em>&gt;.<strong>snvs_indels.tiers.tsv</strong></p>
<p>The SNVs/InDels are organized into different <strong>tiers</strong>
(as defined above for the HTML report)</p>
<p>The following variables are included in the tiered TSV file (VCF tags
issued by the user will be appended at the end):</p>
<pre><code>1. GENOMIC_CHANGE - Identifier for variant at the genome (VCF) level, e.g. 1:g.152382569A&gt;G
      Format: (&lt;chrom&gt;:g.&lt;position&gt;&lt;ref_allele&gt;&gt;&lt;alt_allele&gt;)
2. VAR_ID - Variant identifier
3. GENOTYPE - Variant genotype (heterozygous/homozygous)
4. CPSR_CLASSIFICATION_SOURCE - ClinVar or Other (i.e. not present in ClinVar)
5. GENOME_VERSION - Assembly version, e.g. GRCh37
6. VCF_SAMPLE_ID - Sample identifier
7. VARIANT_CLASS - Variant type, e.g. SNV/insertion/deletion
8. CODING_STATUS - coding/noncoding (wrt. protein alteration and canonical splice site disruption)
9. SYMBOL - Gene symbol
10. GENE_NAME - Gene description
11. CCDS - CCDS identifier
12. ENTREZ_ID - Entrez gene identifier
13. UNIPROT_ID - UniProt protein identifier
14. ENSEMBL_GENE_ID - Ensembl gene identifier
15. ENSEMBL_TRANSCRIPT_ID - Ensembl transcript identifier
16. REFSEQ_MRNA - RefSeq mRNA identifier
17. ONCOGENE - Gene is predicted as an oncogene according to Network of Cancer Genes (NCG) and CancerMine
18. TUMOR_SUPPRESSOR - Gene is predicted as a tumor suppressor gene according to Network of Cancer Genes (NCG) and CancerMine
19. CONSEQUENCE - Variant consequence
20. VEP_ALL_CSQ - All VEP transcript block consequences
21. PROTEIN_CHANGE - Protein change - one letter abbreviation (HGVSp)
22. PROTEIN_DOMAIN - Protein domain (Pfam)
23. DBSNP - dbSNP identifier (rsid)
24. HGVSp - The HGVS protein sequence name
25. HGVSc - The HGVS coding sequence name
26. LAST_EXON - Last exon in gene
27. EXON_POSITION - Relative position of exon variant to nearest intron/exon junction (NearestExonJB plugin)
28. INTRON_POSITION - Relative position of intron variant to nearest intron/exon junction (NearestExonJB plugin)
29. CDS_CHANGE - Coding, transcript-specific sequence annotation
30. MUTATION_HOTSPOT - Cancer mutation hotspot (cancerhotspots.org)
31. RMSK_HIT - RepeatMasker hit
32. PROTEIN_FEATURE - Protein feature (active sites etc.) from UniProt KnowledgeBase
33. EFFECT_PREDICTIONS - Functional effect predictions from multiple algorithms (dbNSFP)
34. LOSS_OF_FUNCTION - Loss-of-function variant, as predicted from VEP's LofTee plugin
35. CANCER_PHENOTYPE - For variants with a ClinVar classification, indication of cancer-associated disease/phenotype (1) or not (0)
36. CLINVAR_CLASSIFICATION - clinical significance of ClinVar Variant (CPSR category)
37. CLINVAR_MSID - measureset identifier of ClinVar variant
38. CLINVAR_VARIANT_ORIGIN - variant origin (somatic/germline) of ClinVar variant
39. CLINVAR_CONFLICTED - indicator of conflicting interpretations
40. CLINVAR_PHENOTYPE - associated phenotype(s) for ClinVar variant
41. CLINVAR_REVIEW_STATUS_STARS
42. DBMTS - variant with potential effect on microRNA target sites (dbMTS). Format: &lt;ensembl_transcript_id&gt;|&lt;microrna_identifier&gt;|&lt;target_prediction_algorithms&gt;|&lt;gain_loss_consensus&gt;. _Target prediction algorithms_ indicate support by different algorithms (separated by '&amp;'), 'TS' = TargetScan, M = 'miRanda', 'R' = RNAhybrid. *Gain_loss_consensus* indicate whether the variant was predicted to disrupt a binding site ('L' = Loss), or create a new target site ('G' = gain) by the different algorithms
43. miRNA_TARGET_HIT - loss, gain, or gain|loss, as given from the hits in DBMTS column
44. miRNA_TARGET_HIT_PREDICTION - links to miRBase, as given from the hits in the DBMTS column
45. TF_BINDING_SITE_VARIANT - Indicates whether a variant overlaps a critical/non-critical position of a transcription factor binding site (TFBS) - as provided by VEP's--regulatory option ('Overlap: non-critical motif position' or 'Overlap: critical motif position')
46. TF_BINDING_SITE_VARIANT_INFO - Comma-separated list of transcription factor binding sites affected by variant. Format per factor: &lt;TRANSCRIPTION_FACTOR&gt;|&lt;MOTIF_NAME&gt;|&lt;MOTIF_POS&gt;|&lt;MOTIF_SCORE_CHANGE&gt;|&lt;HIGH_INF_POS&gt;. *HIGH_INF_POS* indicates whether the variant overlapped a critical motif position ('Y'), or non-critical motif position ('N')
47. GERP_SCORE - Genomic conservation score (GERP)
48. N_INSILICO_CALLED - Number of algorithms with effect prediction (damaging/tolerated) from dbNSFP
49. N_INSILICO_DAMAGING - Number of algorithms with damaging prediction from dbNSFP
50. N_INSILICO_TOLERATED - Number of algorithms with tolerated prediction from dbNSFP
51. N_INSILICO_SPLICING_NEUTRAL - Number of algorithms with splicing neutral prediction from dbscSNV
52. N_INSILICO_SPLICING_AFFECTED - Number of algorithms with splicing affected prediction from dbscSNV
53. GLOBAL_AF_GNOMAD - Global MAF in gnomAD
54. &lt;CUSTOM_POPULATION_GNOMAD&gt; - Population specific MAF in gnomAD control (non-cancer, population configured by user)
55. ACMG_BA1_AD - Very high MAF (&gt; 0.5% in gnomAD non-cancer pop subset) - min AN = 12,000 - Dominant mechanism of disease
56. ACMG_BS1_1_AD - High MAF (&gt; 0.1% in gnomAD non-cancer pop subset) - min AN = 12,000 - Dominant mechanism of disease
57. ACMG_BS1_2_AD - Somewhat high MAF (&gt; 0.005% in gnomAD non-cancer pop subset) - Dominant mechanism of disease
58. ACMG_BA1_AR - Very high MAF (&gt; 1% in gnomAD non-cancer pop subset) - min AN = 12,000 - Recessive mechanism of disease
59. ACMG_BS1_1_AR - High MAF (&gt; 0.3% in gnomAD non-cancer pop subset) - min AN = 12,000 - Recessive mechanism of disease
60. ACMG_BS1_2_AR - Somewhat high MAF (&gt; 0.005% in gnomAD non-cancer pop subset) - Recessive mechanism of disease
61. ACMG_PM2_1 - Allele count within pathogenic range (MAF &lt;= 0.005% in the population-specific non-cancer gnomAD subset)
62. ACMG_PM2_2 - Alternate allele absent in the population-specific non-cancer gnomAD subset
63. ACMG_PVS1_1 - Null variant (frameshift/nonsense) - predicted as LoF by LOFTEE - within pathogenic range - LoF established for gene
64. ACMG_PVS1_2 - Null variant (frameshift/nonsense) - not predicted as LoF by LOFTEE - within pathogenic range - LoF established for gene
65. ACMG_PVS1_3 - Null variant (frameshift/nonsense) - predicted as LoF by LOFTEE - within pathogenic range - LoF not established for gene
66. ACMG_PVS1_4 - Null variant (frameshift/nonsense) - not predicted as LoF by LOFTEE -- within pathogenic range - LoF not established for gene
67. ACMG_PVS1_5 - Start (initiator methionine) lost - within pathogenic range - Lof established for gene
68. ACMG_PVS1_6 - Start (initiator methionine) lost - within pathogenic range - LoF not established for gene
69. ACMG_PVS1_7 - Donor/acceptor variant - predicted as LoF by LOFTEE - within pathogenic range - not last intron - LoF established for gene
70. ACMG_PVS1_8 - Donor/acceptor variant - last intron - within pathogenic range - LoF established for gene
71. ACMG_PVS1_9 - Donor/acceptor variant - not last intron - within pathogenic range - LoF not established for gene
72. ACMG_PVS1_10 - Donor variant at located at the +3, +4 or +5 position of the intron -  within the pathogenic range (i.e. &lt;9 alleles in ExAC))
73. ACMG_PS1 - Same amino acid change as a previously established pathogenic variant (ClinVar) regardless of nucleotide change
74. ACMG_PP2 - Missense variant in a gene that has a relatively low rate of benign missense variation (&lt;20%) and where missense variants are a common mechanism of disease (&gt;50% P/LP (ClinVar))
75. ACMG_PM1 - Missense variant in a somatic mutation hotspot as determined by cancerhotspots.org
76. ACMG_PM4 - Protein length changes due to inframe indels or nonstop variant in non-repetitive regions of genes that harbor variants with a dominant mode of inheritance.
77. ACMG_PPC1 - Protein length changes due to inframe indels or nonstop variant in non-repetitive regions of genes that harbor variants with a recessive mode of inheritance.
78. ACMG_PM5 - Novel missense change at an amino acid residue where a different missense change determined to be pathogenic has been seen before (ClinVar)
79. ACMG_PP3 - Multiple lines (&gt;=5) of computational evidence support a deleterious effect on the gene or gene product (conservation, evolutionary, splicing impact) with maximum two contradictory predictions - from dbNSFP
80. ACMG_BP4 - Multiple lines (&gt;=5) of computational evidence support a benign effect on the gene or gene product (conservation, evolutionary, splicing impact) with maximum two contradictory prediction - from dbNSFP
81. ACMG_BMC1 - Peptide change is at the same location of a known benign change (ClinVar)
82. ACMG_BSC1 - Peptide change is reported as benign (ClinVar)
83. ACMG_BP1 - Missense variant in a gene for which primarily truncating variants are known to cause disease (ClinVar)
84. ACMG_BP3 - Variants in promoter or untranslated regions
85. ACMG_BP7 - Silent/intronic variant outside of the splice site consensus
86. FINAL_CLASSIFICATION - Final variant classification based on the combination of CLINVAR_CLASSIFICTION (for ClinVar-classified variants), and CPSR_CLASSIFICATION (for novel variants)
87. CPSR_CLASSIFICATION - CPSR tier level (P/LP/VUS/LB/B)
88. CPSR_PATHOGENICITY_SCORE - Aggregated CPSR pathogenicity score
89. CPSR_CLASSIFICATION_CODE - Combination of CPSR classification codes assigned to the variant (ACMG)
90. CPSR_CLASSIFICATION_DOC - Verbal description of CPSR classification codes assignted to the variant (ACMG)</code></pre>
<p><strong>NOTE</strong>: The user has the possibility to append the TSV
file with data from other INFO tags in the input VCF (i.e.Â using the
<em>â€“preserved_info_tags</em> option)</p>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by <a href="https://github.com/sigven" class="external-link">Sigve Nakken</a>,
<a href="https://github.com/pdiakumis" class="external-link">Peter Diakumis</a>.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a>
2.0.2.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
