<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="cpsr">
<title>Running • CPSR</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.11/clipboard.min.js" integrity="sha512-7O5pXpc0oCRrxk8RUfDYFgn0nO1t+jLuIOQdOMRp4APB7uZ4vSjspzp5y6YDtDs4VzUSTbWzBFZ/LKJhnyFOKw==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Running">
<meta property="og:description" content="cpsr">
<meta property="og:image" content="https://sigven.github.io/cpsr/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-info" data-bs-theme="light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">CPSR</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">2.0.0</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="../articles/installation.html">Installation</a>
</li>
<li class="active nav-item">
  <a class="nav-link" href="../articles/running.html">Running</a>
</li>
<li class="nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/input.html">Input files</a>
    <a class="dropdown-item" href="../articles/virtual_panels.html">Virtual gene panels</a>
    <a class="dropdown-item" href="../articles/variant_classification.html">Variant classification</a>
    <a class="dropdown-item" href="../articles/output.html">Output files</a>
    <a class="dropdown-item" href="../articles/annotation_resources.html">Annotation resources</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="../articles/CHANGELOG.html">CHANGELOG</a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/sigven/cpsr/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>Running</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/sigven/cpsr/blob/HEAD/vignettes/running.Rmd" class="external-link"><code>vignettes/running.Rmd</code></a></small>
      <div class="d-none name"><code>running.Rmd</code></div>
    </div>

    
    
<p><br></p>
<p>The CPSR software comes with a range of options to configure the
report and analysis.</p>
<div class="section level2">
<h2 id="key-settings">Key settings<a class="anchor" aria-label="anchor" href="#key-settings"></a>
</h2>
<p>Below, we will outline some key settings that are important for the
generation of variant reports with CPSR.</p>
<div class="section level3">
<h3 id="virtual-gene-panel">Virtual gene panel<a class="anchor" aria-label="anchor" href="#virtual-gene-panel"></a>
</h3>
<p>The user can choose the set of cancer predisposition genes for which
input variants should be subject to classification and reporting. The
input VCF with germline variants may thus come from any sequencing assay
(whole-genome, whole-exome or targeted assay), yet the results shown in
the report will always be restricted by the choice of a <a href="virtual_panels.html">virtual gene panel</a>:</p>
<ul>
<li><code>--panel_id &lt;panel_number&gt;</code></li>
</ul>
<div class="section level4">
<h4 id="custom-made-virtual-gene-panels">Custom-made virtual gene panels<a class="anchor" aria-label="anchor" href="#custom-made-virtual-gene-panels"></a>
</h4>
<p>CPSR allows users to create custom virtual gene panels for reporting.
Any set of genes found in the <a href="virtual_panels.html#panel-0">CPSR
superpanel (panel 0)</a> can be used to design a custom virtual gene
panel. Technically, the users need to create a simple one-column text
file with Ensembl gene identifiers, and provide a name for the custom
panel:</p>
<ul>
<li><code>--custom_list &lt;custom_list_tsv&gt;</code></li>
<li><code>--custom_list_name &lt;custom_list_name</code></li>
</ul>
<p>The <a href="https://github.com/sigven/cpsr/tree/master/examples" class="external-link">examples</a>
folder contains an example file for a custom-made virtual gene
panel.</p>
</div>
</div>
<div class="section level3">
<h3 id="acmg-variant-classification">ACMG variant classification<a class="anchor" aria-label="anchor" href="#acmg-variant-classification"></a>
</h3>
<p>The ACMG classification criteria rely heavily upon variant population
frequencies from <a href="https://gnomad.broadinstitute.org/" class="external-link">gnomAD</a>. The user can
specify the preferred population when it comes to variant allele
frequencies used by CPSR:</p>
<ul>
<li><code>--pop_gnomad &lt;pop_code&gt;</code></li>
</ul>
<hr>
<p>There is also an option to set an upper minor allele frequency limit
(gnomAD - global population) for variants to be included in the report.
This is basically a means to exclude the potentially large number of
common and benign variants that can be found in the input VCF:</p>
<ul>
<li><code>--maf_upper_threshold &lt;maf_threshold&gt;</code></li>
</ul>
<hr>
<p>By default, CPSR provides two types of variant classifications:</p>
<ol style="list-style-type: decimal">
<li>Existing variant classifications - as provided by ClinVar</li>
<li>CPSR-derived classifications of novel variants (not found in
ClinVar)</li>
</ol>
<p>The user may opt to show CPSR’s classification of the variants which
are already attributed with classifications in ClinVar (i.e. set
<em>1.</em>):</p>
<ul>
<li><code>--classify_all</code></li>
</ul>
<p>In the final classification, note that CPSR’s classification will not
overrule the classification reported by ClinVar, which takes precedence
for variants with a record here.</p>
<hr>
<p>By default, CPSR do not report variants in the input sample that are
found in cancer predisposition genes, yet <em>not</em> associated with
diseases or syndromes that are cancer relevant. The user may choose to
show all reported ClinVar variants in cancer predisposition genes
through a dedicated option:</p>
<ul>
<li><code>--clinvar_report_noncancer</code></li>
</ul>
</div>
<div class="section level3">
<h3 id="optional-report-contents">Optional report contents<a class="anchor" aria-label="anchor" href="#optional-report-contents"></a>
</h3>
<p>CPSR allows users to report recommended <a href="https://www.ncbi.nlm.nih.gov/clinvar/docs/acmg/" class="external-link">incidental
findings</a>, and also the genotypes of reported cancer risk loci from
<a href="https://www.ebi.ac.uk/gwas/" class="external-link">genome-wide association studies
(GWAS)</a>:</p>
<ul>
<li><code>--secondary_findings</code></li>
<li><code>--gwas_findings</code></li>
</ul>
</div>
</div>
<div class="section level2">
<h2 id="all-options">All options<a class="anchor" aria-label="anchor" href="#all-options"></a>
</h2>
<p>A cancer predisposition report is generated by running the
<strong>cpsr</strong> command, which takes the following arguments and
options:</p>
<pre class="text"><code>usage: cpsr -h [options] 
    --input_vcf &lt;INPUT_VCF&gt;
    --vep_dir &lt;VEP_DIR&gt;
    --refdata_dir &lt;REFDATA_DIR&gt;
    --output_dir &lt;OUTPUT_DIR&gt;
    --genome_assembly &lt;GENOME_ASSEMBLY&gt;
    --sample_id &lt;SAMPLE_ID&gt;

Cancer Predisposition Sequencing Reporter - report of clinically significant cancer-predisposing germline variants

Required arguments:
  --input_vcf INPUT_VCF
                        VCF input file with germline query variants (SNVs/InDels).
  --vep_dir VEP_DIR     Directory of VEP cache, e.g.  $HOME/.vep
  --refdata_dir REFDATA_DIR
                        Directory that contains the PCGR/CPSR reference data, e.g. ~/pcgr-data-1.4.1.9019
  --output_dir OUTPUT_DIR
                        Output directory
  --genome_assembly {grch37,grch38}
                        Genome assembly build: grch37 or grch38
  --sample_id SAMPLE_ID
                        Sample identifier - prefix for output files

Panel options:
  --panel_id VIRTUAL_PANEL_ID
                        Comma-separated string with identifier(s) of predefined virtual cancer predisposition gene panels,
                        choose any combination of the following identifiers (GEP = Genomics England PanelApp):
                        0 = CPSR exploratory cancer predisposition panel (PanelApp genes / TCGA's germline study / Cancer Gene Census / Other)
                        1 = Adult solid tumours cancer susceptibility (GEP)
                        2 = Adult solid tumours for rare disease (GEP)
                        3 = Bladder cancer pertinent cancer susceptibility (GEP)
                        4 = Brain cancer pertinent cancer susceptibility (GEP)
                        5 = Breast cancer pertinent cancer susceptibility (GEP)
                        6 = Childhood solid tumours cancer susceptibility (GEP)
                        7 = Colorectal cancer pertinent cancer susceptibility (GEP)
                        8 = Endometrial cancer pertinent cancer susceptibility (GEP)
                        9 = Familial Tumours Syndromes of the central &amp; peripheral Nervous system (GEP)
                        10 = Familial breast cancer (GEP)
                        11 = Familial melanoma (GEP)
                        12 = Familial prostate cancer (GEP)
                        13 = Familial rhabdomyosarcoma (GEP)
                        14 = GI tract tumours (GEP)
                        15 = Genodermatoses with malignancies (GEP)
                        16 = Haematological malignancies cancer susceptibility (GEP)
                        17 = Haematological malignancies for rare disease (GEP)
                        18 = Head and neck cancer pertinent cancer susceptibility (GEP)
                        19 = Inherited MMR deficiency (Lynch Syndrome) - GEP
                        20 = Inherited non-medullary thyroid cancer (GEP)
                        21 = Inherited ovarian cancer (without breast cancer) (GEP)
                        22 = Inherited pancreatic cancer (GEP)
                        23 = Inherited polyposis and early onset colorectal cancer (GEP)
                        24 = Inherited predisposition to acute myeloid leukaemia (AML) (GEP)
                        25 = Inherited susceptibility to acute lymphoblastoid leukaemia (ALL) (GEP)
                        26 = Inherited predisposition to GIST (GEP)
                        27 = Inherited renal cancer (GEP)
                        28 = Inherited phaeochromocytoma and paraganglioma (GEP)
                        29 = Melanoma pertinent cancer susceptibility (GEP)
                        30 = Multiple endocrine tumours (GEP)
                        31 = Multiple monogenic benign skin tumours (GEP)
                        32 = Neuroendocrine cancer pertinent cancer susceptibility (GEP)
                        33 = Neurofibromatosis Type 1 (GEP)
                        34 = Ovarian cancer pertinent cancer susceptibility (GEP)
                        35 = Parathyroid Cancer (GEP)
                        36 = Prostate cancer pertinent cancer susceptibility (GEP)
                        37 = Renal cancer pertinent cancer susceptibility (GEP)
                        38 = Rhabdoid tumour predisposition (GEP)
                        39 = Sarcoma cancer susceptibility (GEP)
                        40 = Sarcoma susceptibility (GEP)
                        41 = Thyroid cancer pertinent cancer susceptibility (GEP)
                        42 = Tumour predisposition - childhood onset (GEP)
                        43 = Upper gastrointestinal cancer pertinent cancer susceptibility (GEP)
                        44 = DNA repair genes pertinent cancer susceptibility (GEP)
  --custom_list CUSTOM_LIST
                        Provide custom list of genes from virtual panel 0 (single-column .txt/.tsv file with Ensembl gene identifiers),
                         alternative to predefined panels provided with --panel_id)
  --custom_list_name CUSTOM_LIST_NAME
                        Set name for custom made panel/list (single word - no whitespace), will be displayed in the report
  --diagnostic_grade_only
                        For panel_id's 1-44 (Genomics England PanelApp) - consider genes with a GREEN status only, default: False

Variant classification options:
  --secondary_findings  Include variants found in ACMG-recommended list for secondary findings (v3.2), default: False
  --gwas_findings       Report overlap with low to moderate cancer risk variants (tag SNPs) identified from genome-wide association studies, default: False
  --pop_gnomad {afr,amr,eas,sas,asj,nfe,fin,global}
                        Population source in gnomAD (non-cancer subset) used for variant frequency assessment (ACMG classification), default: nfe
  --maf_upper_threshold MAF_UPPER_THRESHOLD
                        Upper MAF limit (gnomAD global population frequency) for variants to be included in the report, default: 0.9
  --classify_all        Provide CPSR variant classifications (TIER 1-5) also for variants with existing ClinVar classifications in output TSV, default: False
  --clinvar_report_noncancer
                        Report also ClinVar-classified variants attributed to phenotypes/conditions NOT directly related to tumor development, default: False

VEP options:
  --vep_n_forks VEP_N_FORKS
                        Number of forks (option '--fork' in VEP), default: 4
  --vep_buffer_size VEP_BUFFER_SIZE
                        Variant buffer size (variants read into memory simultaneously, option '--buffer_size' in VEP) 
                        - set lower to reduce memory usage, default: 500
  --vep_gencode_basic   Consider basic GENCODE transcript set only with Variant Effect Predictor (VEP) (option '--gencode_basic' in VEP).
  --vep_pick_order VEP_PICK_ORDER
                        Comma-separated string of ordered transcript properties for primary variant pick
                         ( option '--pick_order' in VEP), default: mane_select,mane_plus_clinical,canonical,appris,tsl,biotype,ccds,rank,length
  --vep_no_intergenic   Skip intergenic variants during processing (option '--no_intergenic' in VEP), default: False

vcfanno options:
  --vcfanno_n_proc VCFANNO_N_PROC
                        Number of vcfanno processes (option '-p' in vcfanno), default: 4

Other options:
  --force_overwrite     By default, the script will fail with an error if any output file already exists.
                         You can force the overwrite of existing result files by using this flag, default: False
  --version             show program's version number and exit
  --no_reporting        Run functional variant annotation on VCF through VEP/vcfanno, omit classification/report generation (STEP 4), default: False
  --retained_info_tags RETAINED_INFO_TAGS
                        Comma-separated string of VCF INFO tags from query VCF that should be kept in CPSR output TSV
  --ignore_noncoding    Ignore non-coding (i.e. non protein-altering) variants in report, default: False
  --debug               Print full commands to log
  --pcgrr_conda PCGRR_CONDA
                        pcgrr conda env name (default: pcgrr)</code></pre>
</div>
<div class="section level2">
<h2 id="example-run">Example run<a class="anchor" aria-label="anchor" href="#example-run"></a>
</h2>
<p>The <em>cpsr</em> R package comes with a test VCF file (calls from
the GRCh37 human genome assembly) that can be used to test the CPSR
workflow. Please note that these are artificial germline calls, not
originating from an actual patient.</p>
<p>Report generation with the example VCF, using the <a href="https://panelapp.genomicsengland.co.uk/panels/245/" class="external-link">Adult solid
tumours cancer susceptibility</a> as the virtual gene panel, can be
performed through the following command:</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb2-1"><a href="#cb2-1" tabindex="-1"></a><span class="ex">$</span> <span class="er">(</span><span class="ex">base</span><span class="kw">)</span> <span class="ex">conda</span> activate pcgr</span>
<span id="cb2-2"><a href="#cb2-2" tabindex="-1"></a><span class="ex">$</span> <span class="er">(</span><span class="ex">pcgr</span><span class="kw">)</span></span>
<span id="cb2-3"><a href="#cb2-3" tabindex="-1"></a><span class="ex">cpsr</span> <span class="dt">\</span></span>
<span id="cb2-4"><a href="#cb2-4" tabindex="-1"></a>     <span class="at">--input_vcf</span> ~/cpsr-2.0.0/inst/examples/example.vcf.gz <span class="dt">\</span></span>
<span id="cb2-5"><a href="#cb2-5" tabindex="-1"></a>     <span class="at">--vep_dir</span> ~/.vep <span class="dt">\</span></span>
<span id="cb2-6"><a href="#cb2-6" tabindex="-1"></a>     <span class="at">--refdata_dir</span> ~/pcgr_ref_data <span class="dt">\</span></span>
<span id="cb2-7"><a href="#cb2-7" tabindex="-1"></a>     <span class="at">--output_dir</span> ~/cpsr-2.0.0/ <span class="dt">\</span></span>
<span id="cb2-8"><a href="#cb2-8" tabindex="-1"></a>     <span class="at">--genome_assembly</span> grch37 <span class="dt">\</span></span>
<span id="cb2-9"><a href="#cb2-9" tabindex="-1"></a>     <span class="at">--panel_id</span> 1 <span class="dt">\</span></span>
<span id="cb2-10"><a href="#cb2-10" tabindex="-1"></a>     <span class="at">--sample_id</span> example <span class="dt">\</span></span>
<span id="cb2-11"><a href="#cb2-11" tabindex="-1"></a>     <span class="at">--secondary_findings</span> <span class="dt">\</span></span>
<span id="cb2-12"><a href="#cb2-12" tabindex="-1"></a>     <span class="at">--classify_all</span> <span class="dt">\</span></span>
<span id="cb2-13"><a href="#cb2-13" tabindex="-1"></a>     <span class="at">--maf_upper_threshold</span> 0.2 <span class="dt">\</span></span>
<span id="cb2-14"><a href="#cb2-14" tabindex="-1"></a>     <span class="at">--force_overwrite</span></span></code></pre></div>
<p>Note that the example command refers to the PCGR data bundle
directory (<em>refdata_dir</em>), which contains the data necessary for
both <em>PCGR</em> and <em>CPSR</em>.</p>
<p>This command will produce the following output files in the
<em>output</em> folder:</p>
<ol style="list-style-type: decimal">
<li>
<strong>example.cpsr.grch37.vcf.gz (.tbi)</strong> - Bgzipped VCF
file with various variant annotations appended by CPSR</li>
<li>
<strong>example.cpsr.grch37.pass.vcf.gz (.tbi)</strong> - Bgzipped
VCF file with various variant annotations appended by CPSR (PASS
variants only)</li>
<li>
<strong>example.cpsr.grch37.conf.yaml</strong> - CPSR configuration
file - output from pre-reporting annotation (Python) workflow</li>
<li>
<strong>example.cpsr.grch37.pass.tsv.gz</strong> - Compressed TSV
file (generated with <a href="https://github.com/sigven/vcf2tsvpy" class="external-link">vcf2tsvpy</a>) of VCF content
with various annotations appended by CPSR</li>
<li>
<strong>example.cpsr.grch37.xlsx</strong> - An Excel workbook that
contains
<ul>
<li>
<em>i)</em> information on virtual gene panel interrogated for
variants</li>
<li>
<em>ii)</em> classification of clinical significance for variants
overlapping with cancer predisposition genes</li>
<li>
<em>iii)</em> match of variants with existing biomarkers (if any
found)</li>
<li>
<em>iv)</em> secondary findings (if any found)</li>
</ul>
</li>
<li>
<strong>example.cpsr.grch37.html</strong> - Interactive HTML report
with clinically relevant variants in cancer predisposition genes</li>
<li>
<strong>example.cpsr.grch37.snvs_indels.classification.tsv.gz</strong>
- TSV file with key annotations of germline SNVs/InDels classified
according to clinical significance</li>
</ol>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by <a href="https://github.com/sigven" class="external-link">Sigve Nakken</a>, <a href="https://github.com/pdiakumis" class="external-link">Peter Diakumis</a>.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.9.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
