<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Variant classification • CPSR</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Variant classification">
<script async src="https://www.googletagmanager.com/gtag/js?id=G-FYKTRC4655"></script><script>
   window.dataLayer = window.dataLayer || [];
   function gtag(){dataLayer.push(arguments);}
   gtag('js', new Date());

   gtag('config', 'G-FYKTRC4655');
</script>
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-info" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">CPSR</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">2.2.1</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="../articles/installation.html">Installation</a></li>
<li class="nav-item"><a class="nav-link" href="../articles/running.html">Running</a></li>
<li class="active nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><a class="dropdown-item" href="../articles/input.html">Input files</a></li>
    <li><a class="dropdown-item" href="../articles/virtual_panels.html">Virtual gene panels</a></li>
    <li><a class="dropdown-item" href="../articles/variant_classification.html">Variant classification</a></li>
    <li><a class="dropdown-item" href="../articles/output.html">Output files</a></li>
    <li><a class="dropdown-item" href="../articles/annotation_resources.html">Annotation resources</a></li>
  </ul>
</li>
<li class="nav-item"><a class="nav-link" href="../articles/CHANGELOG.html">CHANGELOG</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/sigven/cpsr/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>Variant classification</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/sigven/cpsr/blob/main/vignettes/variant_classification.Rmd" class="external-link"><code>vignettes/variant_classification.Rmd</code></a></small>
      <div class="d-none name"><code>variant_classification.Rmd</code></div>
    </div>

    
    
<p>All coding, <strong>non-ClinVar</strong> variants in the set of genes
subject to screening have been classified according to a <em>standard,
five-level pathogenicity scheme</em> (coined
<strong>CPSR_CLASSIFICATION</strong>). The scheme has the same five
levels as those employed by ClinVar, i.e.</p>
<ul>
<li>pathogenic (<strong>P</strong>)</li>
<li>likely pathogenic (<strong>LP</strong>)</li>
<li>variant of uncertain significance (<strong>VUS</strong>)</li>
<li>likely benign (<strong>LB</strong>)</li>
<li>benign (<strong>B</strong>)</li>
</ul>
<p>The classification performed by CPSR is rule-based, implementing most
of the ACMG criteria related to <em>variant effect</em> and
<em>population frequency</em>, which have been outlined in <a href="https://www.invitae.com/en/variant-classification/" class="external-link">SherLoc</a>
(<a href="https://www.ncbi.nlm.nih.gov/pubmed/28492532" class="external-link">Nykamp et al.,
Genetics in Medicine, 2017</a>), and also some in <a href="https://github.com/ding-lab/CharGer" class="external-link">CharGer</a>. Information on
cancer predisposition genes (mode of inheritance, loss-of-funcion
mechanism etc.) is largely harvested from <a href="https://www.ncbi.nlm.nih.gov/pubmed/27153395" class="external-link">Maxwell et al., Am J
Hum Genet, 2016</a>.</p>
<p>The refined ACMG/AMP criteria listed below form the basis for the
tier assigned to the <strong>CPSR_CLASSIFICATION</strong> variable.
Specifically, the <strong>score</strong> in parenthesis indicates how
much each evidence item contributes to either of the two pathogenicity
poles (positive values indicate pathogenic support, negative values
indicate benign support). Evidence score along each pole (‘B’ and ‘P’)
are aggregated, and if there is conflicting or little evidence it will
be classified as a VUS. This classification scheme has been adopted by
the one outlined in <a href="https://www.ncbi.nlm.nih.gov/pubmed/28492532" class="external-link">SherLoc</a>.</p>
<table class="table">
<colgroup>
<col width="35%">
<col width="65%">
</colgroup>
<thead><tr class="header">
<th>Tag</th>
<th>Description</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>1. <code>ACMG_BA1_AD</code> (<strong>-5</strong>)</td>
<td>Very high MAF (&gt; 0.5% in gnomAD non-cancer pop subset, min AN =
12,000) - dominant mode of inheritance</td>
</tr>
<tr class="even">
<td>2. <code>ACMG_BS1_1_AD</code> (<strong>-3</strong>)</td>
<td>High MAF (&gt; 0.1% in gnomAD non-cancer pop subset, min AN =
12,000) - dominant mode of inheritance</td>
</tr>
<tr class="odd">
<td>3. <code>ACMG_BS1_2_AD</code> (<strong>-1</strong>)</td>
<td>Somewhat high MAF (&gt; 0.005% in gnomAD non-cancer pop subset, min
AN = 12,000) - dominant mode of inheritance</td>
</tr>
<tr class="even">
<td>4. <code>ACMG_BA1_AR</code> (<strong>-5</strong>)</td>
<td>Very high MAF (&gt; 1% in gnomAD non-cancer pop subset, min AN =
12,000) - recessive mode of inheritance</td>
</tr>
<tr class="odd">
<td>5. <code>ACMG_BS1_1_AR</code> (<strong>-3</strong>)</td>
<td>High MAF (&gt; 0.3% in gnomAD non-cancer pop subset, min AN =
12,000) - recessive mode of inheritance</td>
</tr>
<tr class="even">
<td>6. <code>ACMG_BS1_2_AR</code> (<strong>-1</strong>)</td>
<td>Somewhat high MAF (&gt; 0.005% in gnomAD non-cancer pop subset, min
AN = 12,000) - recessive mode of inheritance</td>
</tr>
<tr class="odd">
<td>7. <code>ACMG_PM2_1</code> (<strong>0.5</strong>)</td>
<td>Allele count within pathogenic range (MAF &lt;= 0.005% in the
population-specific non-cancer gnomAD subset, min AN = 12,000)</td>
</tr>
<tr class="even">
<td>8. <code>ACMG_PM2_2</code> (<strong>1</strong>)</td>
<td>Alternate allele absent in the population-specific non-cancer gnomAD
subset</td>
</tr>
<tr class="odd">
<td>9. <code>ACMG_PVS1_1</code> (<strong>5</strong>)</td>
<td>Null variant - predicted as loss-of-function - within pathogenic
range (ACMG_PM2_1/2) - LoF established for gene</td>
</tr>
<tr class="even">
<td>10. <code>ACMG_PVS1_2</code> (<strong>2.5</strong>)</td>
<td>Null variant - not predicted as loss-of-function - within pathogenic
range (ACMG_PM2_1/2) - LoF established mechanism for gene</td>
</tr>
<tr class="odd">
<td>11. <code>ACMG_PVS1_3</code> (<strong>2</strong>)</td>
<td>Null variant - predicted as loss-of-function - within pathogenic
range (ACMG_PM2_1/2) - LoF not established mechanism for gene</td>
</tr>
<tr class="even">
<td>12. <code>ACMG_PVS1_4</code> (<strong>0</strong>)</td>
<td>Null variant - not predicted as loss-of-function - within pathogenic
range (ACMG_PM2_1/2) - LoF not established mechanism for gene</td>
</tr>
<tr class="odd">
<td>13. <code>ACMG_PVS1_5</code> (<strong>2.5</strong>)</td>
<td>Start (initiator methionine) lost - within pathogenic range
(ACMG_PM2_1/2) - Lof established mechanism for gene</td>
</tr>
<tr class="even">
<td>14. <code>ACMG_PVS1_6</code> (<strong>2</strong>)</td>
<td>Start (initiator methionine) lost - within pathogenic range
(ACMG_PM2_1/2) - LoF not established mechanism for gene</td>
</tr>
<tr class="odd">
<td>15. <code>ACMG_PVS1_7</code> (<strong>4</strong>)</td>
<td>Donor/acceptor variant - predicted as loss-of-function - within
pathogenic range (ACMG_PM2_1/2) - not last intron - LoF established for
gene</td>
</tr>
<tr class="even">
<td>16. <code>ACMG_PVS1_8</code> (<strong>2.5</strong>)</td>
<td>Donor/acceptor variant - last intron - within pathogenic range
(ACMG_PM2_1/2) - LoF established for gene</td>
</tr>
<tr class="odd">
<td>17. <code>ACMG_PVS1_9</code> (<strong>2</strong>)</td>
<td>Donor/acceptor variant - not last intron - within pathogenic range
(ACMG_PM2_1/2) - LoF not established mechanism for gene</td>
</tr>
<tr class="even">
<td>18. <code>ACMG_PVS1_10</code> (<strong>2</strong>)</td>
<td>Donor variant at located at the +3, +4 or +5 position of the intron
- within pathogenic range (ACMG_PM2_1/2) - ACMG_PP3 support</td>
</tr>
<tr class="odd">
<td>19. <code>ACMG_PS1</code> (<strong>4</strong>)</td>
<td>Same amino acid change as a previously established pathogenic
variant (ClinVar - minimum two review stars) regardless of nucleotide
change</td>
</tr>
<tr class="even">
<td>20. <code>ACMG_PP2</code> (<strong>0.5</strong>)</td>
<td>Missense variant in a gene that has a relatively low rate of benign
missense variation (&lt;10%) and where missense variants are a common
mechanism of disease (&gt;50% of high-confidence pathogenic variants
(ClinVar))</td>
</tr>
<tr class="odd">
<td>21. <code>ACMG_PM1</code> (<strong>2</strong>)</td>
<td>Missense variant at critical AA residue (as identified in
cancerhotspots.org)</td>
</tr>
<tr class="even">
<td>22. <code>ACMG_PM4</code> (<strong>0</strong>)</td>
<td>Protein length changes due to inframe indels or nonstop variant in
non-repetitive regions (as identified by RepeatMasker) of genes that
harbor variants with a dominant mode of inheritance.</td>
</tr>
<tr class="odd">
<td>23. <code>ACMG_PPC1</code> (<strong>0</strong>)</td>
<td>Protein length changes due to inframe indels or nonstop variant in
non-repetitive regions (as identified by RepeatMasker) of genes that
harbor variants with a recessive mode of inheritance.</td>
</tr>
<tr class="even">
<td>24. <code>ACMG_PM5</code> (<strong>1.5</strong>)</td>
<td>Novel missense change at an amino acid residue where a different
missense change determined to be pathogenic has been seen before
(ClinVar - minimum two review stars)</td>
</tr>
<tr class="odd">
<td>25. <code>ACMG_PM6</code> (<strong>1.5</strong>)</td>
<td>Disruption of essential nucleotide</td>
</tr>
<tr class="even">
<td>26. <code>ACMG_PP3</code> (<strong>0.5</strong>)</td>
<td>Multiple lines of computational evidence support a deleterious
effect on the gene or gene product (conservation, evolutionary, splicing
impact, etc. - from dbNSFP)</td>
</tr>
<tr class="odd">
<td>27. <code>ACMG_BP4</code> (<strong>-1</strong>)</td>
<td>Multiple lines of computational evidence support a benign effect on
the gene or gene product (conservation, evolutionary, splicing impact,
etc. - from dbNSFP)</td>
</tr>
<tr class="even">
<td>28. <code>ACMG_BMC1</code> (<strong>0</strong>)</td>
<td>Peptide change is at the same location of a known benign change
(ClinVar - minimum two review stars)</td>
</tr>
<tr class="odd">
<td>29. <code>ACMG_BSC1</code> (<strong>-3</strong>)</td>
<td>Peptide change is reported as benign (ClinVar - minimum two review
stars)</td>
</tr>
<tr class="even">
<td>30. <code>ACMG_BP1</code> (<strong>-0.5</strong>)</td>
<td>Missense variant in a gene for which primarily truncating variants
(&gt;= 90% of pathogenic variants) are known to cause disease
(ClinVar)</td>
</tr>
<tr class="odd">
<td>31. <code>ACMG_BP3</code> (<strong>-2.5</strong>)</td>
<td>Non-coding variant in the UTR or promoter region</td>
</tr>
<tr class="even">
<td>32. <code>ACMG_BP7</code> (<strong>-2.5</strong>)</td>
<td>Silent and intronic changes outside of the consensus splice
site</td>
</tr>
</tbody>
</table>
<p><br> Currently (as of March 2025), based on a calibration against
ClinVar-classified variants (minimum two review status stars) in n = 105
core cancer predisposition genes, the clinical significance
(<strong>CPSR_CLASSIFICATION</strong>) is determined based on the
following ranges of pathogenicity scores:</p>
<p><br></p>
<p><img src="../img/cpsr_classification.png"></p>
  </main>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by <a href="https://github.com/sigven" class="external-link">Sigve Nakken</a>, <a href="https://github.com/pdiakumis" class="external-link">Peter Diakumis</a>.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

    </footer>
</div>





  </body>
</html>
