<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="cpsr">
<title>Variant classification • CPSR</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.1.0/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.1.0/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.rawgit.com/afeld/bootstrap-toc/v1.0.1/dist/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Variant classification">
<meta property="og:description" content="cpsr">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-dark navbar-expand-lg bg-primary"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">CPSR</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.7.6</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="../index.html">Intro</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../articles/installation.html">Installation</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../articles/running.html">Running</a>
</li>
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/input.html">Input files</a>
    <a class="dropdown-item" href="../articles/virtual_panels.html">Virtual gene panels</a>
    <a class="dropdown-item" href="../articles/variant_classification.html">ACMG classification</a>
    <a class="dropdown-item" href="../articles/output.html">Output files</a>
    <a class="dropdown-item" href="../articles/annotation_resources.html">Annotation resources</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="../articles/CHANGELOG.html">CHANGELOG</a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/sigven/cpsr/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main"><div class="page-header">
      <img src="" class="logo" alt=""><h1>Variant classification</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/sigven/cpsr/blob/HEAD/vignettes/variant_classification.Rmd" class="external-link"><code>vignettes/variant_classification.Rmd</code></a></small>
      <div class="d-none name"><code>variant_classification.Rmd</code></div>
    </div>

    
    
<p>All coding, <b>non-ClinVar</b> variants in the set of genes subject
to screening have been classified according to a <i>standard, five-level
pathogenicity scheme</i> (coined <b>CPSR_CLASSIFICATION</b>). The scheme
has the same five levels as those employed by ClinVar, i.e.</p>
<ul>
<li>pathogenic (<strong>P</strong>)</li>
<li>likely pathogenic (<strong>LP</strong>)</li>
<li>variant of uncertain significance (<strong>VUS</strong>)</li>
<li>likely benign (<strong>LB</strong>)</li>
<li>benign (<strong>B</strong>)</li>
</ul>
<p>The classification performed by CPSR is rule-based, implementing most
of the ACMG criteria related to <i>variant effect</i> and <i>population
frequency</i>, which have been outlined in <a href="https://www.invitae.com/en/variant-classification/" class="external-link">SherLoc</a>
(<a href="https://www.ncbi.nlm.nih.gov/pubmed/28492532" class="external-link">Nykamp et al.,
Genetics in Medicine, 2017</a>), and also some in <a href="https://github.com/ding-lab/CharGer" class="external-link">CharGer</a>. Information on
cancer predisposition genes (mode of inheritance, loss-of-funcion
mechanism etc.) is largely harvested from <a href="https://www.ncbi.nlm.nih.gov/pubmed/27153395" class="external-link">Maxwell et al., Am J
Hum Genet, 2016</a>.</p>
<p>The refined ACMG/AMP criteria listed below form the basis for the
tier assigned to the <b>CPSR_CLASSIFICATION</b> variable. Specifically,
the <b>score</b> in parenthesis indicates how much each evidence item
contributes to either of the two pathogenicity poles (positive values
indicate pathogenic support, negative values indicate benign support).
Evidence score along each pole (‘B’ and ‘P’) are aggregated, and if
there is conflicting or little evidence it will be classified as a VUS.
This classification scheme has been adopted by the one outlined in <a href="https://www.ncbi.nlm.nih.gov/pubmed/28492532" class="external-link">SherLoc</a>.</p>
<ol style="list-style-type: decimal">
<li>ACMG_BA1_AD (<strong>-5</strong>) - Very high MAF (&gt; 0.5% in
gnomAD non-cancer pop subset) - min AN = 12,000 - Dominant mechanism of
disease</li>
<li>ACMG_BS1_1_AD (<strong>-3</strong>)- High MAF (&gt; 0.1% in gnomAD
non-cancer pop subset) - min AN = 12,000 - Dominant mechanism of
disease</li>
<li>ACMG_BS1_2_AD (<strong>-1</strong>) Somewhat high MAF (&gt; 0.005%
in gnomAD non-cancer pop subset) - Dominant mechanism of disease</li>
<li>ACMG_BA1_AR (<strong>-5</strong>) - Very high MAF (&gt; 1% in gnomAD
non-cancer pop subset) - min AN = 12,000 - Recessive mechanism of
disease</li>
<li>ACMG_BS1_1_AR (<strong>-3</strong>)- High MAF (&gt; 0.3% in gnomAD
non-cancer pop subset) - min AN = 12,000 - Recessive mechanism of
disease</li>
<li>ACMG_BS1_2_AR (<strong>-1</strong>)- Somewhat high MAF (&gt; 0.005%
in gnomAD non-cancer pop subset) - Recessive mechanism of disease</li>
<li>ACMG_BP3 (<strong>-2.5</strong>) - Non-coding variant in the UTR or
promoter region</li>
<li>ACMG_BP7 (<strong>-2.5</strong>) - Silent and intronic changes
outside of the consensus splice site</li>
<li>ACMG_BP4 (<strong>-1</strong>)- Multiple lines (&gt;=5) of
computational evidence support a benign effect on the gene or gene
product (conservation, evolutionary, splicing impact, etc. - from
dbNSFP</li>
<li>ACMG_BMC1 (<strong>0</strong>)- Peptide change is at the same
location of a known benign change (ClinVar)</li>
<li>ACMG_BSC1 (<strong>-3</strong>)- Peptide change is reported as
benign (ClinVar)</li>
<li>ACMG_BP1 (<strong>-0.5</strong>)- Missense variant in a gene for
which primarily truncating variants (&gt;90% of pathogenic variants) are
known to cause disease (ClinVar)</li>
<li>ACMG_PM2_1 (<strong>0.5</strong>)- Allele count within pathogenic
range (MAF &lt;= 0.005% in the population-specific non-cancer gnomAD
subset)</li>
<li>ACMG_PM2_2 (<strong>1</strong>)- Alternate allele absent in the
population-specific non-cancer gnomAD subset</li>
<li>ACMG_PVS1_1 (<strong>5</strong>) - Null variant
(frameshift/nonsense) - predicted as LoF by LOFTEE - within pathogenic
range - LoF established for gene</li>
<li>ACMG_PVS1_2 (<strong>2.5</strong>)- Null variant
(frameshift/nonsense) - not predicted as LoF by LOFTEE - within
pathogenic range - LoF established for gene</li>
<li>ACMG_PVS1_3 (<strong>2</strong>) - Null variant
(frameshift/nonsense) - predicted as LoF by LOFTEE - within pathogenic
range - LoF not established for gene</li>
<li>ACMG_PVS1_4 (<strong>0</strong>) - Null variant
(frameshift/nonsense) - not predicted as LoF by LOFTEE – within
pathogenic range - LoF not established for gene</li>
<li>ACMG_PVS1_5 (<strong>2.5</strong>) - Start (initiator methionine)
lost - within pathogenic range - Lof established for gene</li>
<li>ACMG_PVS1_6 (<strong>2</strong>) - Start (initiator methionine) lost
- within pathogenic range - LoF not established for gene</li>
<li>ACMG_PVS1_7 (<strong>4</strong>) - Donor/acceptor variant -
predicted as LoF by LOFTEE - within pathogenic range - not last intron -
LoF established for gene</li>
<li>ACMG_PVS1_8 (<strong>2.5</strong>) - Donor/acceptor variant - last
intron - within pathogenic range - LoF established for gene</li>
<li>ACMG_PVS1_9 (<strong>2</strong>)- Donor/acceptor variant - not last
intron - within pathogenic range - LoF not established for gene</li>
<li>ACMG_PVS1_10 (<strong>2</strong>) - Donor variant at located at the
+3, +4 or +5 position of the intron - within the pathogenic range</li>
<li>ACMG_PS1 (<strong>4</strong>) - Same amino acid change as a
previously established pathogenic variant (ClinVar) regardless of
nucleotide change</li>
<li>ACMG_PP2 (<strong>0.5</strong>) - Missense variant in a gene that
has a relatively low rate of benign missense variation (&lt;30%) and
where missense variants are a common mechanism of disease (&gt;50% P/LP
(ClinVar))</li>
<li>ACMG_PM1 (<strong>2</strong>) - Missense variant in a somatic
mutation hotspot as determined by cancerhotspots.org</li>
<li>ACMG_PM4 (<strong>0</strong>) - Protein length changes due to
inframe indels or nonstop variant in non-repetitive regions of genes
that harbor variants with a dominant mode of inheritance.</li>
<li>ACMG_PPC1 (<strong>0</strong>) - Protein length changes due to
inframe indels or nonstop variant in non-repetitive regions of genes
that harbor variants with a recessive mode of inheritance.</li>
<li>ACMG_PM5 (<strong>1.5</strong>) - Novel missense change at an amino
acid residue where a different missense change determined to be
pathogenic has been seen before (ClinVar)</li>
<li>ACMG_PP3 (<strong>0.5</strong>)- Multiple lines (&gt;=5) of
computational evidence support a deleterious effect on the gene or gene
product (conservation, evolutionary, splicing impact, etc. - from dbNSFP
</li>
</ol>
<p>Currently, the <strong>CPSR_CLASSIFICATION</strong> is determined
based on the following ranges of pathogenicity scores:</p>
<p><br></p>
<div class="figure">
<img src="../img/cpsr_classification.png" alt=""><p class="caption">Classification Scores</p>
</div>
  </main>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by <a href="https://github.com/sigven" class="external-link">Sigve Nakken</a>,
<a href="https://github.com/pdiakumis" class="external-link">Peter Diakumis</a>.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a>
2.0.2.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
